


<!DOCTYPE html>
<html id="htmlId">
<head>
  <title>Coverage Report > TranscriptConsequencePrioritizer</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">org.cbioportal.genome_nexus.component.annotation</a>
</div>

<h1>Coverage Summary for Class: TranscriptConsequencePrioritizer (org.cbioportal.genome_nexus.component.annotation)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">TranscriptConsequencePrioritizer</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (5/5)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    78,6%
  </span>
  <span class="absValue">
    (66/84)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package org.cbioportal.genome_nexus.component.annotation;
&nbsp;
&nbsp;import org.cbioportal.genome_nexus.model.TranscriptConsequence;
&nbsp;import org.jetbrains.annotations.Nullable;
&nbsp;import org.springframework.stereotype.Component;
&nbsp;
&nbsp;import java.util.HashMap;
&nbsp;import java.util.List;
&nbsp;import java.util.Map;
&nbsp;
&nbsp;@Component
<b class="fc">&nbsp;public class TranscriptConsequencePrioritizer</b>
&nbsp;{
<b class="fc">&nbsp;    public static final Map&lt;String, Integer&gt; EFFECT_PRIORITY = initEffectPriority();</b>
&nbsp;
&nbsp;    @Nullable
&nbsp;    public TranscriptConsequence transcriptWithMostSevereConsequence(List&lt;TranscriptConsequence&gt; transcripts,
&nbsp;                                                                     String mostSevereConsequence)
&nbsp;    {
<b class="fc">&nbsp;        if (transcripts == null) {</b>
<b class="fc">&nbsp;            return null;</b>
&nbsp;        }
&nbsp;
<b class="nc">&nbsp;        Integer highestPriority = Integer.MAX_VALUE;</b>
<b class="nc">&nbsp;        TranscriptConsequence highestPriorityTranscript = null;</b>
&nbsp;
<b class="nc">&nbsp;        for (TranscriptConsequence transcript : transcripts)</b>
&nbsp;        {
<b class="nc">&nbsp;            List&lt;String&gt; consequenceTerms = transcript.getConsequenceTerms();</b>
&nbsp;
<b class="nc">&nbsp;            for (String consequenceTerm : consequenceTerms)</b>
&nbsp;            {
<b class="nc">&nbsp;                if (EFFECT_PRIORITY.getOrDefault(consequenceTerm.toLowerCase(), Integer.MAX_VALUE) &lt; highestPriority)</b>
&nbsp;                {
<b class="nc">&nbsp;                    highestPriorityTranscript = transcript;</b>
<b class="nc">&nbsp;                    highestPriority = EFFECT_PRIORITY.getOrDefault(consequenceTerm.toLowerCase(), Integer.MAX_VALUE);</b>
&nbsp;                }
&nbsp;
<b class="nc">&nbsp;                if (consequenceTerm.trim().equals(mostSevereConsequence)) {</b>
<b class="nc">&nbsp;                    return transcript;</b>
&nbsp;                }
<b class="nc">&nbsp;            }</b>
<b class="nc">&nbsp;        }</b>
&nbsp;
&nbsp;        // no match, pick one with the highest priority
<b class="nc">&nbsp;        if (highestPriorityTranscript != null) {</b>
<b class="nc">&nbsp;            return highestPriorityTranscript;</b>
&nbsp;        }
&nbsp;
&nbsp;        // if for whatever reason that is null, just return the first one.
<b class="nc">&nbsp;        if (transcripts.size() &gt; 0) {</b>
<b class="nc">&nbsp;            return transcripts.get(0);</b>
&nbsp;        }
&nbsp;
<b class="nc">&nbsp;        return null;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String pickHighestPriorityConsequence(List&lt;String&gt; consequences)
&nbsp;    {
<b class="fc">&nbsp;        if (consequences == null) {</b>
<b class="nc">&nbsp;            return null;</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        String highestPriorityConsequence = null;</b>
<b class="fc">&nbsp;        Integer highestPriority = Integer.MAX_VALUE;</b>
&nbsp;
<b class="fc">&nbsp;        for (String consequence : consequences)</b>
&nbsp;        {
<b class="fc">&nbsp;            if (EFFECT_PRIORITY.getOrDefault(consequence.toLowerCase(), Integer.MAX_VALUE) &lt; highestPriority)</b>
&nbsp;            {
<b class="fc">&nbsp;                highestPriorityConsequence = consequence;</b>
<b class="fc">&nbsp;                highestPriority = EFFECT_PRIORITY.getOrDefault(consequence.toLowerCase(), Integer.MAX_VALUE);</b>
&nbsp;            }
<b class="fc">&nbsp;        }</b>
&nbsp;
<b class="fc">&nbsp;        return highestPriorityConsequence;</b>
&nbsp;    }
&nbsp;
&nbsp;    /**
&nbsp;     * Prioritize Sequence Ontology terms in order of severity, as estimated by Ensembl.
&nbsp;     *
&nbsp;     * Ensembl: http://useast.ensembl.org/info/genome/variation/predicted_data.html#consequences
&nbsp;     * VCF2MAF mappings: https://github.com/mskcc/vcf2maf/blob/master/vcf2maf.pl (GetEffectPriority)
&nbsp;     */
&nbsp;    private static Map&lt;String, Integer&gt; initEffectPriority()
&nbsp;    {
<b class="fc">&nbsp;        Map&lt;String, Integer&gt; effectPriority = new HashMap&lt;&gt;();</b>
&nbsp;
<b class="fc">&nbsp;        effectPriority.put(&quot;transcript_ablation&quot;, 1); // A feature ablation whereby the deleted region includes a transcript feature</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;exon_loss_variant&quot;, 1); // A sequence variant whereby an exon is lost from the transcript</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;splice_donor_variant&quot;, 2); // A splice variant that changes the 2 base region at the 5&quot; end of an intron</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;splice_acceptor_variant&quot;, 2); // A splice variant that changes the 2 base region at the 3&quot; end of an intron</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;stop_gained&quot;, 3); // A sequence variant whereby at least one base of a codon is changed, resulting in a premature stop codon, leading to a shortened transcript</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;frameshift_variant&quot;, 3); // A sequence variant which causes a disruption of the translational reading frame, because the number of nucleotides inserted or deleted is not a multiple of three</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;stop_lost&quot;, 3); // A sequence variant where at least one base of the terminator codon (stop) is changed, resulting in an elongated transcript</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;start_lost&quot;, 4); // A codon variant that changes at least one base of the canonical start codon</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;initiator_codon_variant&quot;, 4); // A codon variant that changes at least one base of the first codon of a transcript</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;disruptive_inframe_insertion&quot;, 5); // An inframe increase in cds length that inserts one or more codons into the coding sequence within an existing codon</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;disruptive_inframe_deletion&quot;, 5); // An inframe decrease in cds length that deletes bases from the coding sequence starting within an existing codon</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;inframe_insertion&quot;, 5); // An inframe non synonymous variant that inserts bases into the coding sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;inframe_deletion&quot;, 5); // An inframe non synonymous variant that deletes bases from the coding sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;protein_altering_variant&quot;, 5); // A sequence variant which is predicted to change the protein encoded in the coding sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;missense_variant&quot;, 6); // A sequence variant, that changes one or more bases, resulting in a different amino acid sequence but where the length is preserved</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;conservative_missense_variant&quot;, 6); // A sequence variant whereby at least one base of a codon is changed resulting in a codon that encodes for a different but similar amino acid. These variants may or may not be deleterious</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;rare_amino_acid_variant&quot;, 6); // A sequence variant whereby at least one base of a codon encoding a rare amino acid is changed, resulting in a different encoded amino acid</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;transcript_amplification&quot;, 7); // A feature amplification of a region containing a transcript</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;splice_region_variant&quot;, 8); // A sequence variant in which a change has occurred within the region of the splice site, either within 1-3 bases of the exon or 3-8 bases of the intron</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;start_retained_variant&quot;, 9); // A sequence variant where at least one base in the start codon is changed, but the start remains</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;stop_retained_variant&quot;, 9); // A sequence variant where at least one base in the terminator codon is changed, but the terminator remains</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;synonymous_variant&quot;, 9); // A sequence variant where there is no resulting change to the encoded amino acid</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;incomplete_terminal_codon_variant&quot;, 10); // A sequence variant where at least one base of the final codon of an incompletely annotated transcript is changed</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;coding_sequence_variant&quot;, 11); // A sequence variant that changes the coding sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;mature_mirna_variant&quot;, 11); // A transcript variant located with the sequence of the mature miRNA</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;exon_variant&quot;, 11); // A sequence variant that changes exon sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;5_prime_utr_variant&quot;, 12); // A UTR variant of the 5&quot; UTR</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;5_prime_utr_premature_start_codon_gain_variant&quot;, 12); // snpEff-specific effect, creating a start codon in 5&quot; UTR</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;3_prime_utr_variant&quot;, 12); // A UTR variant of the 3&quot; UTR</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;non_coding_exon_variant&quot;, 13); // A sequence variant that changes non-coding exon sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;non_coding_transcript_exon_variant&quot;, 13); // snpEff-specific synonym for non_coding_exon_variant</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;non_coding_transcript_variant&quot;, 14); // A transcript variant of a non coding RNA gene</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;nc_transcript_variant&quot;, 14); // A transcript variant of a non coding RNA gene (older alias for non_coding_transcript_variant)</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;intron_variant&quot;, 14); // A transcript variant occurring within an intron</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;intragenic_variant&quot;, 14); // A variant that occurs within a gene but falls outside of all transcript features. This occurs when alternate transcripts of a gene do not share overlapping sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;intragenic&quot;, 14); // snpEff-specific synonym of intragenic_variant</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;nmd_transcript_variant&quot;, 15); // A variant in a transcript that is the target of NMD</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;upstream_gene_variant&quot;, 16); // A sequence variant located 5&quot; of a gene</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;downstream_gene_variant&quot;, 16); // A sequence variant located 3&quot; of a gene</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;tfbs_ablation&quot;, 17); // A feature ablation whereby the deleted region includes a transcription factor binding site</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;tfbs_amplification&quot;, 17); // A feature amplification of a region containing a transcription factor binding site</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;tf_binding_site_variant&quot;, 17); // A sequence variant located within a transcription factor binding site</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;regulatory_region_ablation&quot;, 17); // A feature ablation whereby the deleted region includes a regulatory region</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;regulatory_region_amplification&quot;, 17); // A feature amplification of a region containing a regulatory region</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;regulatory_region_variant&quot;, 17); // A sequence variant located within a regulatory region</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;regulatory_region&quot;, 17); // snpEff-specific effect that should really be regulatory_region_variant</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;feature_elongation&quot;, 18); // A sequence variant that causes the extension of a genomic feature, with regard to the reference sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;feature_truncation&quot;, 18); // A sequence variant that causes the reduction of a genomic feature, with regard to the reference sequence</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;intergenic_variant&quot;, 19); // A sequence variant located in the intergenic region, between genes</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;intergenic_region&quot;, 19); // snpEff-specific effect that should really be intergenic_variant</b>
<b class="fc">&nbsp;        effectPriority.put(&quot;&quot;, 20);</b>
&nbsp;
<b class="fc">&nbsp;        return effectPriority;</b>
&nbsp;    }
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2024-04-18 22:00</div>
</div>
</body>
</html>
