


<!DOCTYPE html>
<html id="htmlId">
<head>
  <title>Coverage Report > EnsemblController</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">org.cbioportal.genome_nexus.web</a>
</div>

<h1>Coverage Summary for Class: EnsemblController (org.cbioportal.genome_nexus.web)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">EnsemblController</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    18,2%
  </span>
  <span class="absValue">
    (2/11)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    28,6%
  </span>
  <span class="absValue">
    (4/14)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package org.cbioportal.genome_nexus.web;
&nbsp;
&nbsp;import io.swagger.annotations.Api;
&nbsp;import io.swagger.annotations.ApiOperation;
&nbsp;import io.swagger.annotations.ApiParam;
&nbsp;
&nbsp;import org.cbioportal.genome_nexus.model.EnsemblGene;
&nbsp;import org.cbioportal.genome_nexus.model.EnsemblTranscript;
&nbsp;import org.cbioportal.genome_nexus.model.GeneXref;
&nbsp;import org.cbioportal.genome_nexus.service.EnsemblService;
&nbsp;import org.cbioportal.genome_nexus.service.GeneXrefService;
&nbsp;import org.cbioportal.genome_nexus.service.exception.EnsemblTranscriptNotFoundException;
&nbsp;import org.cbioportal.genome_nexus.service.exception.EnsemblWebServiceException;
&nbsp;import org.cbioportal.genome_nexus.service.exception.NoEnsemblGeneIdForHugoSymbolException;
&nbsp;import org.cbioportal.genome_nexus.service.exception.NoEnsemblGeneIdForEntrezGeneIdException;
&nbsp;import org.cbioportal.genome_nexus.web.config.PublicApi;
&nbsp;import org.cbioportal.genome_nexus.web.param.EnsemblFilter;
&nbsp;import org.springframework.beans.factory.annotation.Autowired;
&nbsp;import org.springframework.web.bind.annotation.*;
&nbsp;
&nbsp;import java.util.List;
&nbsp;
&nbsp;@RestController // shorthand for @Controller, @ResponseBody
&nbsp;@CrossOrigin(origins=&quot;*&quot;) // allow all cross-domain requests
&nbsp;@RequestMapping(value= &quot;/&quot;)
&nbsp;@Api(tags = &quot;ensembl-controller&quot;, description = &quot;Ensembl Controller&quot;)
&nbsp;@PublicApi
&nbsp;public class EnsemblController
&nbsp;{
&nbsp;    private final EnsemblService ensemblService;
&nbsp;    private final GeneXrefService geneXrefService;
&nbsp;
&nbsp;    @Autowired
&nbsp;    public EnsemblController(EnsemblService ensemblService, GeneXrefService geneXrefService)
<b class="fc">&nbsp;    {</b>
<b class="fc">&nbsp;        this.ensemblService = ensemblService;</b>
<b class="fc">&nbsp;        this.geneXrefService = geneXrefService;</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves canonical Ensembl Gene ID by Hugo Symbols&quot;,
&nbsp;        nickname = &quot;fetchCanonicalEnsemblGeneIdByHugoSymbolsPOST&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/canonical-gene/hgnc&quot;,
&nbsp;        method = RequestMethod.POST,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public List&lt;EnsemblGene&gt; fetchCanonicalEnsemblGeneIdByHugoSymbolPOST(
&nbsp;        @ApiParam(value = &quot;List of Hugo Symbols. For example [\&quot;TP53\&quot;,\&quot;PIK3CA\&quot;,\&quot;BRCA1\&quot;]&quot;,
&nbsp;            required = true)
&nbsp;        @RequestBody List&lt;String&gt; hugoSymbols)
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getCanonicalEnsemblGeneIdByHugoSymbols(hugoSymbols);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves Ensembl canonical gene id by Hugo Symbol&quot;,
&nbsp;        nickname = &quot;fetchCanonicalEnsemblGeneIdByHugoSymbolGET&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/canonical-gene/hgnc/{hugoSymbol}&quot;,
&nbsp;        method = RequestMethod.GET,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public EnsemblGene fetchCanonicalEnsemblGeneIdByHugoSymbolGET(
&nbsp;        @ApiParam(value = &quot;A Hugo Symbol. For example TP53&quot;,
&nbsp;            required = true)
&nbsp;        @PathVariable String hugoSymbol) throws NoEnsemblGeneIdForHugoSymbolException
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getCanonicalEnsemblGeneIdByHugoSymbol(hugoSymbol);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves canonical Ensembl Gene ID by Entrez Gene Ids&quot;,
&nbsp;        nickname = &quot;fetchCanonicalEnsemblGeneIdByEntrezGeneIdsPOST&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/canonical-gene/entrez&quot;,
&nbsp;        method = RequestMethod.POST,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public List&lt;EnsemblGene&gt; fetchCanonicalEnsemblGeneIdByEntrezGeneIdPOST(
&nbsp;        @ApiParam(value = &quot;List of Entrez Gene Ids. For example [\&quot;23140\&quot;,\&quot;26009\&quot;,\&quot;100131879\&quot;]&quot;,
&nbsp;            required = true)
&nbsp;        @RequestBody List&lt;String&gt; entrezGeneIds)
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getCanonicalEnsemblGeneIdByEntrezGeneIds(entrezGeneIds);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves Ensembl canonical gene id by Entrez Gene Id&quot;,
&nbsp;        nickname = &quot;fetchCanonicalEnsemblGeneIdByEntrezGeneIdGET&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/canonical-gene/entrez/{entrezGeneId}&quot;,
&nbsp;        method = RequestMethod.GET,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public EnsemblGene fetchCanonicalEnsemblGeneIdByEntrezGeneIdGET(
&nbsp;        @ApiParam(value = &quot;An Entrez Gene Id. For example 23140&quot;,
&nbsp;            required = true)
&nbsp;        @PathVariable String entrezGeneId) throws NoEnsemblGeneIdForEntrezGeneIdException
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getCanonicalEnsemblGeneIdByEntrezGeneId(entrezGeneId);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves Ensembl Transcripts by protein ID, and gene ID. &quot; +
&nbsp;        &quot;Retrieves all transcripts in case no query parameter provided&quot;,
&nbsp;        nickname = &quot;fetchEnsemblTranscriptsGET&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/transcript&quot;,
&nbsp;        method = RequestMethod.GET,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public List&lt;EnsemblTranscript&gt; fetchEnsemblTranscriptsGET(
&nbsp;        @ApiParam(value = &quot;An Ensembl gene ID. For example ENSG00000136999&quot;)
&nbsp;        @RequestParam(required = false) String geneId,
&nbsp;        @ApiParam(value = &quot;An Ensembl protein ID. For example ENSP00000439985&quot;)
&nbsp;        @RequestParam(required = false) String proteinId,
&nbsp;        @ApiParam(value = &quot;A Hugo Symbol For example ARF5&quot;)
&nbsp;        @RequestParam(required = false) String hugoSymbol)
&nbsp;    {
<b class="nc">&nbsp;        return ensemblService.getEnsemblTranscripts(geneId, proteinId, hugoSymbol);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves Ensembl Transcripts by Ensembl transcript IDs, hugo Symbols, protein IDs, or gene IDs&quot;,
&nbsp;        nickname = &quot;fetchEnsemblTranscriptsByEnsemblFilterPOST&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/transcript&quot;,
&nbsp;        method = RequestMethod.POST,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public List&lt;EnsemblTranscript&gt; fetchEnsemblTranscriptsByEnsemblFilterPOST(
&nbsp;        @ApiParam(
&nbsp;            value = EnsemblFilter.TRANSCRIPT_ID_DESC + &quot;&lt;br&gt;OR&lt;br&gt;&quot; +
&nbsp;                EnsemblFilter.HUGO_SYMBOL_DESC + &quot;&lt;br&gt;OR&lt;br&gt;&quot; +
&nbsp;                EnsemblFilter.PROTEIN_ID_DESC + &quot;&lt;br&gt;OR&lt;br&gt;&quot; +
&nbsp;                EnsemblFilter.GENE_ID_DESC,
&nbsp;            required = true)
&nbsp;        @RequestBody EnsemblFilter ensemblFilter)
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getEnsemblTranscripts(</b>
<b class="nc">&nbsp;            ensemblFilter.getTranscriptIds(), ensemblFilter.getGeneIds(), ensemblFilter.getProteinIds(), ensemblFilter.getHugoSymbols());</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves the transcript by an Ensembl transcript ID&quot;,
&nbsp;        nickname = &quot;fetchEnsemblTranscriptByTranscriptIdGET&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/transcript/{transcriptId}&quot;,
&nbsp;        method = RequestMethod.GET,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public EnsemblTranscript fetchEnsemblTranscriptByTranscriptIdGET(
&nbsp;        @ApiParam(value = &quot;An Ensembl transcript ID. For example ENST00000361390&quot;,
&nbsp;            required = true)
&nbsp;        @PathVariable String transcriptId) throws EnsemblTranscriptNotFoundException
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getEnsemblTranscriptsByTranscriptId(transcriptId);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves Ensembl canonical transcripts by Hugo Symbols&quot;,
&nbsp;        nickname = &quot;fetchCanonicalEnsemblTranscriptsByHugoSymbolsPOST&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/canonical-transcript/hgnc&quot;,
&nbsp;        method = RequestMethod.POST,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public List&lt;EnsemblTranscript&gt; fetchCanonicalEnsemblTranscriptsByHugoSymbolPOST(
&nbsp;        @ApiParam(value = &quot;List of Hugo Symbols. For example [\&quot;TP53\&quot;,\&quot;PIK3CA\&quot;,\&quot;BRCA1\&quot;]&quot;,
&nbsp;            required = true)
&nbsp;        @RequestBody List&lt;String&gt; hugoSymbols,
&nbsp;        @ApiParam(value=&quot;Isoform override source. For example uniprot&quot;,
&nbsp;            defaultValue=&quot;uniprot&quot;,
&nbsp;            required = false)
&nbsp;        @RequestParam(required = false) String isoformOverrideSource)
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getCanonicalEnsemblTranscriptsByHugoSymbols(hugoSymbols, isoformOverrideSource);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Retrieves Ensembl canonical transcript by Hugo Symbol&quot;,
&nbsp;        nickname = &quot;fetchCanonicalEnsemblTranscriptByHugoSymbolGET&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/canonical-transcript/hgnc/{hugoSymbol}&quot;,
&nbsp;        method = RequestMethod.GET,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public EnsemblTranscript fetchCanonicalEnsemblTranscriptByHugoSymbolGET(
&nbsp;        @ApiParam(value = &quot;A Hugo Symbol. For example TP53&quot;,
&nbsp;            required = true)
&nbsp;        @PathVariable String hugoSymbol,
&nbsp;        @ApiParam(value=&quot;Isoform override source. For example uniprot&quot;,
&nbsp;            defaultValue=&quot;uniprot&quot;,
&nbsp;            required = false)
&nbsp;        @RequestParam(required = false) String isoformOverrideSource) throws EnsemblTranscriptNotFoundException
&nbsp;    {
<b class="nc">&nbsp;        return this.ensemblService.getCanonicalEnsemblTranscriptByHugoSymbol(hugoSymbol, isoformOverrideSource);</b>
&nbsp;    }
&nbsp;
&nbsp;    @ApiOperation(value = &quot;Perform lookups of Ensembl identifiers and retrieve their external references in other databases&quot;,
&nbsp;        nickname = &quot;fetchGeneXrefsGET&quot;)
&nbsp;    @RequestMapping(value = &quot;/ensembl/xrefs&quot;,
&nbsp;        method = RequestMethod.GET,
&nbsp;        produces = &quot;application/json&quot;)
&nbsp;    public List&lt;GeneXref&gt; fetchGeneXrefsGET(
&nbsp;        @ApiParam(value=&quot;Ensembl gene accession. For example ENSG00000169083&quot;,
&nbsp;            required = true)
&nbsp;        @RequestParam String accession) throws EnsemblWebServiceException
&nbsp;    {
<b class="fc">&nbsp;        return geneXrefService.getGeneXrefs(accession);</b>
&nbsp;    }
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2024-05-01 05:19</div>
</div>
</body>
</html>
