import  org.junit.AfterClass;
import  org.junit.BeforeClass;
import  org.junit.Before;
import  org.junit.Test;
import static org.junit.Assert.*;
import  java.util.Map;
import  java.util.List;
import static org.evomaster.client.java.controller.api.EMTestUtils.*;
import  org.evomaster.client.java.controller.SutHandler;
import  io.restassured.RestAssured;
import static io.restassured.RestAssured.given;
import  io.restassured.response.ValidatableResponse;
import static org.hamcrest.Matchers.*;
import  io.restassured.config.JsonConfig;
import  io.restassured.path.json.config.JsonPathConfig;
import static io.restassured.config.RedirectConfig.redirectConfig;
import static org.evomaster.client.java.controller.contentMatchers.NumberMatcher.*;
import static org.evomaster.client.java.controller.contentMatchers.StringMatcher.*;
import static org.evomaster.client.java.controller.contentMatchers.SubStringMatcher.*;
import static org.evomaster.client.java.controller.expect.ExpectationHandler.expectationHandler;
import  org.evomaster.client.java.controller.expect.ExpectationHandler;
import  io.restassured.path.json.JsonPath;
import  java.util.Arrays;




/**
 * This file was automatically generated by EvoMaster on 2024-05-29T13:58:32.579+02:00[Europe/Paris]
 * <br>
 * The generated test suite contains 151 tests
 * <br>
 * Covered targets: 5948
 * <br>
 * Used time: 6h 3m 55s
 * <br>
 * Needed budget for current results: 93%
 * <br>
 * This file contains one example of each category of fault. The test cases in this file are a subset of the set of test cases likely to indicate faults.
 */
public class EvoMaster_fault_representatives_Test {

    
    private static final SutHandler controller = new em.embedded.org.cbioportal.genome_nexus.EmbeddedEvoMasterController();
    private static String baseUrlOfSut;
    /** [ems] - expectations master switch - is the variable that activates/deactivates expectations individual test cases
    * by default, expectations are turned off. The variable needs to be set to [true] to enable expectations
    */
    private static boolean ems = false;
    /**
    * sco - supported code oracle - checking that the response status code is among those supported according to the schema
    */
    private static boolean sco = false;
    /**
    * rso - response structure oracle - checking that the response objects match the responses defined in the schema
    */
    private static boolean rso = false;
    
    
    @BeforeClass
    public static void initClass() {
        controller.setupForGeneratedTest();
        baseUrlOfSut = controller.startSut();
        controller.registerOrExecuteInitSqlCommandsIfNeeded();
        assertNotNull(baseUrlOfSut);
        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
        RestAssured.useRelaxedHTTPSValidation();
        RestAssured.urlEncodingEnabled = false;
        RestAssured.config = RestAssured.config()
            .jsonConfig(JsonConfig.jsonConfig().numberReturnType(JsonPathConfig.NumberReturnType.DOUBLE))
            .redirect(redirectConfig().followRedirects(false));
    }
    
    
    @AfterClass
    public static void tearDown() {
        controller.stopSut();
    }
    
    
    @Before
    public void initTest() {
        controller.resetStateOfSUT();
    }
    
    
    
    
    @Test(timeout = 60000)
    public void test_0() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        try{
            given().accept("*/*")
                    .header("x-EMextraHeader123", "")
                    .get(baseUrlOfSut + "/nucleotide_context/8:ga140452136A%3ET");
        } catch(Exception e){
        }
    }
    
    
    @Test(timeout = 60000)
    public void test_1() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        try{
            given().accept("*/*")
                    .header("x-EMextraHeader123", "42")
                    .contentType("application/json")
                    .body(" [ " + 
                        " { " + 
                        " \"chromosome\": \"7\", " + 
                        " \"start\": 132064528, " + 
                        " \"end\": 140469520, " + 
                        " \"referenceAllele\": \"_EM_1500_XYZ_\", " + 
                        " \"variantAllele\": \"T\" " + 
                        " } " + 
                        " ] ")
                    .post(baseUrlOfSut + "/annotation/genomic?fields=_EM_1499_XYZ_&fields=nucleotide_context");
        } catch(Exception e){
        }
    }
    
    
    @Test(timeout = 60000)
    public void test_2() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        try{
            given().accept("*/*")
                    .header("x-EMextraHeader123", "42")
                    .contentType("application/json")
                    .body(" [ " + 
                        " { " + 
                        " \"chromosome\": \"7\", " + 
                        " \"start\": 140457289, " + 
                        " \"end\": 174007625, " + 
                        " \"referenceAllele\": \"T\", " + 
                        " \"variantAllele\": \"_EM_1698_XYZ_\" " + 
                        " }, " + 
                        " { " + 
                        " \"chromosome\": \"17\", " + 
                        " \"start\": 37880220, " + 
                        " \"end\": 38404508, " + 
                        " \"referenceAllele\": \"_EM_1699_XYZ_\", " + 
                        " \"variantAllele\": \"C\" " + 
                        " }, " + 
                        " { " + 
                        " \"chromosome\": \"17\", " + 
                        " \"start\": 37879794, " + 
                        " \"end\": 37879794, " + 
                        " \"referenceAllele\": \"G\", " + 
                        " \"variantAllele\": \"C\" " + 
                        " }, " + 
                        " { " + 
                        " \"chromosome\": \"12\", " + 
                        " \"start\": 25390092, " + 
                        " \"end\": 25398285, " + 
                        " \"referenceAllele\": \"CC\", " + 
                        " \"variantAllele\": \"GA\" " + 
                        " }, " + 
                        " { " + 
                        " \"chromosome\": \"12\", " + 
                        " \"start\": 25380271, " + 
                        " \"end\": 25380271, " + 
                        " \"referenceAllele\": \"C\", " + 
                        " \"variantAllele\": \"T\" " + 
                        " }, " + 
                        " { " + 
                        " \"chromosome\": \"10\", " + 
                        " \"start\": 89692940, " + 
                        " \"end\": 89692940, " + 
                        " \"referenceAllele\": \"C\", " + 
                        " \"variantAllele\": \"T\" " + 
                        " } " + 
                        " ] ")
                    .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42");
        } catch(Exception e){
        }
    }
    
    
    @Test(timeout = 60000)
    public void test_3() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .get(baseUrlOfSut + "/v2/api-docs")
                .then()
                .statusCode(200);
        
        expectationHandler.expect(ems)
            /*
             Note: No supported codes appear to be defined. https://swagger.io/docs/specification/describing-responses/.
             This is somewhat unexpected, so the code below is likely to lead to a failed expectation
            */
            .that(sco, Arrays.asList().contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_4() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_177_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_172_XYZ_\", " + 
                    " \"_EM_173_XYZ_\", " + 
                    " \"_EM_174_XYZ_\", " + 
                    " \"_EM_175_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/pfam/domain")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_5() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation?" + 
                    "hugoGeneSymbol=BRCA2&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_6() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/cancer_hotspots/transcript/ENST00000288602?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_7() throws Exception {
        
        given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript/_EM_36_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .body(isEmptyOrNullString());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_8() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript?EMextraParam123=_EM_3160_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_9() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_975_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript?geneId=_EM_972_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_10() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_185_XYZ_")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [ " + 
                    " \"EZghwoXnX\", " + 
                    " \"_EM_180_XYZ_\", " + 
                    " \"_EM_181_XYZ_\", " + 
                    " \"_EM_182_XYZ_\", " + 
                    " \"_EM_183_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/signal/mutation")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_11() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/version")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'genomeNexus'.'server'.'version'", containsString("1.0.2"))
                .body("'genomeNexus'.'server'.'static'", equalTo(true))
                .body("'genomeNexus'.'database'.'version'", containsString("3.6.2"))
                .body("'genomeNexus'.'database'.'static'", equalTo(true))
                .body("'vep'.'server'.'version'", containsString("NA"))
                .body("'vep'.'server'.'static'", equalTo(false))
                .body("'vep'.'cache'.'version'", containsString("NA"))
                .body("'vep'.'cache'.'static'", equalTo(false))
                .body("'vep'.'comment'", containsString("VEP annotations are currently externally sourced from ENSEMBL. Results are subject to change without notice."));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_12() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_62_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_59_XYZ_&" + 
                    "proteinId=_EM_60_XYZ_&" + 
                    "EMextraParam123=_EM_61_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_13() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/ptm/experimental?" + 
                    "ensemblTranscriptId=ENST00000371953&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_14() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript?proteinId=_EM_393_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_15() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_1238_XYZ_&" + 
                    "hugoSymbol=_EM_1240_XYZ_&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_16() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_511_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_507_XYZ_&" + 
                    "proteinId=_EM_508_XYZ_&" + 
                    "hugoSymbol=_EM_509_XYZ_&" + 
                    "EMextraParam123=_EM_510_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_17() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?" + 
                    "keyword=&" + 
                    "limit=243")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_18() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_1946_XYZ_")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=&" + 
                    "limit=1048827&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_19() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "hugoSymbol=_EM_722_XYZ_&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_20() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=&" + 
                    "limit=-16776309")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_21() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "proteinId=_EM_3423_XYZ_&" + 
                    "hugoSymbol=&" + 
                    "EMextraParam123=_EM_3160_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_22() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"transcriptIds\": [ " + 
                    " \"ENST00000420316\", " + 
                    " \"ENST00000646891\", " + 
                    " \"ENST00000371953\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ptm/experimental?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_23() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" { " + 
                    " \"geneIds\": [ " + 
                    " \"\" " + 
                    " ], " + 
                    " \"hugoSymbols\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_24() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" { " + 
                    " \"geneIds\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript?EMextraParam123=_EM_165_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_25() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"geneIds\": [ " + 
                    " \"_EM_383_XYZ_\", " + 
                    " \"y3\" " + 
                    " ], " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_384_XYZ_\", " + 
                    " \"_EM_385_XYZ_\" " + 
                    " ], " + 
                    " \"proteinIds\": [ " + 
                    " \"_EM_386_XYZ_\", " + 
                    " \"_EM_387_XYZ_\", " + 
                    " \"_EM_388_XYZ_\", " + 
                    " \"_EM_389_XYZ_\", " + 
                    " \"_EM_390_XYZ_\" " + 
                    " ], " + 
                    " \"transcriptIds\": [ " + 
                    " \"hWP\", " + 
                    " \"_EM_391_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_26() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"transcriptIds\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_27() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_1198_XYZ_")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_1191_XYZ_\", " + 
                    " \"_EM_1192_XYZ_\" " + 
                    " ], " + 
                    " \"proteinIds\": [ " + 
                    " \"_EM_1193_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_28() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_29() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_14595_XYZ_\", " + 
                    " \"Y4qjXQiAGBS9OaE\", " + 
                    " \"_EM_14596_XYZ_\", " + 
                    " \"ICTWRTLCsey\", " + 
                    " \"_EM_14597_XYZ_\" " + 
                    " ], " + 
                    " \"proteinIds\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_30() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_84_XYZ_")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=_EM_82_XYZ_&" + 
                    "EMextraParam123=_EM_83_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_31() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=VDumZ&" + 
                    "limit=907&" + 
                    "EMextraParam123=_EM_3162_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_32() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"ENST00000288602\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/transcript?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_33() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"178\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-gene/entrez")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_34() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=c.&" + 
                    "limit=7337765")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'queryType'", containsString("HGVSG"))
                .body("[0].'results'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_35() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=g.&" + 
                    "limit=1077377488")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'queryType'", containsString("HGVSG"))
                .body("[0].'results'.size()", equalTo(1))
                .body("[0].'results'[0]", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_36() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"transcriptId\": \"ENST00000397752\", " + 
                    " \"start\": 1009, " + 
                    " \"end\": 1009, " + 
                    " \"mutationType\": \"Splice_Region\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/proteinLocations?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'proteinLocation'.'transcriptId'", containsString("ENST00000397752"))
                .body("[0].'proteinLocation'.'start'", numberMatches(1009.0))
                .body("[0].'proteinLocation'.'end'", numberMatches(1009.0))
                .body("[0].'proteinLocation'.'mutationType'", containsString("Splice_Region"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_37() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_95_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_90_XYZ_\", " + 
                    " \"Mm_FEzhWfJEJ\", " + 
                    " \"_EM_91_XYZ_\", " + 
                    " \"_EM_92_XYZ_\", " + 
                    " \"_EM_93_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-gene/hgnc")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_38() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_100_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_96_XYZ_\", " + 
                    " \"\", " + 
                    " \"_EM_97_XYZ_\", " + 
                    " \"_EM_98_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-transcript/hgnc?" + 
                    "isoformOverrideSource=uniprot&" + 
                    "EMextraParam123=_EM_99_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_39() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?keyword=_EM_77_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_40() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?" + 
                    "keyword=_EM_455_XYZ_&" + 
                    "limit=969&" + 
                    "EMextraParam123=_EM_456_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_41() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?" + 
                    "keyword=dtBV&" + 
                    "limit=-33685444&" + 
                    "EMextraParam123=_EM_2453_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_42() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/xrefs?accession=ENSG00000135999")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(9))
                .body("[0].'display_id'", containsString("OTTHUMG00000153739"))
                .body("[0].'primary_id'", containsString("OTTHUMG00000153739"))
                .body("[0].'version'", containsString("3"))
                .body("[0].'dbname'", containsString("OTTG"))
                .body("[0].'synonyms'.size()", equalTo(0))
                .body("[0].'info_text'", containsString(""))
                .body("[0].'info_types'", containsString("NONE"))
                .body("[0].'db_display_name'", containsString("Havana gene"))
                .body("[1].'display_id'", containsString("Hs.23270"))
                .body("[1].'primary_id'", containsString("Hs.23270"))
                .body("[1].'version'", containsString("0"))
                .body("[1].'description'", containsString("Enhancer of polycomb homolog 2 (Drosophila)"))
                .body("[1].'dbname'", containsString("UniGene"))
                .body("[1].'synonyms'.size()", equalTo(0))
                .body("[1].'info_text'", containsString(""))
                .body("[1].'info_types'", containsString("SEQUENCE_MATCH"))
                .body("[1].'db_display_name'", containsString("UniGene"))
                .body("[2].'display_id'", containsString(" ENHANCER OF POLYCOMB, DROSOPHILA,  [*611000]"))
                .body("[2].'primary_id'", containsString("611000"))
                .body("[2].'version'", containsString("0"))
                .body("[2].'description'", containsString(" ENHANCER OF POLYCOMB, DROSOPHILA, HOMOLOG OF, 2; EPC2"))
                .body("[2].'dbname'", containsString("MIM_GENE"))
                .body("[2].'synonyms'.size()", equalTo(0))
                .body("[2].'info_text'", containsString(""))
                .body("[2].'info_types'", containsString("DEPENDENT"))
                .body("[2].'db_display_name'", containsString("MIM gene"))
                ; // Skipping assertions on the remaining 6 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(4)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(5)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(6)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(7)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(8)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")));
    }
    
    
    @Test(timeout = 60000)
    public void test_43() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/10:g.89624230_89624231delAC?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_44() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/17:g.7574003_7574004insGAACATCTCGAAGCGCTCACGCCCAC?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_45() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/13:g.32893378C%3ET?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_46() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/cancer_hotspots/hgvs/_EM_32_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_47() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/dbsnp/COSM8?" + 
                    "isoformOverrideSource=K65IhlGCE&" + 
                    "token=GuvZluq2&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("COSM8"))
                .body("'originalVariantQuery'", containsString("COSM8"))
                .body("'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_48() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_23881_XYZ_\", " + 
                    " \"\", " + 
                    " \"_EM_23882_XYZ_\", " + 
                    " \"\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/my_variant_info/variant")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_49() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/pdb/header/2npp")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'pdbId'", containsString("2npp"))
                .body("'title'", containsString("structure of the protein phosphatase 2a holoenzyme"))
                .body("'compound'.'1'.'chain'.size()", equalTo(2))
                .body("'compound'.'1'.'chain'", hasItems("a", "d"))
                .body("'compound'.'1'.'fragment'", containsString("scaffolding subunit"))
                .body("'compound'.'1'.'synonym'", containsString("protein phosphatase 2a"))
                .body("'compound'.'1'.'engineered'", containsString("yes"))
                .body("'compound'.'1'.'molecule'", containsString("protein phosphatase 2, regulatory subunit a (pr 65), alpha isoform"))
                .body("'compound'.'1'.'mol_id'", containsString("1"))
                .body("'compound'.'2'.'chain'.size()", equalTo(2))
                .body("'compound'.'2'.'chain'", hasItems("b", "e"))
                .body("'compound'.'2'.'synonym'", containsString("pp2a, b subunit, b' gamma isoform; pp2a, b subunit, b56 gamma isoform; pp2a, b subunit, pr61 gamma isoform; pp2a, b subunit, r5 gamma isoform; ny-ren-29 antigen"))
                .body("'compound'.'2'.'engineered'", containsString("yes"))
                .body("'compound'.'2'.'molecule'", containsString("serine/threonine-protein phosphatase 2a 56 kda regulatory subunit gamma isoform"))
                .body("'compound'.'2'.'mol_id'", containsString("2"))
                .body("'compound'.'3'.'chain'.size()", equalTo(2))
                .body("'compound'.'3'.'chain'", hasItems("c", "f"))
                .body("'compound'.'3'.'fragment'", containsString("catalytic subunit"))
                .body("'compound'.'3'.'synonym'", containsString("serine/threonine protein phosphatase; pp2a-alpha"))
                .body("'compound'.'3'.'engineered'", containsString("yes"))
                .body("'compound'.'3'.'molecule'", containsString("serine/threonine-protein phosphatase 2a catalytic subunit alpha isoform"))
                .body("'compound'.'3'.'mol_id'", containsString("3"))
                .body("'compound'.'3'.'ec'", containsString("3.1.3.16"))
                .body("'compound'.'4'.'chain'.size()", equalTo(2))
                .body("'compound'.'4'.'chain'", hasItems("x", "y"))
                .body("'compound'.'4'.'molecule'", containsString("microcystin lr"))
                .body("'compound'.'4'.'mol_id'", containsString("4"))
                .body("'source'.'1'.'expression_system_plasmid'", containsString("pgex-2t"))
                .body("'source'.'1'.'gene'.size()", equalTo(1))
                .body("'source'.'1'.'gene'", hasItems("ppp2r1a"))
                .body("'source'.'1'.'organism_scientific'", containsString("homo sapiens"))
                .body("'source'.'1'.'organism_common'", containsString("human"))
                .body("'source'.'1'.'expression_system_vector_type'", containsString("plasmid"))
                .body("'source'.'1'.'organism_taxid'", containsString("9606"))
                .body("'source'.'1'.'expression_system_strain'", containsString("bl21(de3)"))
                .body("'source'.'1'.'expression_system_taxid'", containsString("469008"))
                .body("'source'.'1'.'expression_system'", containsString("escherichia coli"))
                .body("'source'.'1'.'mol_id'", containsString("1"))
                .body("'source'.'2'.'expression_system_plasmid'", containsString("pgex-2t"))
                .body("'source'.'2'.'gene'.size()", equalTo(2))
                .body("'source'.'2'.'gene'", hasItems("ppp2r5c", "kiaa0044"))
                .body("'source'.'2'.'organism_scientific'", containsString("homo sapiens"))
                .body("'source'.'2'.'organism_common'", containsString("human"))
                .body("'source'.'2'.'expression_system_vector_type'", containsString("plasmid"))
                .body("'source'.'2'.'organism_taxid'", containsString("9606"))
                .body("'source'.'2'.'expression_system_strain'", containsString("bl21(de3)"))
                .body("'source'.'2'.'expression_system_taxid'", containsString("469008"))
                .body("'source'.'2'.'expression_system'", containsString("escherichia coli"))
                .body("'source'.'2'.'mol_id'", containsString("2"))
                .body("'source'.'3'.'gene'.size()", equalTo(1))
                .body("'source'.'3'.'gene'", hasItems("ppp2ca"))
                .body("'source'.'3'.'organism_scientific'", containsString("homo sapiens"))
                .body("'source'.'3'.'organism_common'", containsString("human"))
                .body("'source'.'3'.'expression_system_vector_type'", containsString("baculovirus"))
                .body("'source'.'3'.'organism_taxid'", containsString("9606"))
                .body("'source'.'3'.'expression_system_common'", containsString("fall armyworm"))
                .body("'source'.'3'.'expression_system_strain'", containsString("hi-5"))
                .body("'source'.'3'.'expression_system_taxid'", containsString("7108"))
                .body("'source'.'3'.'expression_system'", containsString("spodoptera frugiperda"))
                .body("'source'.'3'.'mol_id'", containsString("3"))
                .body("'source'.'4'.'synthetic'", containsString("yes"))
                .body("'source'.'4'.'organism_scientific'", containsString("cyanobacteria"))
                .body("'source'.'4'.'mol_id'", containsString("4"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_50() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/pdb/header/1yy9?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'pdbId'", containsString("1yy9"))
                .body("'title'", containsString("structure of the extracellular domain of the epidermal growth factor receptor in complex with the fab fragment of cetuximab/erbitux/imc-c225"))
                .body("'compound'.'1'.'chain'.size()", equalTo(1))
                .body("'compound'.'1'.'chain'", hasItems("a"))
                .body("'compound'.'1'.'fragment'", containsString("unp residues 25-642"))
                .body("'compound'.'1'.'synonym'", containsString("receptor tyrosine-protein kinase erbb-1"))
                .body("'compound'.'1'.'engineered'", containsString("yes"))
                .body("'compound'.'1'.'molecule'", containsString("epidermal growth factor receptor"))
                .body("'compound'.'1'.'mol_id'", containsString("1"))
                .body("'compound'.'1'.'ec'", containsString("2.7.10.1"))
                .body("'compound'.'2'.'chain'.size()", equalTo(1))
                .body("'compound'.'2'.'chain'", hasItems("c"))
                .body("'compound'.'2'.'engineered'", containsString("yes"))
                .body("'compound'.'2'.'molecule'", containsString("cetuximab fab light chain"))
                .body("'compound'.'2'.'mol_id'", containsString("2"))
                .body("'compound'.'3'.'chain'.size()", equalTo(1))
                .body("'compound'.'3'.'chain'", hasItems("d"))
                .body("'compound'.'3'.'molecule'", containsString("cetuximab fab heavy chain"))
                .body("'compound'.'3'.'mol_id'", containsString("3"))
                .body("'source'.'1'.'gene'.size()", equalTo(2))
                .body("'source'.'1'.'gene'", hasItems("egfr", "erbb1"))
                .body("'source'.'1'.'organism_scientific'", containsString("homo sapiens"))
                .body("'source'.'1'.'organism_common'", containsString("human"))
                .body("'source'.'1'.'expression_system_vector_type'", containsString("baculovirus"))
                .body("'source'.'1'.'organism_taxid'", containsString("9606"))
                .body("'source'.'1'.'expression_system_common'", containsString("fall armyworm"))
                .body("'source'.'1'.'expression_system_atcc_number'", containsString("sf9"))
                .body("'source'.'1'.'expression_system_taxid'", containsString("7108"))
                .body("'source'.'1'.'expression_system'", containsString("spodoptera frugiperda"))
                .body("'source'.'1'.'mol_id'", containsString("1"))
                .body("'source'.'2'.'strain'", containsString(","))
                .body("'source'.'2'.'expression_system_plasmid'", containsString("pdhl2"))
                .body("'source'.'2'.'organism_common'", containsString("house mouse, human"))
                .body("'source'.'2'.'expression_system_vector_type'", containsString("plasmid"))
                .body("'source'.'2'.'expression_system_cell'", containsString("mouse myeloma cell line"))
                .body("'source'.'2'.'expression_system'", containsString("mus musculus"))
                .body("'source'.'2'.'organism_scientific'", containsString("mus musculus, homo sapiens"))
                .body("'source'.'2'.'organism_taxid'", containsString("10090,9606"))
                .body("'source'.'2'.'expression_system_common'", containsString("house mouse"))
                .body("'source'.'2'.'expression_system_strain'", containsString("sp2/0-ag15"))
                .body("'source'.'2'.'expression_system_atcc_number'", containsString("crl-1581"))
                .body("'source'.'2'.'expression_system_taxid'", containsString("10090"))
                .body("'source'.'2'.'mol_id'", containsString("2"))
                .body("'source'.'3'.'strain'", containsString(","))
                .body("'source'.'3'.'organism_scientific'", containsString("mus musculus, homo sapiens"))
                .body("'source'.'3'.'organism_common'", containsString("house mouse, human"))
                .body("'source'.'3'.'expression_system_vector_type'", containsString("plasmid"))
                .body("'source'.'3'.'organism_taxid'", containsString("10090,9606"))
                .body("'source'.'3'.'expression_system_common'", containsString("house mouse"))
                .body("'source'.'3'.'expression_system_strain'", containsString("sp2/0-ag15"))
                .body("'source'.'3'.'expression_system_atcc_number'", containsString("crl-1581"))
                .body("'source'.'3'.'expression_system_cell'", containsString("mouse myeloma cell line"))
                .body("'source'.'3'.'expression_system_taxid'", containsString("10090"))
                .body("'source'.'3'.'expression_system'", containsString("mus musculus"))
                .body("'source'.'3'.'mol_id'", containsString("3"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_51() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_3193_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_3188_XYZ_\", " + 
                    " \"_EM_3189_XYZ_\", " + 
                    " \"_EM_3190_XYZ_\", " + 
                    " \"nA8TR8ibU8E\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/nucleotide_context")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_52() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_197_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_192_XYZ_\", " + 
                    " \"_EM_193_XYZ_\", " + 
                    " \"_EM_194_XYZ_\", " + 
                    " \"_EM_195_XYZ_\", " + 
                    " \"_EM_196_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/mutation_assessor?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_53() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"\", " + 
                    " \"jn7KZ0OejSxICpmX\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/hgvs")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString(""))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'variant'", containsString("jn7KZ0OejSxICpmX"))
                .body("[1].'hotspots'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_54() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"1a37\", " + 
                    " \"1a4o\", " + 
                    " \"3njp\", " + 
                    " \"1yy9\", " + 
                    " \"INVALID\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/pdb/header?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(4))
                .body("[0].'pdbId'", containsString("1a37"))
                .body("[0].'title'", containsString("14-3-3 protein zeta bound to ps-raf259 peptide"))
                .body("[0].'compound'.'1'.'other_details'", containsString("complexed with phosphoserine-containing peptide derived from raf"))
                .body("[0].'compound'.'1'.'chain'.size()", equalTo(2))
                .body("[0].'compound'.'1'.'chain'", hasItems("a", "b"))
                .body("[0].'compound'.'1'.'engineered'", containsString("yes"))
                .body("[0].'compound'.'1'.'molecule'", containsString("14-3-3 protein zeta"))
                .body("[0].'compound'.'1'.'mol_id'", containsString("1"))
                .body("[0].'compound'.'2'.'chain'.size()", equalTo(2))
                .body("[0].'compound'.'2'.'chain'", hasItems("p", "q"))
                .body("[0].'compound'.'2'.'molecule'", containsString("ps-raf259 peptide lsqrqrst(sep)tpnvhm"))
                .body("[0].'compound'.'2'.'mol_id'", containsString("2"))
                .body("[0].'source'.'1'.'organism_scientific'", containsString("bos taurus"))
                .body("[0].'source'.'1'.'organism_common'", containsString("cattle"))
                .body("[0].'source'.'1'.'organism_taxid'", containsString("9913"))
                .body("[0].'source'.'1'.'expression_system_taxid'", containsString("562"))
                .body("[0].'source'.'1'.'expression_system'", containsString("escherichia coli"))
                .body("[0].'source'.'1'.'mol_id'", containsString("1"))
                .body("[1].'pdbId'", containsString("1a4o"))
                .body("[1].'title'", containsString("14-3-3 protein zeta isoform"))
                .body("[1].'compound'.'1'.'chain'.size()", equalTo(4))
                .body("[1].'compound'.'1'.'chain'", hasItems("a", "b", "c", "d"))
                .body("[1].'compound'.'1'.'molecule'", containsString("14-3-3 protein zeta"))
                .body("[1].'compound'.'1'.'mol_id'", containsString("1"))
                .body("[1].'source'.'1'.'organism_scientific'", containsString("bos taurus"))
                .body("[1].'source'.'1'.'organism_common'", containsString("cattle"))
                .body("[1].'source'.'1'.'organism_taxid'", containsString("9913"))
                .body("[1].'source'.'1'.'expression_system'", containsString("escherichia coli"))
                .body("[1].'source'.'1'.'mol_id'", containsString("1"))
                .body("[2].'pdbId'", containsString("3njp"))
                .body("[2].'title'", containsString("the extracellular and transmembrane domain interfaces in epidermal growth factor receptor signaling"))
                .body("[2].'compound'.'1'.'chain'.size()", equalTo(2))
                .body("[2].'compound'.'1'.'chain'", hasItems("a", "b"))
                .body("[2].'compound'.'1'.'synonym'", containsString("receptor tyrosine-protein kinase erbb-1, proto-oncogene c- erbb-1"))
                .body("[2].'compound'.'1'.'engineered'", containsString("yes"))
                .body("[2].'compound'.'1'.'molecule'", containsString("epidermal growth factor receptor"))
                .body("[2].'compound'.'1'.'mol_id'", containsString("1"))
                .body("[2].'compound'.'1'.'ec'", containsString("2.7.10.1"))
                .body("[2].'compound'.'2'.'chain'.size()", equalTo(2))
                .body("[2].'compound'.'2'.'chain'", hasItems("c", "d"))
                .body("[2].'compound'.'2'.'molecule'", containsString("epidermal growth factor"))
                .body("[2].'compound'.'2'.'mol_id'", containsString("2"))
                .body("[2].'source'.'1'.'expression_system_plasmid'", containsString("pcdna3.1/zeo(+)"))
                .body("[2].'source'.'1'.'gene'.size()", equalTo(2))
                .body("[2].'source'.'1'.'gene'", hasItems("egfr", "erbb1"))
                .body("[2].'source'.'1'.'organism_scientific'", containsString("homo sapiens"))
                .body("[2].'source'.'1'.'organism_common'", containsString("human"))
                .body("[2].'source'.'1'.'expression_system_vector_type'", containsString("plasmid"))
                .body("[2].'source'.'1'.'organism_taxid'", containsString("9606"))
                .body("[2].'source'.'1'.'expression_system_common'", containsString("chinese hamster"))
                .body("[2].'source'.'1'.'expression_system_strain'", containsString("lec8"))
                .body("[2].'source'.'1'.'expression_system_organ'", containsString("ovary"))
                .body("[2].'source'.'1'.'expression_system_taxid'", containsString("10029"))
                .body("[2].'source'.'1'.'expression_system'", containsString("cricetulus griseus"))
                .body("[2].'source'.'1'.'mol_id'", containsString("1"))
                .body("[2].'source'.'2'.'gene'.size()", equalTo(1))
                .body("[2].'source'.'2'.'gene'", hasItems("egf"))
                .body("[2].'source'.'2'.'organism_scientific'", containsString("homo sapiens"))
                .body("[2].'source'.'2'.'organism_common'", containsString("human"))
                .body("[2].'source'.'2'.'organism_taxid'", containsString("9606"))
                .body("[2].'source'.'2'.'expression_system_common'", containsString("chinese hamster"))
                .body("[2].'source'.'2'.'expression_system_taxid'", containsString("10029"))
                .body("[2].'source'.'2'.'expression_system'", containsString("cricetulus griseus"))
                .body("[2].'source'.'2'.'mol_id'", containsString("2"))
                ; // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_55() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/my_variant_info/variant/17:g.41276045_41276046del?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("chr17:g.41276045_41276046del"))
                .body("'query'", containsString("chr17:g.41276045_41276046del"))
                .body("'hgvs'", containsString("chr17:g.41276045_41276046del"))
                .body("'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("'vcf'.'alt'", containsString("A"))
                .body("'vcf'.'position'", containsString("41276044"))
                .body("'vcf'.'ref'", containsString("ACT"))
                .body("'gnomadExome'.'alleleCount'.'ac'", numberMatches(57.0))
                .body("'gnomadExome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_amr'", numberMatches(1.0))
                .body("'gnomadExome'.'alleleCount'.'ac_asj'", numberMatches(41.0))
                .body("'gnomadExome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_nfe'", numberMatches(11.0))
                .body("'gnomadExome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_sas'", numberMatches(4.0))
                .body("'gnomadExome'.'alleleNumber'.'an'", numberMatches(251050.0))
                .body("'gnomadExome'.'alleleNumber'.'an_afr'", numberMatches(16250.0))
                .body("'gnomadExome'.'alleleNumber'.'an_amr'", numberMatches(34592.0))
                .body("'gnomadExome'.'alleleNumber'.'an_asj'", numberMatches(10078.0))
                .body("'gnomadExome'.'alleleNumber'.'an_eas'", numberMatches(18394.0))
                .body("'gnomadExome'.'alleleNumber'.'an_fin'", numberMatches(21644.0))
                .body("'gnomadExome'.'alleleNumber'.'an_nfe'", numberMatches(113354.0))
                .body("'gnomadExome'.'alleleNumber'.'an_oth'", numberMatches(6130.0))
                .body("'gnomadExome'.'alleleNumber'.'an_sas'", numberMatches(30608.0))
                .body("'gnomadExome'.'alleleFrequency'.'af'", numberMatches(2.27046E-4))
                .body("'gnomadExome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_amr'", numberMatches(2.89084E-5))
                .body("'gnomadExome'.'alleleFrequency'.'af_asj'", numberMatches(0.00406827))
                .body("'gnomadExome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_nfe'", numberMatches(9.70411E-5))
                .body("'gnomadExome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_sas'", numberMatches(1.30685E-4))
                .body("'gnomadExome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_sas'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac'", numberMatches(1.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_amr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_asj'", numberMatches(1.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_nfe'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_sas'", nullValue())
                .body("'gnomadGenome'.'alleleNumber'.'an'", numberMatches(31392.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_afr'", numberMatches(8712.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_amr'", numberMatches(848.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_asj'", numberMatches(290.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_eas'", numberMatches(1560.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_fin'", numberMatches(3468.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_nfe'", numberMatches(15426.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_oth'", numberMatches(1088.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_sas'", nullValue())
                .body("'gnomadGenome'.'alleleFrequency'.'af'", numberMatches(3.18552E-5))
                .body("'gnomadGenome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_amr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_asj'", numberMatches(0.00344828))
                .body("'gnomadGenome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_nfe'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_sas'", nullValue())
                .body("'gnomadGenome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_sas'", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_56() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/dbsnp/rs626?" + 
                    "isoformOverrideSource=genome_nexus&" + 
                    "token=%7B%0A++++++++++++++++++++%22_EM_10758_XYZ_%22%3A%22s%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_10759_XYZ_%22%3A%22TyFC%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_10760_XYZ_%22%3A%22h7yi%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_10761_XYZ_%22%3A%22QjyCJ%22%0A++++++++++++++++++++%7D&" + 
                    "fields=bU&fields=_EM_28244_XYZ_&fields=SaglVm1Ik&fields=_EM_28245_XYZ_&fields=UEM_22U50%5BXYZa")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("rs626"))
                .body("'originalVariantQuery'", containsString("rs626"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("HG1453_PATCH"))
                .body("'start'", numberMatches(1.40527617E8))
                .body("'end'", numberMatches(1.40527617E8))
                .body("'allele_string'", containsString("A/C/G/T"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("intergenic_variant"))
                .body("'intergenic_consequences'.size()", equalTo(3))
                .body("'intergenic_consequences'[0].'impact'", containsString("MODIFIER"))
                .body("'intergenic_consequences'[0].'variantAllele'", containsString("C"))
                .body("'intergenic_consequences'[0].'consequenceTerms'.size()", equalTo(1))
                .body("'intergenic_consequences'[0].'consequenceTerms'", hasItems("intergenic_variant"))
                .body("'intergenic_consequences'[1].'impact'", containsString("MODIFIER"))
                .body("'intergenic_consequences'[1].'variantAllele'", containsString("G"))
                .body("'intergenic_consequences'[1].'consequenceTerms'.size()", equalTo(1))
                .body("'intergenic_consequences'[1].'consequenceTerms'", hasItems("intergenic_variant"))
                .body("'intergenic_consequences'[2].'impact'", containsString("MODIFIER"))
                .body("'intergenic_consequences'[2].'variantAllele'", containsString("T"))
                .body("'intergenic_consequences'[2].'consequenceTerms'.size()", equalTo(1))
                .body("'intergenic_consequences'[2].'consequenceTerms'", hasItems("intergenic_variant"))
                .body("'successfully_annotated'", equalTo(true));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_57() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_587_XYZ_")
                .get(baseUrlOfSut + "/annotation/summary/_EM_584_XYZ_?" + 
                    "isoformOverrideSource=_EM_585_XYZ_&" + 
                    "projection=CANONICAL")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("_EM_584_XYZ_"))
                .body("'genomicLocation'.'chromosome'", nullValue())
                .body("'genomicLocation'.'start'", nullValue())
                .body("'genomicLocation'.'end'", nullValue())
                .body("'genomicLocation'.'referenceAllele'", nullValue())
                .body("'genomicLocation'.'variantAllele'", nullValue());
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_58() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/summary/undefined?isoformOverrideSource=_EM_1038_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("undefined"))
                .body("'genomicLocation'.'chromosome'", nullValue())
                .body("'genomicLocation'.'start'", nullValue())
                .body("'genomicLocation'.'end'", nullValue())
                .body("'genomicLocation'.'referenceAllele'", nullValue())
                .body("'genomicLocation'.'variantAllele'", nullValue())
                .body("'transcriptConsequences'.size()", equalTo(0))
                .body("'transcriptConsequenceSummaries'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_59() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"gHX2JUVU\", " + 
                    " \"jVUXQUDHqUIsgHrVfI\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/dbsnp/?" + 
                    "isoformOverrideSource=&" + 
                    "token=%7B%0A++++++++++++++++++++%22j%5B%7CtW87VvgxV1G%22%3A%22lUIXQTDHrWIreUrTd%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22Up7E%22%3A%22Lz%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22STOQTJkG%22%3A%22W2%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22uXJ0hju5euTzuPPGt%22%3A%22F59mfvveq%22%0A++++++++++++++++++++%7D&" + 
                    "fields=clinvar&fields=yjF&fields=fLU1UsIUuf")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("gHX2JUVU"))
                .body("[0].'originalVariantQuery'", containsString("gHX2JUVU"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[0].'clinvar'.'annotation'", nullValue())
                .body("[1].'variant'", containsString("jVUXQUDHqUIsgHrVfI"))
                .body("[1].'originalVariantQuery'", containsString("jVUXQUDHqUIsgHrVfI"))
                .body("[1].'successfully_annotated'", equalTo(false))
                .body("[1].'clinvar'.'annotation'", nullValue());
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_60() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/summary/jPBG%5Dl?" + 
                    "isoformOverrideSource=d_c4bjQw8thfm&" + 
                    "projection=ALL")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("jPBG]l"))
                .body("'genomicLocation'.'chromosome'", nullValue())
                .body("'genomicLocation'.'start'", nullValue())
                .body("'genomicLocation'.'end'", nullValue())
                .body("'genomicLocation'.'referenceAllele'", nullValue())
                .body("'genomicLocation'.'variantAllele'", nullValue())
                .body("'transcriptConsequences'.size()", equalTo(0))
                .body("'transcriptConsequenceSummaries'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_61() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"7vAFI4HLTp\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?" + 
                    "isoformOverrideSource=&" + 
                    "projection=ALL")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("7vAFI4HLTp"))
                .body("[0].'genomicLocation'.'chromosome'", nullValue())
                .body("[0].'genomicLocation'.'start'", nullValue())
                .body("[0].'genomicLocation'.'end'", nullValue())
                .body("[0].'genomicLocation'.'referenceAllele'", nullValue())
                .body("[0].'genomicLocation'.'variantAllele'", nullValue())
                .body("[0].'transcriptConsequences'.size()", equalTo(0))
                .body("[0].'transcriptConsequenceSummaries'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_62() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"VD\", " + 
                    " \"b\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/dbsnp/?" + 
                    "token=khkUTXUVUGsFm&" + 
                    "fields=oncokb&fields=oSEBuVNXW3tVPYUh&fields=TvzR1%5CsTU5Xo%60UdVaeV&fields=Se")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("VD"))
                .body("[0].'originalVariantQuery'", containsString("VD"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[1].'variant'", containsString("b"))
                .body("[1].'originalVariantQuery'", containsString("b"))
                .body("[1].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_63() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"y:U\", " + 
                    " \"start\": 738008576, " + 
                    " \"end\": -1048576, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"_EM_17896_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'genomicLocation'.'chromosome'", containsString("y:U"))
                .body("[0].'genomicLocation'.'start'", numberMatches(7.38008576E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(-1048576.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("-"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("_EM_17896_XYZ_"))
                .body("[0].'variant'", containsString("y:U:g.738008576_738008577ins_EM_17896_XYZ_"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_64() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"7:g.140453136A\u003eT\", " + 
                    " \"17:g.41276045_41276046del\", " + 
                    " \"rs12190874\", " + 
                    " \"INVALID\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/my_variant_info/variant?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(3))
                .body("[0].'variant'", containsString("chr7:g.140453136A>T"))
                .body("[0].'query'", containsString("chr7:g.140453136A>T"))
                .body("[0].'hgvs'", containsString("7:g.140453136A>T"))
                .body("[0].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[0].'vcf'.'alt'", containsString("T"))
                .body("[0].'vcf'.'position'", containsString("140453136"))
                .body("[0].'vcf'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[0].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[0].'dbsnp'.'alleles'[0].'allele'", containsString("A"))
                .body("[0].'dbsnp'.'alleles'[1].'allele'", containsString("T"))
                .body("[0].'dbsnp'.'alt'", containsString("T"))
                .body("[0].'dbsnp'.'chrom'", containsString("7"))
                .body("[0].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[0].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[0].'dbsnp'.'flags'[0]", nullValue())
                .body("[0].'dbsnp'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'dbsnp'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'dbsnp'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'rsid'", containsString("rs113488022"))
                .body("[0].'dbsnp'.'vartype'", containsString("snv"))
                .body("[0].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[0].'cosmic'.'alt'", containsString("A"))
                .body("[0].'cosmic'.'chrom'", containsString("7"))
                .body("[0].'cosmic'.'cosmicId'", containsString("COSM476"))
                .body("[0].'cosmic'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'cosmic'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'cosmic'.'mutFreq'", numberMatches(2.83))
                .body("[0].'cosmic'.'mutNt'", containsString("T>A"))
                .body("[0].'cosmic'.'ref'", containsString("T"))
                .body("[0].'cosmic'.'tumorSite'", containsString("biliary_tract"))
                .body("[0].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[0].'clinVar'.'alleleId'", numberMatches(29000.0))
                .body("[0].'clinVar'.'alt'", containsString("T"))
                .body("[0].'clinVar'.'chrom'", containsString("7"))
                .body("[0].'clinVar'.'cytogenic'", containsString("7q34"))
                .body("[0].'clinVar'.'gene'.'symbol'", containsString("BRAF"))
                .body("[0].'clinVar'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'clinVar'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'clinVar'.'hg38'.'start'", containsString("140753336"))
                .body("[0].'clinVar'.'hg38'.'end'", containsString("140753336"))
                .body("[0].'clinVar'.'hgvs'.'coding'.size()", equalTo(14))
                .body("[0].'clinVar'.'hgvs'.'coding'", hasItems("LRG_299t1:c.1799T>A", "NM_001354609.2:c.1799T>A", "NM_001374244.1:c.1919T>A", "NM_001374258.1:c.1919T>A", "NM_001378467.1:c.1808T>A", "NM_001378468.1:c.1799T>A", "NM_001378469.1:c.1733T>A", "NM_001378470.1:c.1697T>A", "NM_001378471.1:c.1688T>A", "NM_001378472.1:c.1643T>A", "NM_001378473.1:c.1643T>A", "NM_001378474.1:c.1799T>A", "NM_001378475.1:c.1535T>A", "NM_004333.6:c.1799T>A"))
                .body("[0].'clinVar'.'hgvs'.'genomic'.size()", equalTo(5))
                .body("[0].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_299:g.176429T>A", "NC_000007.12:g.140099605A>T", "NC_000007.13:g.140453136A>T", "NC_000007.14:g.140753336A>T", "NG_007873.3:g.176429T>A"))
                .body("[0].'clinVar'.'hgvs'.'protein'.size()", equalTo(17))
                .body("[0].'clinVar'.'hgvs'.'protein'", hasItems("LRG_299p1:p.Val600Glu", "NP_001341538.1:p.Val600Glu", "NP_001361173.1:p.Val640Glu", "NP_001361187.1:p.Val640Glu", "NP_001365396.1:p.Val603Glu", "NP_001365397.1:p.Val600Glu", "NP_001365398.1:p.Val578Glu", "NP_001365399.1:p.Val566Glu", "NP_001365400.1:p.Val563Glu", "NP_001365401.1:p.Val548Glu", "NP_001365402.1:p.Val548Glu", "NP_001365403.1:p.Val600Glu", "NP_001365404.1:p.Val512Glu", "NP_004324.2:p.Val600Glu", "NP_004324.2:p.Val600Glu", "P15056:p.Val600Glu", "p.V600E"))
                .body("[0].'clinVar'.'rcv'.size()", equalTo(31))
                .body("[0].'clinVar'.'rcv'[0].'accession'", containsString("RCV000014992"))
                .body("[0].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_004333.6(BRAF):c.1799T>A (p.Val600Glu)"))
                .body("[0].'clinVar'.'rcv'[1].'accession'", containsString("RCV000014993"))
                .body("[0].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[1].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_004333.6(BRAF):c.1799T>A (p.Val600Glu)"))
                .body("[0].'clinVar'.'rcv'[2].'accession'", containsString("RCV000014994"))
                .body("[0].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[2].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_004333.6(BRAF):c.1799T>A (p.Val600Glu)"))
                // Skipping assertions on the remaining 28 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'clinVar'.'variantId'", numberMatches(13961.0))
                .body("[0].'mutdb'.'rsid'", containsString("rs113488022"))
                .body("[0].'mutdb'.'ref'", containsString("T"))
                .body("[0].'mutdb'.'alt'", containsString("A"))
                .body("[0].'mutdb'.'uniprotId'", containsString("VAR_018629"))
                .body("[0].'mutdb'.'mutpredScore'", numberMatches(0.705))
                .body("[0].'mutdb'.'cosmicId'", containsString("476, 1131"))
                .body("[0].'mutdb'.'chrom'", containsString("7"))
                .body("[0].'mutdb'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'mutdb'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'gnomadExome'.'alleleCount'.'ac'", numberMatches(1.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_amr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_asj'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_nfe'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_sas'", numberMatches(1.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an'", numberMatches(251260.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_afr'", numberMatches(16252.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_amr'", numberMatches(34528.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_asj'", numberMatches(10076.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_eas'", numberMatches(18392.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_fin'", numberMatches(21638.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_nfe'", numberMatches(113638.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_oth'", numberMatches(6124.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_sas'", numberMatches(30612.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af'", numberMatches(3.97994E-6))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_amr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_asj'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_nfe'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_sas'", numberMatches(3.26669E-5))
                .body("[0].'gnomadExome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_sas'", numberMatches(0.0))
                .body("[1].'variant'", containsString("chr17:g.41276045_41276046del"))
                .body("[1].'query'", containsString("chr17:g.41276045_41276046del"))
                .body("[1].'hgvs'", containsString("17:g.41276045_41276046del"))
                .body("[1].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[1].'vcf'.'alt'", containsString("A"))
                .body("[1].'vcf'.'position'", containsString("41276044"))
                .body("[1].'vcf'.'ref'", containsString("ACT"))
                .body("[1].'gnomadExome'.'alleleCount'.'ac'", numberMatches(57.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_amr'", numberMatches(1.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_asj'", numberMatches(41.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_nfe'", numberMatches(11.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_sas'", numberMatches(4.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an'", numberMatches(251050.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_afr'", numberMatches(16250.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_amr'", numberMatches(34592.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_asj'", numberMatches(10078.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_eas'", numberMatches(18394.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_fin'", numberMatches(21644.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_nfe'", numberMatches(113354.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_oth'", numberMatches(6130.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_sas'", numberMatches(30608.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af'", numberMatches(2.27046E-4))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_amr'", numberMatches(2.89084E-5))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_asj'", numberMatches(0.00406827))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_nfe'", numberMatches(9.70411E-5))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_sas'", numberMatches(1.30685E-4))
                .body("[1].'gnomadExome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_sas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac'", numberMatches(1.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_amr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_asj'", numberMatches(1.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_nfe'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_sas'", nullValue())
                .body("[1].'gnomadGenome'.'alleleNumber'.'an'", numberMatches(31392.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_afr'", numberMatches(8712.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_amr'", numberMatches(848.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_asj'", numberMatches(290.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_eas'", numberMatches(1560.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_fin'", numberMatches(3468.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_nfe'", numberMatches(15426.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_oth'", numberMatches(1088.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_sas'", nullValue())
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af'", numberMatches(3.18552E-5))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_amr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_asj'", numberMatches(0.00344828))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_nfe'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_sas'", nullValue())
                .body("[1].'gnomadGenome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_sas'", nullValue())
                .body("[2].'variant'", containsString("chr6:g.42454850G>A"))
                .body("[2].'query'", containsString("rs12190874"))
                .body("[2].'hgvs'", containsString("rs12190874"))
                .body("[2].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[2].'vcf'.'alt'", containsString("A"))
                .body("[2].'vcf'.'position'", containsString("42454850"))
                .body("[2].'vcf'.'ref'", containsString("G"))
                .body("[2].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[2].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[2].'dbsnp'.'alleles'[0].'allele'", containsString("G"))
                .body("[2].'dbsnp'.'alleles'[1].'allele'", containsString("A"))
                .body("[2].'dbsnp'.'alt'", containsString("A"))
                .body("[2].'dbsnp'.'chrom'", containsString("6"))
                .body("[2].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[2].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[2].'dbsnp'.'flags'[0]", nullValue())
                .body("[2].'dbsnp'.'hg19'.'start'", numberMatches(4.245485E7))
                .body("[2].'dbsnp'.'hg19'.'end'", numberMatches(4.245485E7))
                .body("[2].'dbsnp'.'ref'", containsString("G"))
                .body("[2].'dbsnp'.'rsid'", containsString("rs12190874"))
                .body("[2].'dbsnp'.'vartype'", containsString("snv"))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac'", numberMatches(3282.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_afr'", numberMatches(646.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_amr'", numberMatches(94.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_asj'", numberMatches(12.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_eas'", numberMatches(271.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_fin'", numberMatches(433.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_nfe'", numberMatches(1698.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_oth'", numberMatches(128.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_sas'", nullValue())
                .body("[2].'gnomadGenome'.'alleleNumber'.'an'", numberMatches(31340.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_afr'", numberMatches(8696.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_amr'", numberMatches(846.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_asj'", numberMatches(290.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_eas'", numberMatches(1556.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_fin'", numberMatches(3474.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_nfe'", numberMatches(15394.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_oth'", numberMatches(1084.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_sas'", nullValue())
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af'", numberMatches(0.104722))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_afr'", numberMatches(0.074287))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_amr'", numberMatches(0.111111))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_asj'", numberMatches(0.0413793))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_eas'", numberMatches(0.174165))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_fin'", numberMatches(0.12464))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_nfe'", numberMatches(0.110303))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_oth'", numberMatches(0.118081))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_sas'", nullValue())
                .body("[2].'gnomadGenome'.'homozygotes'.'hom'", numberMatches(179.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_afr'", numberMatches(15.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_amr'", numberMatches(7.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_asj'", numberMatches(1.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_eas'", numberMatches(19.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_fin'", numberMatches(26.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_nfe'", numberMatches(103.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_oth'", numberMatches(8.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_sas'", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_65() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"chr17:g.7577566_7577567insA\", " + 
                    " \"chr17:g.7577567A\u003eC\", " + 
                    " \"chr17:g.7577567_7577567del\", " + 
                    " \"chr17:g.7577567_7577568insCACATGTAGTTGT\", " + 
                    " \"chr17:g.7577568C\u003eA\", " + 
                    " \"chr17:g.7577568C\u003eG\", " + 
                    " \"chr17:g.7577568C\u003eT\", " + 
                    " \"chr17:g.7577569A\u003eC\", " + 
                    " \"chr17:g.7577569A\u003eG\", " + 
                    " \"chr17:g.7577569_7577569del\", " + 
                    " \"chr17:g.7577569_7577572del\", " + 
                    " \"chr17:g.7577569_7577580del\", " + 
                    " \"chr17:g.7577570C\u003eA\", " + 
                    " \"chr17:g.7577570C\u003eG\", " + 
                    " \"chr17:g.7577570C\u003eT\", " + 
                    " \"chr17:g.7577571A\u003eT\", " + 
                    " \"chr17:g.7577571_7577572insTGTAGT\", " + 
                    " \"chr17:g.7577572_7577572del\", " + 
                    " \"chr17:g.7577572_7577574del\", " + 
                    " \"chr17:g.7577573G\u003eC\", " + 
                    " \"chr17:g.7577573G\u003eT\", " + 
                    " \"chr17:g.7577573_7577573del\", " + 
                    " \"chr17:g.7577574T\u003eC\", " + 
                    " \"chr17:g.7577574T\u003eG\", " + 
                    " \"chr17:g.7577575A\u003eC\", " + 
                    " \"chr17:g.7577575A\u003eG\", " + 
                    " \"chr17:g.7577575A\u003eT\", " + 
                    " \"chr17:g.7577576_7577578del\", " + 
                    " \"chr17:g.7577576_7577582del\", " + 
                    " \"chr17:g.7577577T\u003eC\", " + 
                    " \"chr17:g.7577578T\u003eC\", " + 
                    " \"chr17:g.7577579G\u003eC\", " + 
                    " \"chr17:g.7577579G\u003eT\", " + 
                    " \"chr17:g.7577579_7577581del\", " + 
                    " \"chr17:g.7577580T\u003eC\", " + 
                    " \"chr17:g.7577580T\u003eG\", " + 
                    " \"chr17:g.7577581A\u003eC\", " + 
                    " \"chr17:g.7577581A\u003eG\", " + 
                    " \"chr17:g.7577581A\u003eT\", " + 
                    " \"chr17:g.7577581_7577595del\", " + 
                    " \"chr17:g.7577584_7577585insCTAC\", " + 
                    " \"chr17:g.7577585_7577590del\", " + 
                    " \"chr17:g.7577586A\u003eC\", " + 
                    " \"chr17:g.7577586A\u003eT\", " + 
                    " \"chr17:g.7577586_7577587insTGGTGGTACAGTCAG\", " + 
                    " \"chr17:g.7577587T\u003eA\", " + 
                    " \"chr17:g.7577587_7577587del\", " + 
                    " \"chr17:g.7577588_7577595del\", " + 
                    " \"chr17:g.7577589G\u003eT\", " + 
                    " \"chr17:g.7577589_7577590insT\", " + 
                    " \"chr17:g.7577591_7577596del\", " + 
                    " \"chr17:g.7577592_7577602del\", " + 
                    " \"chr17:g.7577593_7577593del\", " + 
                    " \"chr17:g.7577593_7577594del\", " + 
                    " \"chr17:g.7577594A\u003eT\", " + 
                    " \"chr17:g.7577594_7577595del\", " + 
                    " \"chr17:g.7577595C\u003eT\", " + 
                    " \"chr17:g.7577596_7577596del\", " + 
                    " \"chr17:g.7577599C\u003eG\", " + 
                    " \"chr17:g.7577599_7577600insA\", " + 
                    " \"chr17:g.7577604C\u003eT\", " + 
                    " \"chr17:g.7577608C\u003eT\", " + 
                    " \"chr17:g.7577609C\u003eA\", " + 
                    " \"chr17:g.7577609C\u003eG\", " + 
                    " \"chr17:g.7577609C\u003eT\", " + 
                    " \"chr17:g.7577610T\u003eA\", " + 
                    " \"chr17:g.7577610T\u003eC\", " + 
                    " \"chr17:g.7577610T\u003eG\", " + 
                    " \"chr17:g.7578098_7578176del\", " + 
                    " \"chr17:g.7578175A\u003eG\", " + 
                    " \"chr17:g.7578175_7578233del\", " + 
                    " \"chr17:g.7578176C\u003eA\", " + 
                    " \"chr17:g.7578176C\u003eT\", " + 
                    " \"chr17:g.7578177C\u003eA\", " + 
                    " \"chr17:g.7578177C\u003eG\", " + 
                    " \"chr17:g.7578179C\u003eA\", " + 
                    " \"chr17:g.7578181G\u003eA\", " + 
                    " \"chr17:g.7578181_7578181del\", " + 
                    " \"chr17:g.7578184_7578184del\", " + 
                    " \"chr17:g.7578184_7578187del\", " + 
                    " \"chr17:g.7578184_7578208del\", " + 
                    " \"chr17:g.7578188C\u003eA\", " + 
                    " \"chr17:g.7578189A\u003eC\", " + 
                    " \"chr17:g.7578189A\u003eT\", " + 
                    " \"chr17:g.7578190T\u003eC\", " + 
                    " \"chr17:g.7578190T\u003eG\", " + 
                    " \"chr17:g.7578191A\u003eC\", " + 
                    " \"chr17:g.7578191A\u003eG\", " + 
                    " \"chr17:g.7578191A\u003eT\", " + 
                    " \"chr17:g.7578192_7578192del\", " + 
                    " \"chr17:g.7578192_7578193delinsA\", " + 
                    " \"chr17:g.7578192_7578193insGGCACCACCACACTATGTCGAAAAGTGT\", " + 
                    " \"chr17:g.7578193G\u003eC\", " + 
                    " \"chr17:g.7578194_7578195insCAC\", " + 
                    " \"chr17:g.7578195_7578197del\", " + 
                    " \"chr17:g.7578196A\u003eC\", " + 
                    " \"chr17:g.7578196_7578197insC\", " + 
                    " \"chr17:g.7578196_7578197insCC\", " + 
                    " \"chr17:g.7578196_7578197insCCACCACACTAT\", " + 
                    " \"chr17:g.7578197C\u003eA\", " + 
                    " \"chr17:g.7578197_7578198delinsTT\", " + 
                    " \"chr17:g.7578199A\u003eC\", " + 
                    " \"chr17:g.7578199_7578199del\", " + 
                    " \"chr17:g.7578200_7578222del\", " + 
                    " \"chr17:g.7578201_7578202del\", " + 
                    " \"chr17:g.7578202A\u003eC\", " + 
                    " \"chr17:g.7578202A\u003eT\", " + 
                    " \"chr17:g.7578203C\u003eA\", " + 
                    " \"chr17:g.7578203C\u003eG\", " + 
                    " \"chr17:g.7578203C\u003eT\", " + 
                    " \"chr17:g.7578204A\u003eC\", " + 
                    " \"chr17:g.7578204A\u003eT\", " + 
                    " \"chr17:g.7578205C\u003eA\", " + 
                    " \"chr17:g.7578205C\u003eG\", " + 
                    " \"chr17:g.7578205C\u003eT\", " + 
                    " \"chr17:g.7578205_7578206insTATGTCGAAAA\", " + 
                    " \"chr17:g.7578205_7578207delinsTTT\", " + 
                    " \"chr17:g.7578206T\u003eA\", " + 
                    " \"chr17:g.7578206T\u003eC\", " + 
                    " \"chr17:g.7578206T\u003eG\", " + 
                    " \"chr17:g.7578206_7578207del\", " + 
                    " \"chr17:g.7578207A\u003eC\", " + 
                    " \"chr17:g.7578207_7578215del\", " + 
                    " \"chr17:g.7578208T\u003eA\", " + 
                    " \"chr17:g.7578208T\u003eC\", " + 
                    " \"chr17:g.7578208T\u003eG\", " + 
                    " \"chr17:g.7578208_7578208del\", " + 
                    " \"chr17:g.7578208_7578217del\", " + 
                    " \"chr17:g.7578208_7578219del\", " + 
                    " \"chr17:g.7578209_7578209del\", " + 
                    " \"chr17:g.7578210_7578210del\", " + 
                    " \"chr17:g.7578211C\u003eA\", " + 
                    " \"chr17:g.7578211C\u003eT\", " + 
                    " \"chr17:g.7578211_7578212delinsA\", " + 
                    " \"chr17:g.7578212G\u003eA\", " + 
                    " \"chr17:g.7578212G\u003eC\", " + 
                    " \"chr17:g.7578212_7578213insA\", " + 
                    " \"chr17:g.7578213_7578213del\", " + 
                    " \"chr17:g.7578213_7578214del\", " + 
                    " \"chr17:g.7578213_7578231del\", " + 
                    " \"chr17:g.7578215_7578227del\", " + 
                    " \"chr17:g.7578217G\u003eA\", " + 
                    " \"chr17:g.7578219_7578220insT\", " + 
                    " \"chr17:g.7578220_7578221del\", " + 
                    " \"chr17:g.7578220_7578221insTTCTGTCATCCAAATACTCCACACG\", " + 
                    " \"chr17:g.7578222_7578223del\", " + 
                    " \"chr17:g.7578223C\u003eT\", " + 
                    " \"chr17:g.7578224T\u003eA\", " + 
                    " \"chr17:g.7578225G\u003eT\", " + 
                    " \"chr17:g.7578226T\u003eA\", " + 
                    " \"chr17:g.7578226_7578229del\", " + 
                    " \"chr17:g.7578227C\u003eT\", " + 
                    " \"chr17:g.7578229_7578237del\", " + 
                    " \"chr17:g.7578230_7578230del\", " + 
                    " \"chr17:g.7578231_7578232insAAAT\", " + 
                    " \"chr17:g.7578235T\u003eA\", " + 
                    " \"chr17:g.7578235T\u003eC\", " + 
                    " \"chr17:g.7578235T\u003eG\", " + 
                    " \"chr17:g.7578236A\u003eG\", " + 
                    " \"chr17:g.7578238_7578255del\", " + 
                    " \"chr17:g.7578239C\u003eA\", " + 
                    " \"chr17:g.7578239_7578240delinsAG\", " + 
                    " \"chr17:g.7578239_7578240insCA\", " + 
                    " \"chr17:g.7578242C\u003eA\", " + 
                    " \"chr17:g.7578244_7578261del\", " + 
                    " \"chr17:g.7578245G\u003eA\", " + 
                    " \"chr17:g.7578245_7578277del\", " + 
                    " \"chr17:g.7578247A\u003eT\", " + 
                    " \"chr17:g.7578247_7578247del\", " + 
                    " \"chr17:g.7578247_7578248del\", " + 
                    " \"chr17:g.7578248_7578249insAT\", " + 
                    " \"chr17:g.7578250_7578250del\", " + 
                    " \"chr17:g.7578252_7578299del\", " + 
                    " \"chr17:g.7578253C\u003eA\", " + 
                    " \"chr17:g.7578253C\u003eT\", " + 
                    " \"chr17:g.7578254C\u003eA\", " + 
                    " \"chr17:g.7578257C\u003eA\", " + 
                    " \"chr17:g.7578257_7578257del\", " + 
                    " \"chr17:g.7578257_7578258delinsAA\", " + 
                    " \"chr17:g.7578259A\u003eT\", " + 
                    " \"chr17:g.7578260C\u003eT\", " + 
                    " \"chr17:g.7578262C\u003eA\", " + 
                    " \"chr17:g.7578262C\u003eG\", " + 
                    " \"chr17:g.7578262C\u003eT\", " + 
                    " \"chr17:g.7578263G\u003eA\", " + 
                    " \"chr17:g.7578263_7578282del\", " + 
                    " \"chr17:g.7578264_7578265delinsTT\", " + 
                    " \"chr17:g.7578264_7578338delinsTT\", " + 
                    " \"chr17:g.7578265A\u003eC\", " + 
                    " \"chr17:g.7578265A\u003eG\", " + 
                    " \"chr17:g.7578265A\u003eT\", " + 
                    " \"chr17:g.7578265_7578266insT\", " + 
                    " \"chr17:g.7578266T\u003eA\", " + 
                    " \"chr17:g.7578266_7578266del\", " + 
                    " \"chr17:g.7578266_7578267insA\", " + 
                    " \"chr17:g.7578267_7578268insAG\", " + 
                    " \"chr17:g.7578268A\u003eC\", " + 
                    " \"chr17:g.7578268A\u003eG\", " + 
                    " \"chr17:g.7578268_7578280del\", " + 
                    " \"chr17:g.7578269G\u003eA\", " + 
                    " \"chr17:g.7578270A\u003eT\", " + 
                    " \"chr17:g.7578271T\u003eA\", " + 
                    " \"chr17:g.7578271T\u003eC\", " + 
                    " \"chr17:g.7578271T\u003eG\", " + 
                    " \"chr17:g.7578272G\u003eA\", " + 
                    " \"chr17:g.7578272G\u003eC\", " + 
                    " \"chr17:g.7578272G\u003eT\", " + 
                    " \"chr17:g.7578272_7578273delinsAA\", " + 
                    " \"chr17:g.7578274T\u003eA\", " + 
                    " \"chr17:g.7578275G\u003eA\", " + 
                    " \"chr17:g.7578275_7578277del\", " + 
                    " \"chr17:g.7578275_7578293del\", " + 
                    " \"chr17:g.7578277_7578281del\", " + 
                    " \"chr17:g.7578280G\u003eA\", " + 
                    " \"chr17:g.7578280_7578280del\", " + 
                    " \"chr17:g.7578281G\u003eA\", " + 
                    " \"chr17:g.7578281G\u003eT\", " + 
                    " \"chr17:g.7578282_7578306del\", " + 
                    " \"chr17:g.7578283_7578284insC\", " + 
                    " \"chr17:g.7578284_7578284del\", " + 
                    " \"chr17:g.7578284_7578288del\", " + 
                    " \"chr17:g.7578284_7578297del\", " + 
                    " \"chr17:g.7578284_7578310del\", " + 
                    " \"chr17:g.7578285_7578296del\", " + 
                    " \"chr17:g.7578286A\u003eT\", " + 
                    " \"chr17:g.7578286_7578286del\", " + 
                    " \"chr17:g.7578287_7578287del\", " + 
                    " \"chr17:g.7578289C\u003eT\", " + 
                    " \"chr17:g.7578289_7578290delinsTT\", " + 
                    " \"chr17:g.7578290C\u003eA\", " + 
                    " \"chr17:g.7578290C\u003eG\", " + 
                    " \"chr17:g.7578290C\u003eT\", " + 
                    " \"chr17:g.7578291T\u003eA\", " + 
                    " \"chr17:g.7578291T\u003eC\", " + 
                    " \"chr17:g.7578291T\u003eG\", " + 
                    " \"chr17:g.7578293_7578977del\", " + 
                    " \"chr17:g.7578313_7578425del\", " + 
                    " \"chr17:g.7578359_7578373del\", " + 
                    " \"chr17:g.7578359_7578375del\", " + 
                    " \"chr17:g.7578360_7578373del\", " + 
                    " \"chr17:g.7578362_7578385del\", " + 
                    " \"chr17:g.7578365_7578383del\", " + 
                    " \"chr17:g.7578370C\u003eA\", " + 
                    " \"chr17:g.7578370C\u003eG\", " + 
                    " \"chr17:g.7578370C\u003eT\", " + 
                    " \"chr17:g.7578371C\u003eG\", " + 
                    " \"chr17:g.7578371_7578382del\", " + 
                    " \"chr17:g.7578372_7578373delinsC\", " + 
                    " \"chr17:g.7578376_7578376del\", " + 
                    " \"chr17:g.7578376_7578379del\", " + 
                    " \"chr17:g.7578377_7578377del\", " + 
                    " \"chr17:g.7578377_7578378del\", " + 
                    " \"chr17:g.7578378A\u003eT\", " + 
                    " \"chr17:g.7578378_7578379insTC\", " + 
                    " \"chr17:g.7578378_7578395del\", " + 
                    " \"chr17:g.7578379_7578379del\", " + 
                    " \"chr17:g.7578382G\u003eC\", " + 
                    " \"chr17:g.7578382_7578382del\", " + 
                    " \"chr17:g.7578384_7578401del\", " + 
                    " \"chr17:g.7578388C\u003eG\", " + 
                    " \"chr17:g.7578388C\u003eT\", " + 
                    " \"chr17:g.7578389G\u003eA\", " + 
                    " \"chr17:g.7578389_7578402delinsT\", " + 
                    " \"chr17:g.7578390C\u003eG\", " + 
                    " \"chr17:g.7578392C\u003eA\", " + 
                    " \"chr17:g.7578392C\u003eT\", " + 
                    " \"chr17:g.7578393A\u003eC\", " + 
                    " \"chr17:g.7578393A\u003eT\", " + 
                    " \"chr17:g.7578393_7578405del\", " + 
                    " \"chr17:g.7578394T\u003eA\", " + 
                    " \"chr17:g.7578394T\u003eC\", " + 
                    " \"chr17:g.7578394T\u003eG\", " + 
                    " \"chr17:g.7578394_7578395del\", " + 
                    " \"chr17:g.7578394_7578396del\", " + 
                    " \"chr17:g.7578395G\u003eA\", " + 
                    " \"chr17:g.7578395G\u003eC\", " + 
                    " \"chr17:g.7578395G\u003eT\", " + 
                    " \"chr17:g.7578397T\u003eG\", " + 
                    " \"chr17:g.7578397_7578397del\", " + 
                    " \"chr17:g.7578397_7578398insG\", " + 
                    " \"chr17:g.7578398G\u003eC\", " + 
                    " \"chr17:g.7578398_7578398del\", " + 
                    " \"chr17:g.7578400G\u003eC\", " + 
                    " \"chr17:g.7578401_7578402delinsT\", " + 
                    " \"chr17:g.7578402G\u003eC\", " + 
                    " \"chr17:g.7578402G\u003eT\", " + 
                    " \"chr17:g.7578403C\u003eA\", " + 
                    " \"chr17:g.7578403C\u003eT\", " + 
                    " \"chr17:g.7578404A\u003eC\", " + 
                    " \"chr17:g.7578404A\u003eG\", " + 
                    " \"chr17:g.7578404A\u003eT\", " + 
                    " \"chr17:g.7578406C\u003eA\", " + 
                    " \"chr17:g.7578406C\u003eT\", " + 
                    " \"chr17:g.7578406_7578406del\", " + 
                    " \"chr17:g.7578407G\u003eA\", " + 
                    " \"chr17:g.7578407G\u003eC\", " + 
                    " \"chr17:g.7578407_7578412delinsCCC\", " + 
                    " \"chr17:g.7578410T\u003eA\", " + 
                    " \"chr17:g.7578410T\u003eC\", " + 
                    " \"chr17:g.7578411_7578411del\", " + 
                    " \"chr17:g.7578412A\u003eC\", " + 
                    " \"chr17:g.7578412A\u003eG\", " + 
                    " \"chr17:g.7578413C\u003eA\", " + 
                    " \"chr17:g.7578413C\u003eG\", " + 
                    " \"chr17:g.7578413C\u003eT\", " + 
                    " \"chr17:g.7578413_7578413del\", " + 
                    " \"chr17:g.7578414_7578414del\", " + 
                    " \"chr17:g.7578415A\u003eT\", " + 
                    " \"chr17:g.7578416C\u003eA\", " + 
                    " \"chr17:g.7578417_7578419delinsTG\", " + 
                    " \"chr17:g.7578418T\u003eC\", " + 
                    " \"chr17:g.7578419C\u003eA\", " + 
                    " \"chr17:g.7578419_7578419del\", " + 
                    " \"chr17:g.7578419_7578420insCGTCATGTG\", " + 
                    " \"chr17:g.7578421G\u003eA\", " + 
                    " \"chr17:g.7578422_7578423insCATGTGCTG\", " + 
                    " \"chr17:g.7578422_7578424del\", " + 
                    " \"chr17:g.7578423_7578424insA\", " + 
                    " \"chr17:g.7578423_7579121del\", " + 
                    " \"chr17:g.7578424_7578425del\", " + 
                    " \"chr17:g.7578424_7578427del\", " + 
                    " \"chr17:g.7578427T\u003eC\", " + 
                    " \"chr17:g.7578427T\u003eG\", " + 
                    " \"chr17:g.7578428_7578431delinsACTA\", " + 
                    " \"chr17:g.7578428_7578627del\", " + 
                    " \"chr17:g.7578429_7578430insTG\", " + 
                    " \"chr17:g.7578430_7578431del\", " + 
                    " \"chr17:g.7578431G\u003eA\", " + 
                    " \"chr17:g.7578433G\u003eC\", " + 
                    " \"chr17:g.7578433G\u003eT\", " + 
                    " \"chr17:g.7578435C\u003eA\", " + 
                    " \"chr17:g.7578437G\u003eA\", " + 
                    " \"chr17:g.7578439_7578439del\", " + 
                    " \"chr17:g.7578440T\u003eA\", " + 
                    " \"chr17:g.7578440T\u003eC\", " + 
                    " \"chr17:g.7578441G\u003eT\", " + 
                    " \"chr17:g.7578441_7578442delinsA\", " + 
                    " \"chr17:g.7578442T\u003eC\", " + 
                    " \"chr17:g.7578442_7578450delinsCCG\", " + 
                    " \"chr17:g.7578443A\u003eC\", " + 
                    " \"chr17:g.7578443A\u003eG\", " + 
                    " \"chr17:g.7578443A\u003eT\", " + 
                    " \"chr17:g.7578443_7578691del\", " + 
                    " \"chr17:g.7578445A\u003eC\", " + 
                    " \"chr17:g.7578445A\u003eT\", " + 
                    " \"chr17:g.7578445_7578449delinsCTGGG\", " + 
                    " \"chr17:g.7578446T\u003eA\", " + 
                    " \"chr17:g.7578447_7578447del\", " + 
                    " \"chr17:g.7578448G\u003eT\", " + 
                    " \"chr17:g.7578449C\u003eA\", " + 
                    " \"chr17:g.7578449C\u003eT\", " + 
                    " \"chr17:g.7578452_7578453insGGCGCG\", " + 
                    " \"chr17:g.7578453_7578455del\", " + 
                    " \"chr17:g.7578453_7578458del\", " + 
                    " \"chr17:g.7578454G\u003eA\", " + 
                    " \"chr17:g.7578454_7578455del\", " + 
                    " \"chr17:g.7578454_7578457del\", " + 
                    " \"chr17:g.7578454_7578464delinsACGC\", " + 
                    " \"chr17:g.7578454_7578467del\", " + 
                    " \"chr17:g.7578455C\u003eG\", " + 
                    " \"chr17:g.7578455_7578460del\", " + 
                    " \"chr17:g.7578457C\u003eA\", " + 
                    " \"chr17:g.7578457C\u003eG\", " + 
                    " \"chr17:g.7578457C\u003eT\", " + 
                    " \"chr17:g.7578457_7578458delinsAA\", " + 
                    " \"chr17:g.7578458G\u003eA\", " + 
                    " \"chr17:g.7578458G\u003eC\", " + 
                    " \"chr17:g.7578458_7578459insA\", " + 
                    " \"chr17:g.7578458_7578464del\", " + 
                    " \"chr17:g.7578460A\u003eC\", " + 
                    " \"chr17:g.7578460A\u003eG\", " + 
                    " \"chr17:g.7578461C\u003eA\", " + 
                    " \"chr17:g.7578461_7578473del\", " + 
                    " \"chr17:g.7578462_7578463insC\", " + 
                    " \"chr17:g.7578463C\u003eA\", " + 
                    " \"chr17:g.7578463C\u003eG\", " + 
                    " \"chr17:g.7578463C\u003eT\", " + 
                    " \"chr17:g.7578463_7578464insA\", " + 
                    " \"chr17:g.7578464G\u003eC\", " + 
                    " \"chr17:g.7578464_7578464del\", " + 
                    " \"chr17:g.7578464_7578476del\", " + 
                    " \"chr17:g.7578465_7578470del\", " + 
                    " \"chr17:g.7578466G\u003eA\", " + 
                    " \"chr17:g.7578466G\u003eT\", " + 
                    " \"chr17:g.7578467T\u003eG\", " + 
                    " \"chr17:g.7578467_7578468insG\", " + 
                    " \"chr17:g.7578467_7578468insGCCGGGC\", " + 
                    " \"chr17:g.7578467_7578498del\", " + 
                    " \"chr17:g.7578469C\u003eA\", " + 
                    " \"chr17:g.7578469C\u003eT\", " + 
                    " \"chr17:g.7578469_7578469del\", " + 
                    " \"chr17:g.7578469_7578470insCGGG\", " + 
                    " \"chr17:g.7578469_7578470insGGG\", " + 
                    " \"chr17:g.7578469_7578474del\", " + 
                    " \"chr17:g.7578470_7578471delinsA\", " + 
                    " \"chr17:g.7578470_7578471insG\", " + 
                    " \"chr17:g.7578470_7578473del\", " + 
                    " \"chr17:g.7578470_7578482del\", " + 
                    " \"chr17:g.7578470_7578502del\", " + 
                    " \"chr17:g.7578471_7578471del\", " + 
                    " \"chr17:g.7578471_7578478del\", " + 
                    " \"chr17:g.7578471_7578483del\", " + 
                    " \"chr17:g.7578472G\u003eA\", " + 
                    " \"chr17:g.7578472_7578491del\", " + 
                    " \"chr17:g.7578473_7578474insA\", " + 
                    " \"chr17:g.7578473_7578474insC\", " + 
                    " \"chr17:g.7578474_7578475insG\", " + 
                    " \"chr17:g.7578475G\u003eA\", " + 
                    " \"chr17:g.7578475G\u003eC\", " + 
                    " \"chr17:g.7578475_7578475del\", " + 
                    " \"chr17:g.7578475_7578476delinsTT\", " + 
                    " \"chr17:g.7578476G\u003eA\", " + 
                    " \"chr17:g.7578476G\u003eT\", " + 
                    " \"chr17:g.7578476_7578477insGGGT\", " + 
                    " \"chr17:g.7578478G\u003eC\", " + 
                    " \"chr17:g.7578478G\u003eT\", " + 
                    " \"chr17:g.7578478_7578479delinsAA\", " + 
                    " \"chr17:g.7578478_7578480del\", " + 
                    " \"chr17:g.7578479G\u003eA\", " + 
                    " \"chr17:g.7578479G\u003eC\", " + 
                    " \"chr17:g.7578479G\u003eT\", " + 
                    " \"chr17:g.7578480_7578488del\", " + 
                    " \"chr17:g.7578480_7578495del\", " + 
                    " \"chr17:g.7578481_7578481del\", " + 
                    " \"chr17:g.7578481_7578482insT\", " + 
                    " \"chr17:g.7578482_7578482del\", " + 
                    " \"chr17:g.7578484_7578485insA\", " + 
                    " \"chr17:g.7578484_7578502del\", " + 
                    " \"chr17:g.7578485_7578485del\", " + 
                    " \"chr17:g.7578486A\u003eT\", " + 
                    " \"chr17:g.7578488_7578488del\", " + 
                    " \"chr17:g.7578490A\u003eC\", " + 
                    " \"chr17:g.7578491_7578491del\", " + 
                    " \"chr17:g.7578492C\u003eG\", " + 
                    " \"chr17:g.7578492C\u003eT\", " + 
                    " \"chr17:g.7578493C\u003eT\", " + 
                    " \"chr17:g.7578493_7578500del\", " + 
                    " \"chr17:g.7578494A\u003eC\", " + 
                    " \"chr17:g.7578495_7578507del\", " + 
                    " \"chr17:g.7578496A\u003eC\", " + 
                    " \"chr17:g.7578496A\u003eG\", " + 
                    " \"chr17:g.7578496A\u003eT\", " + 
                    " \"chr17:g.7578499T\u003eA\", " + 
                    " \"chr17:g.7578499T\u003eC\", " + 
                    " \"chr17:g.7578499T\u003eG\", " + 
                    " \"chr17:g.7578499_7578499del\", " + 
                    " \"chr17:g.7578500G\u003eA\", " + 
                    " \"chr17:g.7578500_7578500del\", " + 
                    " \"chr17:g.7578500_7578524del\", " + 
                    " \"chr17:g.7578501_7578502insA\", " + 
                    " \"chr17:g.7578502A\u003eC\", " + 
                    " \"chr17:g.7578502A\u003eG\", " + 
                    " \"chr17:g.7578503C\u003eT\", " + 
                    " \"chr17:g.7578503_7578507del\", " + 
                    " \"chr17:g.7578503_7578518del\", " + 
                    " \"chr17:g.7578504_7578505insGGG\", " + 
                    " \"chr17:g.7578505G\u003eA\", " + 
                    " \"chr17:g.7578505_7578505del\", " + 
                    " \"chr17:g.7578505_7578510del\", " + 
                    " \"chr17:g.7578507G\u003eC\", " + 
                    " \"chr17:g.7578507G\u003eT\", " + 
                    " \"chr17:g.7578508C\u003eT\", " + 
                    " \"chr17:g.7578508_7578535delinsG\", " + 
                    " \"chr17:g.7578509A\u003eC\", " + 
                    " \"chr17:g.7578509A\u003eG\", " + 
                    " \"chr17:g.7578510_7578510del\", " + 
                    " \"chr17:g.7578511G\u003eA\", " + 
                    " \"chr17:g.7578513C\u003eG\", " + 
                    " \"chr17:g.7578513_7578549del\", " + 
                    " \"chr17:g.7578514T\u003eG\", " + 
                    " \"chr17:g.7578515T\u003eA\", " + 
                    " \"chr17:g.7578515_7578516insGGCCAGTTGGCAAAACATCTTGTTGAG\", " + 
                    " \"chr17:g.7578516_7578516del\", " + 
                    " \"chr17:g.7578517G\u003eA\", " + 
                    " \"chr17:g.7578518C\u003eG\", " + 
                    " \"chr17:g.7578520A\u003eT\", " + 
                    " \"chr17:g.7578520_7578525del\", " + 
                    " \"chr17:g.7578521_7578521del\", " + 
                    " \"chr17:g.7578521_7578531del\", " + 
                    " \"chr17:g.7578522_7578522del\", " + 
                    " \"chr17:g.7578523T\u003eG\", " + 
                    " \"chr17:g.7578523_7578548del\", " + 
                    " \"chr17:g.7578524G\u003eA\", " + 
                    " \"chr17:g.7578524G\u003eC\", " + 
                    " \"chr17:g.7578525G\u003eC\", " + 
                    " \"chr17:g.7578525G\u003eT\", " + 
                    " \"chr17:g.7578526C\u003eA\", " + 
                    " \"chr17:g.7578526C\u003eG\", " + 
                    " \"chr17:g.7578526C\u003eT\", " + 
                    " \"chr17:g.7578526_7578527insA\", " + 
                    " \"chr17:g.7578527A\u003eC\", " + 
                    " \"chr17:g.7578527A\u003eG\", " + 
                    " \"chr17:g.7578527A\u003eT\", " + 
                    " \"chr17:g.7578527_7578527del\", " + 
                    " \"chr17:g.7578527_7578528del\", " + 
                    " \"chr17:g.7578528A\u003eC\", " + 
                    " \"chr17:g.7578528A\u003eT\", " + 
                    " \"chr17:g.7578529A\u003eC\", " + 
                    " \"chr17:g.7578530A\u003eC\", " + 
                    " \"chr17:g.7578530A\u003eG\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/my_variant_info/variant?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(280))
                .body("[0].'variant'", containsString("chr17:g.7577567A>C"))
                .body("[0].'query'", containsString("chr17:g.7577567A>C"))
                .body("[0].'hgvs'", containsString("chr17:g.7577567A>C"))
                .body("[0].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[0].'vcf'.'alt'", containsString("C"))
                .body("[0].'vcf'.'position'", containsString("7577567"))
                .body("[0].'vcf'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[0].'dbsnp'.'alleles'.size()", equalTo(1))
                .body("[0].'dbsnp'.'alleles'[0]", nullValue())
                .body("[0].'dbsnp'.'alt'", containsString("C"))
                .body("[0].'dbsnp'.'chrom'", containsString("17"))
                .body("[0].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[0].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[0].'dbsnp'.'flags'[0]", nullValue())
                .body("[0].'dbsnp'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'dbsnp'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[0].'dbsnp'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'rsid'", containsString("rs193920789"))
                .body("[0].'dbsnp'.'vartype'", containsString("snv"))
                .body("[0].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[0].'cosmic'.'alt'", containsString("G"))
                .body("[0].'cosmic'.'chrom'", containsString("17"))
                .body("[0].'cosmic'.'cosmicId'", containsString("COSM44676"))
                .body("[0].'cosmic'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'cosmic'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[0].'cosmic'.'mutFreq'", numberMatches(0.03))
                .body("[0].'cosmic'.'mutNt'", containsString("T>G"))
                .body("[0].'cosmic'.'ref'", containsString("T"))
                .body("[0].'cosmic'.'tumorSite'", containsString("urinary_tract"))
                .body("[0].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[0].'clinVar'.'alleleId'", numberMatches(171615.0))
                .body("[0].'clinVar'.'alt'", containsString("C"))
                .body("[0].'clinVar'.'chrom'", containsString("17"))
                .body("[0].'clinVar'.'cytogenic'", containsString("17p13.1"))
                .body("[0].'clinVar'.'gene'.'symbol'", containsString("TP53"))
                .body("[0].'clinVar'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'clinVar'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[0].'clinVar'.'hg38'.'start'", containsString("7674249"))
                .body("[0].'clinVar'.'hg38'.'end'", containsString("7674249"))
                .body("[0].'clinVar'.'hgvs'.'coding'.size()", equalTo(17))
                .body("[0].'clinVar'.'hgvs'.'coding'", hasItems("LRG_321t1:c.714T>G", "LRG_321t5:c.318T>G", "NM_000546.6:c.714T>G", "NM_001126112.3:c.714T>G", "NM_001126113.3:c.714T>G", "NM_001126114.3:c.714T>G", "NM_001126115.2:c.318T>G", "NM_001126116.2:c.318T>G", "NM_001126117.2:c.318T>G", "NM_001126118.2:c.597T>G", "NM_001276695.3:c.597T>G", "NM_001276696.3:c.597T>G", "NM_001276697.3:c.237T>G", "NM_001276698.3:c.237T>G", "NM_001276699.3:c.237T>G", "NM_001276760.3:c.597T>G", "NM_001276761.3:c.597T>G"))
                .body("[0].'clinVar'.'hgvs'.'genomic'.size()", equalTo(4))
                .body("[0].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_321:g.18302T>G", "NC_000017.10:g.7577567A>C", "NC_000017.11:g.7674249A>C", "NG_017013.2:g.18302T>G"))
                .body("[0].'clinVar'.'hgvs'.'protein'.size()", equalTo(19))
                .body("[0].'clinVar'.'hgvs'.'protein'", hasItems("LRG_321p1:p.Cys238Trp", "LRG_321p5:p.Cys106Trp", "NP_000537.3:p.Cys238Trp", "NP_000537.3:p.Cys238Trp", "NP_001119584.1:p.Cys238Trp", "NP_001119585.1:p.Cys238Trp", "NP_001119586.1:p.Cys238Trp", "NP_001119587.1:p.Cys106Trp", "NP_001119587.1:p.Cys106Trp", "NP_001119588.1:p.Cys106Trp", "NP_001119589.1:p.Cys106Trp", "NP_001119590.1:p.Cys199Trp", "NP_001263624.1:p.Cys199Trp", "NP_001263625.1:p.Cys199Trp", "NP_001263626.1:p.Cys79Trp", "NP_001263627.1:p.Cys79Trp", "NP_001263628.1:p.Cys79Trp", "NP_001263689.1:p.Cys199Trp", "NP_001263690.1:p.Cys199Trp"))
                .body("[0].'clinVar'.'rcv'.size()", equalTo(3))
                .body("[0].'clinVar'.'rcv'[0].'accession'", containsString("RCV000149049"))
                .body("[0].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Uncertain significance"))
                .body("[0].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000546.6(TP53):c.714T>G (p.Cys238Trp)"))
                .body("[0].'clinVar'.'rcv'[1].'accession'", containsString("RCV000203823"))
                .body("[0].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[1].'origin'", containsString("germline"))
                .body("[0].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000546.6(TP53):c.714T>G (p.Cys238Trp)"))
                .body("[0].'clinVar'.'rcv'[2].'accession'", containsString("RCV002362784"))
                .body("[0].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[0].'clinVar'.'rcv'[2].'origin'", containsString("germline"))
                .body("[0].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_000546.6(TP53):c.714T>G (p.Cys238Trp)"))
                .body("[0].'clinVar'.'variantId'", numberMatches(161515.0))
                .body("[0].'mutdb'.'ref'", containsString("T"))
                .body("[0].'mutdb'.'alt'", containsString("G"))
                .body("[0].'mutdb'.'uniprotId'", containsString("VAR_045203"))
                .body("[0].'mutdb'.'mutpredScore'", numberMatches(0.963))
                .body("[0].'mutdb'.'cosmicId'", containsString("44676"))
                .body("[0].'mutdb'.'chrom'", containsString("17"))
                .body("[0].'mutdb'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'mutdb'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[1].'variant'", containsString("chr17:g.7577568C>A"))
                .body("[1].'query'", containsString("chr17:g.7577568C>A"))
                .body("[1].'hgvs'", containsString("chr17:g.7577568C>A"))
                .body("[1].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[1].'vcf'.'alt'", containsString("A"))
                .body("[1].'vcf'.'position'", containsString("7577568"))
                .body("[1].'vcf'.'ref'", containsString("C"))
                .body("[1].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[1].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[1].'dbsnp'.'alleles'[0].'allele'", containsString("C"))
                .body("[1].'dbsnp'.'alleles'[1].'allele'", containsString("T"))
                .body("[1].'dbsnp'.'alt'", containsString("A"))
                .body("[1].'dbsnp'.'chrom'", containsString("17"))
                .body("[1].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[1].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[1].'dbsnp'.'flags'[0]", nullValue())
                .body("[1].'dbsnp'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'dbsnp'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[1].'dbsnp'.'ref'", containsString("C"))
                .body("[1].'dbsnp'.'rsid'", containsString("rs730882005"))
                .body("[1].'dbsnp'.'vartype'", containsString("snv"))
                .body("[1].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[1].'cosmic'.'alt'", containsString("T"))
                .body("[1].'cosmic'.'chrom'", containsString("17"))
                .body("[1].'cosmic'.'cosmicId'", containsString("COSM43778"))
                .body("[1].'cosmic'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'cosmic'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[1].'cosmic'.'mutFreq'", numberMatches(0.63))
                .body("[1].'cosmic'.'mutNt'", containsString("G>T"))
                .body("[1].'cosmic'.'ref'", containsString("G"))
                .body("[1].'cosmic'.'tumorSite'", containsString("meninges"))
                .body("[1].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[1].'clinVar'.'alleleId'", numberMatches(363453.0))
                .body("[1].'clinVar'.'alt'", containsString("A"))
                .body("[1].'clinVar'.'chrom'", containsString("17"))
                .body("[1].'clinVar'.'cytogenic'", containsString("17p13.1"))
                .body("[1].'clinVar'.'gene'.'symbol'", containsString("TP53"))
                .body("[1].'clinVar'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'clinVar'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[1].'clinVar'.'hg38'.'start'", containsString("7674250"))
                .body("[1].'clinVar'.'hg38'.'end'", containsString("7674250"))
                .body("[1].'clinVar'.'hgvs'.'coding'.size()", equalTo(16))
                .body("[1].'clinVar'.'hgvs'.'coding'", hasItems("LRG_321t1:c.713G>T", "NM_000546.6:c.713G>T", "NM_001126112.3:c.713G>T", "NM_001126113.3:c.713G>T", "NM_001126114.3:c.713G>T", "NM_001126115.2:c.317G>T", "NM_001126116.2:c.317G>T", "NM_001126117.2:c.317G>T", "NM_001126118.2:c.596G>T", "NM_001276695.3:c.596G>T", "NM_001276696.3:c.596G>T", "NM_001276697.3:c.236G>T", "NM_001276698.3:c.236G>T", "NM_001276699.3:c.236G>T", "NM_001276760.3:c.596G>T", "NM_001276761.3:c.596G>T"))
                .body("[1].'clinVar'.'hgvs'.'genomic'.size()", equalTo(4))
                .body("[1].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_321:g.18301G>T", "NC_000017.10:g.7577568C>A", "NC_000017.11:g.7674250C>A", "NG_017013.2:g.18301G>T"))
                .body("[1].'clinVar'.'hgvs'.'protein'.size()", equalTo(17))
                .body("[1].'clinVar'.'hgvs'.'protein'", hasItems("LRG_321p1:p.Cys238Phe", "NP_000537.3:p.Cys238Phe", "NP_000537.3:p.Cys238Phe", "NP_001119584.1:p.Cys238Phe", "NP_001119585.1:p.Cys238Phe", "NP_001119586.1:p.Cys238Phe", "NP_001119587.1:p.Cys106Phe", "NP_001119588.1:p.Cys106Phe", "NP_001119589.1:p.Cys106Phe", "NP_001119590.1:p.Cys199Phe", "NP_001263624.1:p.Cys199Phe", "NP_001263625.1:p.Cys199Phe", "NP_001263626.1:p.Cys79Phe", "NP_001263627.1:p.Cys79Phe", "NP_001263628.1:p.Cys79Phe", "NP_001263689.1:p.Cys199Phe", "NP_001263690.1:p.Cys199Phe"))
                .body("[1].'clinVar'.'rcv'.size()", equalTo(19))
                .body("[1].'clinVar'.'rcv'[0].'accession'", containsString("RCV000417498"))
                .body("[1].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[1].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[1].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000546.6(TP53):c.713G>T (p.Cys238Phe)"))
                .body("[1].'clinVar'.'rcv'[1].'accession'", containsString("RCV000418555"))
                .body("[1].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[1].'clinVar'.'rcv'[1].'origin'", containsString("somatic"))
                .body("[1].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000546.6(TP53):c.713G>T (p.Cys238Phe)"))
                .body("[1].'clinVar'.'rcv'[2].'accession'", containsString("RCV000422481"))
                .body("[1].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[1].'clinVar'.'rcv'[2].'origin'", containsString("somatic"))
                .body("[1].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_000546.6(TP53):c.713G>T (p.Cys238Phe)"))
                // Skipping assertions on the remaining 16 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'clinVar'.'variantId'", numberMatches(376574.0))
                .body("[1].'mutdb'.'ref'", containsString("G"))
                .body("[1].'mutdb'.'alt'", containsString("T"))
                .body("[1].'mutdb'.'uniprotId'", containsString("VAR_005966"))
                .body("[1].'mutdb'.'mutpredScore'", numberMatches(0.976))
                .body("[1].'mutdb'.'cosmicId'", containsString("43778, 99626, 99624, 99625"))
                .body("[1].'mutdb'.'chrom'", containsString("17"))
                .body("[1].'mutdb'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'mutdb'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'variant'", containsString("chr17:g.7577568C>G"))
                .body("[2].'query'", containsString("chr17:g.7577568C>G"))
                .body("[2].'hgvs'", containsString("chr17:g.7577568C>G"))
                .body("[2].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[2].'vcf'.'alt'", containsString("G"))
                .body("[2].'vcf'.'position'", containsString("7577568"))
                .body("[2].'vcf'.'ref'", containsString("C"))
                .body("[2].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[2].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[2].'dbsnp'.'alleles'[0].'allele'", containsString("C"))
                .body("[2].'dbsnp'.'alleles'[1].'allele'", containsString("T"))
                .body("[2].'dbsnp'.'alt'", containsString("G"))
                .body("[2].'dbsnp'.'chrom'", containsString("17"))
                .body("[2].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[2].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[2].'dbsnp'.'flags'[0]", nullValue())
                .body("[2].'dbsnp'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'dbsnp'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'dbsnp'.'ref'", containsString("C"))
                .body("[2].'dbsnp'.'rsid'", containsString("rs730882005"))
                .body("[2].'dbsnp'.'vartype'", containsString("snv"))
                .body("[2].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[2].'cosmic'.'alt'", containsString("C"))
                .body("[2].'cosmic'.'chrom'", containsString("17"))
                .body("[2].'cosmic'.'cosmicId'", containsString("COSM44653"))
                .body("[2].'cosmic'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'cosmic'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'cosmic'.'mutFreq'", numberMatches(0.04))
                .body("[2].'cosmic'.'mutNt'", containsString("G>C"))
                .body("[2].'cosmic'.'ref'", containsString("G"))
                .body("[2].'cosmic'.'tumorSite'", containsString("central_nervous_system"))
                .body("[2].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[2].'clinVar'.'alleleId'", numberMatches(363454.0))
                .body("[2].'clinVar'.'alt'", containsString("G"))
                .body("[2].'clinVar'.'chrom'", containsString("17"))
                .body("[2].'clinVar'.'cytogenic'", containsString("17p13.1"))
                .body("[2].'clinVar'.'gene'.'symbol'", containsString("TP53"))
                .body("[2].'clinVar'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'clinVar'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'clinVar'.'hg38'.'start'", containsString("7674250"))
                .body("[2].'clinVar'.'hg38'.'end'", containsString("7674250"))
                .body("[2].'clinVar'.'hgvs'.'coding'.size()", equalTo(16))
                .body("[2].'clinVar'.'hgvs'.'coding'", hasItems("LRG_321t1:c.713G>C", "NM_000546.6:c.713G>C", "NM_001126112.3:c.713G>C", "NM_001126113.3:c.713G>C", "NM_001126114.3:c.713G>C", "NM_001126115.2:c.317G>C", "NM_001126116.2:c.317G>C", "NM_001126117.2:c.317G>C", "NM_001126118.2:c.596G>C", "NM_001276695.3:c.596G>C", "NM_001276696.3:c.596G>C", "NM_001276697.3:c.236G>C", "NM_001276698.3:c.236G>C", "NM_001276699.3:c.236G>C", "NM_001276760.3:c.596G>C", "NM_001276761.3:c.596G>C"))
                .body("[2].'clinVar'.'hgvs'.'genomic'.size()", equalTo(4))
                .body("[2].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_321:g.18301G>C", "NC_000017.10:g.7577568C>G", "NC_000017.11:g.7674250C>G", "NG_017013.2:g.18301G>C"))
                .body("[2].'clinVar'.'hgvs'.'protein'.size()", equalTo(17))
                .body("[2].'clinVar'.'hgvs'.'protein'", hasItems("LRG_321p1:p.Cys238Ser", "NP_000537.3:p.Cys238Ser", "NP_000537.3:p.Cys238Ser", "NP_001119584.1:p.Cys238Ser", "NP_001119585.1:p.Cys238Ser", "NP_001119586.1:p.Cys238Ser", "NP_001119587.1:p.Cys106Ser", "NP_001119588.1:p.Cys106Ser", "NP_001119589.1:p.Cys106Ser", "NP_001119590.1:p.Cys199Ser", "NP_001263624.1:p.Cys199Ser", "NP_001263625.1:p.Cys199Ser", "NP_001263626.1:p.Cys79Ser", "NP_001263627.1:p.Cys79Ser", "NP_001263628.1:p.Cys79Ser", "NP_001263689.1:p.Cys199Ser", "NP_001263690.1:p.Cys199Ser"))
                .body("[2].'clinVar'.'rcv'.size()", equalTo(21))
                .body("[2].'clinVar'.'rcv'[0].'accession'", containsString("RCV000419485"))
                .body("[2].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[2].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[2].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000546.6(TP53):c.713G>C (p.Cys238Ser)"))
                .body("[2].'clinVar'.'rcv'[1].'accession'", containsString("RCV000418923"))
                .body("[2].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[2].'clinVar'.'rcv'[1].'origin'", containsString("somatic"))
                .body("[2].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000546.6(TP53):c.713G>C (p.Cys238Ser)"))
                .body("[2].'clinVar'.'rcv'[2].'accession'", containsString("RCV000423226"))
                .body("[2].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[2].'clinVar'.'rcv'[2].'origin'", containsString("somatic"))
                .body("[2].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_000546.6(TP53):c.713G>C (p.Cys238Ser)"))
                // Skipping assertions on the remaining 18 elements. This limit of 3 elements can be increased in the configurations
                .body("[2].'clinVar'.'variantId'", numberMatches(376575.0))
                .body("[2].'mutdb'.'ref'", containsString("G"))
                .body("[2].'mutdb'.'alt'", containsString("C"))
                .body("[2].'mutdb'.'uniprotId'", containsString("VAR_045202"))
                .body("[2].'mutdb'.'mutpredScore'", numberMatches(0.977))
                .body("[2].'mutdb'.'cosmicId'", containsString("44653"))
                .body("[2].'mutdb'.'chrom'", containsString("17"))
                .body("[2].'mutdb'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'mutdb'.'hg19'.'end'", numberMatches(7577568.0))
                ; // Skipping assertions on the remaining 277 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_66() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/nucleotide_context/7:g.140453136A%3ET?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'molecule'", containsString("dna"))
                .body("'seq'", containsString("CAC"))
                .body("'hgvs'", containsString("7:g.140453136A>T"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_67() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/dbsnp/rs63?" + 
                    "isoformOverrideSource=mskcc&" + 
                    "token=d0km_L73bK0fHyk&" + 
                    "fields=bUs&fields=jYqwBI0lnqDkIJ&fields=_EM_8921_XYZ_&fields=_EM_8922_XYZ_&fields=_EM_8923_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("rs63"))
                .body("'originalVariantQuery'", containsString("rs63"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("7"))
                .body("'start'", numberMatches(1.1620002E7))
                .body("'end'", numberMatches(1.1620002E7))
                .body("'allele_string'", containsString("T/G"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("intron_variant"))
                .body("'transcript_consequences'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000423059"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000423059.4:c.1453+10085A>C"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("G"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000406482"))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("THSD7A"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000005108"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("22207"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_015204.2"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("intron_variant"))
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_68() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/MT:g.10360G%3EA?" + 
                    "token=_EM_25317_XYZ_&" + 
                    "fields=32rmxCI")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("MT:g.10360G>A"))
                .body("'originalVariantQuery'", containsString("MT:g.10360G>A"))
                .body("'hgvsg'", containsString("MT:g.10360G>A"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("MT"))
                .body("'start'", numberMatches(10360.0))
                .body("'end'", numberMatches(10360.0))
                .body("'allele_string'", containsString("G/A"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("missense_variant"))
                .body("'transcript_consequences'.size()", equalTo(26))
                .body("'transcript_consequences'[0].'exon'", containsString("1/1"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000361227"))
                .body("'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000355206.2:p.Ser101Asn"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000361227.2:c.302G>A"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[0].'codons'", containsString("aGt/aAt"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000355206"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(101.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(101.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("MT-ND3"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000198840"))
                .body("'transcript_consequences'[0].'amino_acids'", containsString("S/N"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("7458"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'polyphen_score'", numberMatches(0.569))
                .body("'transcript_consequences'[0].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[0].'sift_prediction'", containsString("deleterious_low_confidence"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000361335"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000354728"))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("MT-ND4L"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000212907"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("7460"))
                .body("'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("upstream_gene_variant"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000361381"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000354961"))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("MT-ND4"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000198886"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("7459"))
                .body("'transcript_consequences'[2].'canonical'", containsString("1"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("upstream_gene_variant"))
                // Skipping assertions on the remaining 23 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_69() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"RTbDChnV2ym9XPK\", " + 
                    " \"start\": -283, " + 
                    " \"end\": 131622, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"_EM_1478_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"gMNYz\", " + 
                    " \"start\": 2098109, " + 
                    " \"end\": 520, " + 
                    " \"referenceAllele\": \"_EM_466_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_467_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"\", " + 
                    " \"start\": -33554293, " + 
                    " \"end\": 16778143, " + 
                    " \"referenceAllele\": \"NA\", " + 
                    " \"variantAllele\": \"NA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_471_XYZ_\", " + 
                    " \"start\": 81, " + 
                    " \"end\": 134218082, " + 
                    " \"referenceAllele\": \"_EM_472_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1479_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=mhpgpR_FWyt&" + 
                    "token=_EM_1477_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("gMNYz:g.2098115_520delins7_XYZ_"))
                .body("[0].'originalVariantQuery'", containsString("gMNYz,2098109,520,_EM_466_XYZ_,_EM_467_XYZ_"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[1].'variant'", containsString("_EM_471_XYZ_:g.85_134218082delins1479_XYZ_"))
                .body("[1].'originalVariantQuery'", containsString("_EM_471_XYZ_,81,134218082,_EM_472_XYZ_,_EM_1479_XYZ_"))
                .body("[1].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_70() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"MT:g.10360G\u003eA\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("MT:g.10360G>A"))
                .body("[0].'originalVariantQuery'", containsString("MT:g.10360G>A"))
                .body("[0].'hgvsg'", containsString("MT:g.10360G>A"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("MT"))
                .body("[0].'start'", numberMatches(10360.0))
                .body("[0].'end'", numberMatches(10360.0))
                .body("[0].'allele_string'", containsString("G/A"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("missense_variant"))
                .body("[0].'transcript_consequences'.size()", equalTo(26))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("1/1"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000361227"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000355206.2:p.Ser101Asn"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000361227.2:c.302G>A"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("aGt/aAt"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000355206"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(101.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(101.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("MT-ND3"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000198840"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("S/N"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("7458"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.569))
                .body("[0].'transcript_consequences'[0].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[0].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious_low_confidence"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000361335"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000354728"))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("MT-ND4L"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000212907"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("7460"))
                .body("[0].'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("upstream_gene_variant"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000361381"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000354961"))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("MT-ND4"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000198886"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("7459"))
                .body("[0].'transcript_consequences'[2].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("upstream_gene_variant"))
                // Skipping assertions on the remaining 23 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_71() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"Tiqza_Zj5iJ\", " + 
                    " \"rs122\", " + 
                    " \"_EM_912_XYZ_\", " + 
                    " \"rs400\", " + 
                    " \"Q5mrTgOYFi\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/dbsnp/?" + 
                    "isoformOverrideSource=_EM_911_XYZ_&" + 
                    "token=mVh9")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(5))
                .body("[0].'variant'", containsString("Tiqza_Zj5iJ"))
                .body("[0].'originalVariantQuery'", containsString("Tiqza_Zj5iJ"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[1].'variant'", containsString("rs122"))
                .body("[1].'originalVariantQuery'", containsString("rs122"))
                .body("[1].'assembly_name'", containsString("GRCh37"))
                .body("[1].'seq_region_name'", containsString("7"))
                .body("[1].'start'", numberMatches(2.496659E7))
                .body("[1].'end'", numberMatches(2.496659E7))
                .body("[1].'allele_string'", containsString("G/A/C/T"))
                .body("[1].'strand'", numberMatches(1.0))
                .body("[1].'most_severe_consequence'", containsString("intron_variant"))
                .body("[1].'transcript_consequences'.size()", equalTo(15))
                .body("[1].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000313367"))
                .body("[1].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000313367.2:c.-149-34350C>T"))
                .body("[1].'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000315410"))
                .body("[1].'transcript_consequences'[0].'gene_symbol'", containsString("OSBPL3"))
                .body("[1].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000070882"))
                .body("[1].'transcript_consequences'[0].'hgnc_id'", containsString("16370"))
                .body("[1].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[1].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_015550.2"))
                .body("[1].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[0].'consequence_terms'", hasItems("intron_variant"))
                .body("[1].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000313367"))
                .body("[1].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000313367.2:c.-149-34350C>G"))
                .body("[1].'transcript_consequences'[1].'variant_allele'", containsString("C"))
                .body("[1].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000315410"))
                .body("[1].'transcript_consequences'[1].'gene_symbol'", containsString("OSBPL3"))
                .body("[1].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000070882"))
                .body("[1].'transcript_consequences'[1].'hgnc_id'", containsString("16370"))
                .body("[1].'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("[1].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_015550.2"))
                .body("[1].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[1].'consequence_terms'", hasItems("intron_variant"))
                .body("[1].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000313367"))
                .body("[1].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000313367.2:c.-149-34350C>A"))
                .body("[1].'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("[1].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000315410"))
                .body("[1].'transcript_consequences'[2].'gene_symbol'", containsString("OSBPL3"))
                .body("[1].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000070882"))
                .body("[1].'transcript_consequences'[2].'hgnc_id'", containsString("16370"))
                .body("[1].'transcript_consequences'[2].'canonical'", containsString("1"))
                .body("[1].'transcript_consequences'[2].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[2].'refseq_transcript_ids'", hasItems("NM_015550.2"))
                .body("[1].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[2].'consequence_terms'", hasItems("intron_variant"))
                // Skipping assertions on the remaining 12 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'successfully_annotated'", equalTo(true))
                .body("[2].'variant'", containsString("_EM_912_XYZ_"))
                .body("[2].'originalVariantQuery'", containsString("_EM_912_XYZ_"))
                .body("[2].'successfully_annotated'", equalTo(false))
                ; // Skipping assertions on the remaining 2 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(4)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_72() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"x\", " + 
                    " \"start\": 32, " + 
                    " \"end\": 0, " + 
                    " \"referenceAllele\": \"9KwHc8K0Zb1in6\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"B3SFN\", " + 
                    " \"start\": 592, " + 
                    " \"end\": -67109, " + 
                    " \"referenceAllele\": \"Mt_yhDM3odmDGf\", " + 
                    " \"variantAllele\": \"Mt_ygDM3odmDGf\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'genomicLocation'.'chromosome'", containsString("x"))
                .body("[0].'genomicLocation'.'start'", numberMatches(32.0))
                .body("[0].'genomicLocation'.'end'", numberMatches(0.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("9KwHc8K0Zb1in6"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'variant'", containsString("x:g.32_0del"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("B3SFN"))
                .body("[1].'genomicLocation'.'start'", numberMatches(592.0))
                .body("[1].'genomicLocation'.'end'", numberMatches(-67109.0))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("Mt_yhDM3odmDGf"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("Mt_ygDM3odmDGf"))
                .body("[1].'variant'", containsString("B3SFN:g.596_-67109delinsgDM3odmDGf"))
                .body("[1].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_73() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"_EM_682_XYZ_\", " + 
                    " \"start\": -301279840, " + 
                    " \"end\": 582, " + 
                    " \"referenceAllele\": \"_EM_683_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_684_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_685_XYZ_\", " + 
                    " \"start\": 424, " + 
                    " \"end\": 150, " + 
                    " \"referenceAllele\": \"_EM_686_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_687_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_688_XYZ_\", " + 
                    " \"start\": 440402638, " + 
                    " \"end\": 729, " + 
                    " \"referenceAllele\": \"_EM_689_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_690_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_691_XYZ_\", " + 
                    " \"start\": -2109159905, " + 
                    " \"end\": 856, " + 
                    " \"referenceAllele\": \"_EM_692_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_693_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_694_XYZ_\", " + 
                    " \"start\": 56, " + 
                    " \"end\": 653, " + 
                    " \"referenceAllele\": \"_EM_695_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_696_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?fields=")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(3))
                .body("[0].'variant'", containsString("_EM_685_XYZ_:g.430_150delins7_XYZ_"))
                .body("[0].'originalVariantQuery'", containsString("_EM_685_XYZ_,424,150,_EM_686_XYZ_,_EM_687_XYZ_"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[1].'variant'", containsString("_EM_688_XYZ_:g.440402643_729delins90_XYZ_"))
                .body("[1].'originalVariantQuery'", containsString("_EM_688_XYZ_,440402638,729,_EM_689_XYZ_,_EM_690_XYZ_"))
                .body("[1].'successfully_annotated'", equalTo(false))
                .body("[2].'variant'", containsString("_EM_694_XYZ_:g.62_653delins6_XYZ_"))
                .body("[2].'originalVariantQuery'", containsString("_EM_694_XYZ_,56,653,_EM_695_XYZ_,_EM_696_XYZ_"))
                .body("[2].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_74() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"8\", " + 
                    " \"start\": 620, " + 
                    " \"end\": 630, " + 
                    " \"referenceAllele\": \"19zVotO4LeaFTQj\", " + 
                    " \"variantAllele\": \"6\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"k\", " + 
                    " \"start\": 362, " + 
                    " \"end\": 370, " + 
                    " \"referenceAllele\": \"hWhurEf4Q\", " + 
                    " \"variantAllele\": \"Z\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"\", " + 
                    " \"start\": 360, " + 
                    " \"end\": 195, " + 
                    " \"referenceAllele\": \"M\", " + 
                    " \"variantAllele\": \"1lHvsbNc1_uwsNOe\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(3))
                .body("[0].'genomicLocation'.'chromosome'", containsString("8"))
                .body("[0].'genomicLocation'.'start'", numberMatches(620.0))
                .body("[0].'genomicLocation'.'end'", numberMatches(630.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("19zVotO4LeaFTQj"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("6"))
                .body("[0].'variant'", containsString("8:g.620_630delins6"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("k"))
                .body("[1].'genomicLocation'.'start'", numberMatches(362.0))
                .body("[1].'genomicLocation'.'end'", numberMatches(370.0))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("hWhurEf4Q"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("Z"))
                .body("[1].'variant'", containsString("k:g.362_370delinsZ"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString(""))
                .body("[2].'genomicLocation'.'start'", numberMatches(360.0))
                .body("[2].'genomicLocation'.'end'", numberMatches(195.0))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("M"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("1lHvsbNc1_uwsNOe"))
                .body("[2].'variant'", containsString(":g.360delins1lHvsbNc1_uwsNOe"))
                .body("[2].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_75() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 73344272, " + 
                    " \"end\": 140453072, " + 
                    " \"referenceAllele\": \"_EM_2533_XYZ_\", " + 
                    " \"variantAllele\": \"NA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=&" + 
                    "token=h7vPh&" + 
                    "fields=annotation_summary&fields=_EM_2341_XYZ_&fields=T_AJHMfJ")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("7:g.73344272_140453072del"))
                .body("[0].'originalVariantQuery'", containsString("7,73344272,140453072,_EM_2533_XYZ_,NA"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_76() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/cancer_hotspots/genomic/19,10267171,10267171,T,C?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_77() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/genomic/7,140453136,140453136,A,T?" + 
                    "isoformOverrideSource=genome_nexus&" + 
                    "token=_EM_3958_XYZ_&" + 
                    "fields=_EM_3962_XYZ_&fields=_EM_2416_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("7:g.140453136A>T"))
                .body("'originalVariantQuery'", containsString("7,140453136,140453136,A,T"))
                .body("'hgvsg'", containsString("7:g.140453136A>T"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("7"))
                .body("'start'", numberMatches(1.40453136E8))
                .body("'end'", numberMatches(1.40453136E8))
                .body("'allele_string'", containsString("A/T"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("missense_variant"))
                .body("'transcript_consequences'.size()", equalTo(4))
                .body("'transcript_consequences'[0].'exon'", containsString("15/18"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000288602"))
                .body("'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000288602.6:p.Val600Glu"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[0].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000288602"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[0].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'polyphen_score'", numberMatches(0.963))
                .body("'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_004333.4"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[1].'exon'", containsString("2/6"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000479537"))
                .body("'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000418033.1:p.Val28Glu"))
                .body("'transcript_consequences'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[1].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000418033"))
                .body("'transcript_consequences'[1].'protein_start'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'protein_end'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[1].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[1].'polyphen_score'", numberMatches(0.991))
                .body("'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[1].'sift_score'", numberMatches(0.01))
                .body("'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(2))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant", "NMD_transcript_variant"))
                .body("'transcript_consequences'[2].'exon'", containsString("6/10"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000496384"))
                .body("'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000419060.1:p.Val208Glu"))
                .body("'transcript_consequences'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[2].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000419060"))
                .body("'transcript_consequences'[2].'protein_start'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'protein_end'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[2].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[2].'polyphen_score'", numberMatches(0.817))
                .body("'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_78() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1339_XYZ_\", " + 
                    " \"start\": 140453189, " + 
                    " \"end\": 4, " + 
                    " \"referenceAllele\": \"_EM_1340_XYZ_\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_2685_XYZ_\", " + 
                    " \"start\": 37880228, " + 
                    " \"end\": 37945754, " + 
                    " \"referenceAllele\": \"_EM_2686_XYZ_\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37881834, " + 
                    " \"end\": 37945330, " + 
                    " \"referenceAllele\": \"\", " + 
                    " \"variantAllele\": \"_EM_1915_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_605_XYZ_\", " + 
                    " \"start\": 25398412, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"_EM_2687_XYZ_\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1341_XYZ_\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_2688_XYZ_\", " + 
                    " \"start\": 16777216, " + 
                    " \"end\": 0, " + 
                    " \"referenceAllele\": \"_EM_1342_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1343_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("_EM_1339_XYZ_"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453189E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(4.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("_EM_1340_XYZ_"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("_EM_1339_XYZ_:g.140453189_4delinsC"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("_EM_2685_XYZ_"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.7880228E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.7945754E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("_EM_2686_XYZ_"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[1].'variant'", containsString("_EM_2685_XYZ_:g.37880228_37945754del"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7881834E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.794533E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString(""))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("_EM_1915_XYZ_"))
                .body("[2].'variant'", containsString("17:g.37881834_37881835ins_EM_1915_XYZ_"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_79() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"7:g.140453136A\u003eT\", " + 
                    " \"12:g.25398285C\u003eA\", " + 
                    " \"17:g.41242962_41242963insGA\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/nucleotide_context?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'molecule'", containsString("dna"))
                .body("[0].'seq'", containsString("CAC"))
                .body("[0].'hgvs'", containsString("7:g.140453136A>T"))
                .body("[1].'molecule'", containsString("dna"))
                .body("[1].'seq'", containsString("CCA"))
                .body("[1].'hgvs'", containsString("12:g.25398285C>A"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_80() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 55242466, " + 
                    " \"end\": 55242480, " + 
                    " \"referenceAllele\": \"GAATTAAGAGAAGCA\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 55249002, " + 
                    " \"end\": 55249003, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"CAGCGTGGA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(5.5242466E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(5.524248E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("GAATTAAGAGAAGCA"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'variant'", containsString("7:g.55242466_55242480del"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[1].'genomicLocation'.'start'", numberMatches(5.5249002E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(5.5249003E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("-"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("CAGCGTGGA"))
                .body("[1].'variant'", containsString("7:g.55249002_55249003insCAGCGTGGA"))
                .body("[1].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_81() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/summary/9:g.21994291C%3ET?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("9:g.21994291C>T"))
                .body("'genomicLocation'.'chromosome'", containsString("9"))
                .body("'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("'genomicLocation'.'variantAllele'", containsString("T"))
                .body("'strandSign'", containsString("+"))
                .body("'variantType'", containsString("SNP"))
                .body("'assemblyName'", containsString("GRCh37"))
                .body("'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequences'.size()", equalTo(23))
                .body("'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("'transcriptConsequences'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("'transcriptConsequences'[0].'aminoAcids'", containsString("A/T"))
                .body("'transcriptConsequences'[0].'aminoAcidRef'", containsString("A"))
                .body("'transcriptConsequences'[0].'aminoAcidAlt'", containsString("T"))
                .body("'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("'transcriptConsequences'[0].'hgvspShort'", containsString("p.A55T"))
                .body("'transcriptConsequences'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("'transcriptConsequences'[0].'refSeq'", containsString("NM_058195.3"))
                .body("'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'transcriptConsequences'[0].'exon'", containsString("1/3"))
                .body("'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.014))
                .body("'transcriptConsequences'[0].'polyphenPrediction'", containsString("benign"))
                .body("'transcriptConsequences'[0].'siftScore'", numberMatches(0.09))
                .body("'transcriptConsequences'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("'transcriptConsequences'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequences'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequences'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequences'[1].'variantClassification'", containsString("5'Flank"))
                .body("'transcriptConsequences'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("'transcriptConsequences'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequences'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequences'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("'transcriptConsequenceSummaries'.size()", equalTo(23))
                .body("'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("'transcriptConsequenceSummaries'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("A/T"))
                .body("'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("A"))
                .body("'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("T"))
                .body("'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.A55T"))
                .body("'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_058195.3"))
                .body("'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'transcriptConsequenceSummaries'[0].'exon'", containsString("1/3"))
                .body("'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.014))
                .body("'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("benign"))
                .body("'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.09))
                .body("'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("5'Flank"))
                .body("'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_82() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"9:g.21994291C\u003eT\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?projection=CANONICAL")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("9:g.21994291C>T"))
                .body("[0].'genomicLocation'.'chromosome'", containsString("9"))
                .body("[0].'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'strandSign'", containsString("+"))
                .body("[0].'variantType'", containsString("SNP"))
                .body("[0].'assemblyName'", containsString("GRCh37"))
                .body("[0].'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'[0].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[0].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_83() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_602_XYZ_\", " + 
                    " \"start\": 37880228, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"_EM_603_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_604_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879794, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_605_XYZ_\", " + 
                    " \"start\": 25398412, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("7:g.140453193T>C"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("_EM_602_XYZ_"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.7880228E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("_EM_603_XYZ_"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("_EM_604_XYZ_"))
                .body("[1].'variant'", containsString("_EM_602_XYZ_:g.37880234_37880220delins4_XYZ_"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[2].'variant'", containsString("17:g.37879794G>C"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_84() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"9:g.21994291C\u003eT\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("9:g.21994291C>T"))
                .body("[0].'genomicLocation'.'chromosome'", containsString("9"))
                .body("[0].'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'strandSign'", containsString("+"))
                .body("[0].'variantType'", containsString("SNP"))
                .body("[0].'assemblyName'", containsString("GRCh37"))
                .body("[0].'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'.size()", equalTo(23))
                .body("[0].'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("[0].'transcriptConsequences'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("[0].'transcriptConsequences'[0].'aminoAcids'", containsString("A/T"))
                .body("[0].'transcriptConsequences'[0].'aminoAcidRef'", containsString("A"))
                .body("[0].'transcriptConsequences'[0].'aminoAcidAlt'", containsString("T"))
                .body("[0].'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("[0].'transcriptConsequences'[0].'hgvspShort'", containsString("p.A55T"))
                .body("[0].'transcriptConsequences'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("[0].'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("[0].'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("[0].'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("[0].'transcriptConsequences'[0].'refSeq'", containsString("NM_058195.3"))
                .body("[0].'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'transcriptConsequences'[0].'exon'", containsString("1/3"))
                .body("[0].'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.014))
                .body("[0].'transcriptConsequences'[0].'polyphenPrediction'", containsString("benign"))
                .body("[0].'transcriptConsequences'[0].'siftScore'", numberMatches(0.09))
                .body("[0].'transcriptConsequences'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("[0].'transcriptConsequences'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[1].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequences'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("[0].'transcriptConsequences'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'transcriptConsequenceSummaries'.size()", equalTo(23))
                .body("[0].'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("[0].'transcriptConsequenceSummaries'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("A/T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.A55T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("[0].'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("[0].'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_058195.3"))
                .body("[0].'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'transcriptConsequenceSummaries'[0].'exon'", containsString("1/3"))
                .body("[0].'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.014))
                .body("[0].'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("benign"))
                .body("[0].'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.09))
                .body("[0].'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("[0].'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("[0].'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_85() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37880220, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879794, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692936, " + 
                    " \"end\": 358128396, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"_EM_1792_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("7:g.140453193T>C"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[1].'variant'", containsString("17:g.37880220T>C"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[2].'variant'", containsString("17:g.37879794G>C"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_86() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140445001, " + 
                    " \"end\": 140452681, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"_EM_1798_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 33685916, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37617650, " + 
                    " \"end\": 37912562, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"_EM_1799_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1800_XYZ_\", " + 
                    " \"start\": 25398220, " + 
                    " \"end\": 25406477, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"_EM_1801_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380015, " + 
                    " \"referenceAllele\": \"_EM_1802_XYZ_\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40445001E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40452681E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("_EM_1798_XYZ_"))
                .body("[0].'variant'", containsString("7:g.140445001delins_EM_1798_XYZ_"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.3685916E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[1].'variant'", containsString("17:g.33685916T>C"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.761765E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7912562E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("_EM_1799_XYZ_"))
                .body("[2].'variant'", containsString("17:g.37617650delins_EM_1799_XYZ_"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_87() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/genomic/7,140453136,140453136,A,T?" + 
                    "isoformOverrideSource=foo&" + 
                    "fields=annotation_summary&fields=nucleotide_context")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("7:g.140453136A>T"))
                .body("'originalVariantQuery'", containsString("7,140453136,140453136,A,T"))
                .body("'hgvsg'", containsString("7:g.140453136A>T"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("7"))
                .body("'start'", numberMatches(1.40453136E8))
                .body("'end'", numberMatches(1.40453136E8))
                .body("'allele_string'", containsString("A/T"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("missense_variant"))
                .body("'transcript_consequences'.size()", equalTo(4))
                .body("'transcript_consequences'[0].'exon'", containsString("15/18"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000288602"))
                .body("'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000288602.6:p.Val600Glu"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[0].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000288602"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[0].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'polyphen_score'", numberMatches(0.963))
                .body("'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_004333.4"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[1].'exon'", containsString("2/6"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000479537"))
                .body("'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000418033.1:p.Val28Glu"))
                .body("'transcript_consequences'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[1].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000418033"))
                .body("'transcript_consequences'[1].'protein_start'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'protein_end'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[1].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[1].'polyphen_score'", numberMatches(0.991))
                .body("'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[1].'sift_score'", numberMatches(0.01))
                .body("'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(2))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant", "NMD_transcript_variant"))
                .body("'transcript_consequences'[2].'exon'", containsString("6/10"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000496384"))
                .body("'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000419060.1:p.Val208Glu"))
                .body("'transcript_consequences'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[2].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000419060"))
                .body("'transcript_consequences'[2].'protein_start'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'protein_end'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[2].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[2].'polyphen_score'", numberMatches(0.817))
                .body("'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true))
                .body("'nucleotide_context'.'license'", nullValue())
                .body("'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("'nucleotide_context'.'annotation'.'seq'", containsString("CAC"))
                .body("'nucleotide_context'.'annotation'.'hgvs'", containsString("7:g.140453136A>T"))
                .body("'annotation_summary'.'variant'", containsString("7:g.140453136A>T"))
                .body("'annotation_summary'.'genomicLocation'.'chromosome'", containsString("7"))
                .body("'annotation_summary'.'genomicLocation'.'start'", numberMatches(1.40453136E8))
                .body("'annotation_summary'.'genomicLocation'.'end'", numberMatches(1.40453136E8))
                .body("'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("A"))
                .body("'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("T"))
                .body("'annotation_summary'.'strandSign'", containsString("+"))
                .body("'annotation_summary'.'variantType'", containsString("SNP"))
                .body("'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.V600E"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.963))
                .body("'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.V600E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.963))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000479537"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant,NMD_transcript_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.V28E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Val28Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(28.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(28.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/6"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.991))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.01))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000496384"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.V208E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Val208Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(208.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(208.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("6/10"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.817))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.V600E"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Val600Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.963))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_88() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"4\", " + 
                    " \"start\": 55152094, " + 
                    " \"end\": 55152107, " + 
                    " \"referenceAllele\": \"ATCATGCATGATT\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=foo&" + 
                    "fields=annotation_summary&fields=nucleotide_context&fields=Rk4Isyv")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("4:g.55152095_55152107del"))
                .body("[0].'originalVariantQuery'", containsString("4,55152094,55152107,ATCATGCATGATT,A"))
                .body("[0].'hgvsg'", containsString("4:g.55152095_55152107del"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("4"))
                .body("[0].'start'", numberMatches(5.5152095E7))
                .body("[0].'end'", numberMatches(5.5152107E7))
                .body("[0].'allele_string'", containsString("ATCATGCATGATT/-"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("frameshift_variant"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("18/23"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000257290"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000257290.5:p.Ile843ArgfsTer13"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000257290.5:c.2527_2539del"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("-"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("ATCATGCATGATTcg/cg"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000257290"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(843.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(847.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("PDGFRA"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000134853"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("IMHDS/X"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("8803"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_006206.4"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("frameshift_variant"))
                .body("[0].'transcript_consequences'[1].'exon'", containsString("19/24"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000507166"))
                .body("[0].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000423325.1:p.Ile603ArgfsTer13"))
                .body("[0].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000507166.1:c.1807_1819del"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("-"))
                .body("[0].'transcript_consequences'[1].'codons'", containsString("ATCATGCATGATTcg/cg"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000423325"))
                .body("[0].'transcript_consequences'[1].'protein_start'", numberMatches(603.0))
                .body("[0].'transcript_consequences'[1].'protein_end'", numberMatches(607.0))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("FIP1L1"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000145216"))
                .body("[0].'transcript_consequences'[1].'amino_acids'", containsString("IMHDS/X"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("19124"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("frameshift_variant"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000507536"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("-"))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("PDGFRA"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000134853"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("8803"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("downstream_gene_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'annotation_summary'.'variant'", containsString("4:g.55152095_55152107del"))
                .body("[0].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("4"))
                .body("[0].'annotation_summary'.'genomicLocation'.'start'", numberMatches(5.5152095E7))
                .body("[0].'annotation_summary'.'genomicLocation'.'end'", numberMatches(5.5152107E7))
                .body("[0].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("ATCATGCATGATT"))
                .body("[0].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[0].'annotation_summary'.'variantType'", containsString("DEL"))
                .body("[0].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[0].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000257290"))
                .body("[0].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000257290"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("ATCATGCATGATTcg/cg"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("IMHDS/X"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("IMHDS"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("X"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("frameshift_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("PDGFRA"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.I843Rfs*13"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Ile843ArgfsTer13"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000257290.5:c.2527_2539del"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(843.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(847.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_006206.4"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Frame_Shift_Del"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("18/23"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000257290"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("ATCATGCATGATTcg/cg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("IMHDS/X"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("IMHDS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("X"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("frameshift_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("PDGFRA"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.I843Rfs*13"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Ile843ArgfsTer13"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000257290.5:c.2527_2539del"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(843.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(847.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_006206.4"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Frame_Shift_Del"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("18/23"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000507166"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("ATCATGCATGATTcg/cg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("IMHDS/X"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("IMHDS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("X"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("frameshift_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("FIP1L1"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.I603Rfs*13"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Ile603ArgfsTer13"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000507166.1:c.1807_1819del"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(603.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(607.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Frame_Shift_Del"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("19/24"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000507536"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("downstream_gene_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("PDGFRA"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("3'Flank"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000257290"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("ATCATGCATGATTcg/cg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("IMHDS/X"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("IMHDS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("X"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("frameshift_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("PDGFRA"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.I843Rfs*13"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Ile843ArgfsTer13"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000257290.5:c.2527_2539del"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(843.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(847.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_006206.4"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Frame_Shift_Del"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("18/23"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_89() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"4\", " + 
                    " \"start\": 55151585, " + 
                    " \"end\": 55152107, " + 
                    " \"referenceAllele\": \"E5VVWQa\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=ensembl&" + 
                    "token=goo&" + 
                    "fields=annotation_summary&fields=A%60")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("4:g.55151585_55152107del"))
                .body("[0].'originalVariantQuery'", containsString("4,55151585,55152107,E5VVWQa,-"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_90() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"4\", " + 
                    " \"start\": 55135711, " + 
                    " \"end\": 55152107, " + 
                    " \"referenceAllele\": \"ATCATGCATGATT\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=foo&" + 
                    "token=foo&" + 
                    "fields=annotation_summary&fields=_EM_4775_XYZ_&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("4:g.55135712_55152107del"))
                .body("[0].'originalVariantQuery'", containsString("4,55135711,55152107,ATCATGCATGATT,A"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_91() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"4\", " + 
                    " \"start\": 55151583, " + 
                    " \"end\": 55152107, " + 
                    " \"referenceAllele\": \"ATCATGCATGATT\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=_EM_4881_XYZ_&" + 
                    "fields=annotation_summary&fields=_EM_4880_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("4:g.55151584_55152107del"))
                .body("[0].'originalVariantQuery'", containsString("4,55151583,55152107,ATCATGCATGATT,A"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_92() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453136, " + 
                    " \"end\": 140453136, " + 
                    " \"referenceAllele\": \"A\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("7:g.140453136A>T"))
                .body("[0].'originalVariantQuery'", containsString("7,140453136,140453136,A,T"))
                .body("[0].'hgvsg'", containsString("7:g.140453136A>T"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("7"))
                .body("[0].'start'", numberMatches(1.40453136E8))
                .body("[0].'end'", numberMatches(1.40453136E8))
                .body("[0].'allele_string'", containsString("A/T"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("missense_variant"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("15/18"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000288602"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000288602.6:p.Val600Glu"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("gTg/gAg"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000288602"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(600.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(600.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("BRAF"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000157764"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("V/E"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("1097"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.963))
                .body("[0].'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_004333.4"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[1].'exon'", containsString("2/6"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000479537"))
                .body("[0].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000418033.1:p.Val28Glu"))
                .body("[0].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[1].'codons'", containsString("gTg/gAg"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000418033"))
                .body("[0].'transcript_consequences'[1].'protein_start'", numberMatches(28.0))
                .body("[0].'transcript_consequences'[1].'protein_end'", numberMatches(28.0))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("BRAF"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000157764"))
                .body("[0].'transcript_consequences'[1].'amino_acids'", containsString("V/E"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("1097"))
                .body("[0].'transcript_consequences'[1].'polyphen_score'", numberMatches(0.991))
                .body("[0].'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[1].'sift_score'", numberMatches(0.01))
                .body("[0].'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(2))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant", "NMD_transcript_variant"))
                .body("[0].'transcript_consequences'[2].'exon'", containsString("6/10"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000496384"))
                .body("[0].'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000419060.1:p.Val208Glu"))
                .body("[0].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[2].'codons'", containsString("gTg/gAg"))
                .body("[0].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000419060"))
                .body("[0].'transcript_consequences'[2].'protein_start'", numberMatches(208.0))
                .body("[0].'transcript_consequences'[2].'protein_end'", numberMatches(208.0))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("BRAF"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000157764"))
                .body("[0].'transcript_consequences'[2].'amino_acids'", containsString("V/E"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("1097"))
                .body("[0].'transcript_consequences'[2].'polyphen_score'", numberMatches(0.817))
                .body("[0].'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[0].'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'nucleotide_context'.'license'", nullValue())
                .body("[0].'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("[0].'nucleotide_context'.'annotation'.'seq'", containsString("CAC"))
                .body("[0].'nucleotide_context'.'annotation'.'hgvs'", containsString("7:g.140453136A>T"))
                .body("[0].'annotation_summary'.'variant'", containsString("7:g.140453136A>T"))
                .body("[0].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'annotation_summary'.'genomicLocation'.'start'", numberMatches(1.40453136E8))
                .body("[0].'annotation_summary'.'genomicLocation'.'end'", numberMatches(1.40453136E8))
                .body("[0].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("A"))
                .body("[0].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[0].'annotation_summary'.'variantType'", containsString("SNP"))
                .body("[0].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[0].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.V600E"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_004333.4"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("15/18"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.963))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.V600E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_004333.4"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("15/18"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.963))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000479537"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant,NMD_transcript_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.V28E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Val28Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(28.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(28.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.991))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.01))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000496384"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.V208E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Val208Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(208.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(208.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("6/10"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.817))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.V600E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Val600Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_004333.4"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("15/18"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.963))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_93() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453136, " + 
                    " \"end\": 140518672, " + 
                    " \"referenceAllele\": \"_EM_2423_XYZ_\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=foo&" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("7:g.140453136_140518672delinsT"))
                .body("[0].'originalVariantQuery'", containsString("7,140453136,140518672,_EM_2423_XYZ_,T"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_94() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89624245, " + 
                    " \"end\": 89624245, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10.0\", " + 
                    " \"start\": 89624242, " + 
                    " \"end\": 89624243, " + 
                    " \"referenceAllele\": \"AA\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "fields=my_variant_info&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("10:g.89624245G>T"))
                .body("[0].'originalVariantQuery'", containsString("10,89624245,89624245,G,T"))
                .body("[0].'hgvsg'", containsString("10:g.89624245G>T"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("10"))
                .body("[0].'start'", numberMatches(8.9624245E7))
                .body("[0].'end'", numberMatches(8.9624245E7))
                .body("[0].'allele_string'", containsString("G/T"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("stop_gained"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("1/9"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000371953"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000361021.3:p.Glu7Ter"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000371953.3:c.19G>T"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("Gag/Tag"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000361021"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(7.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(7.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("PTEN"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000171862"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("E/*"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("9588"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_000314.4"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("stop_gained"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000445946"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000392204"))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("KLLN"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000227268"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("37212"))
                .body("[0].'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_001126049.1"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("upstream_gene_variant"))
                .body("[0].'transcript_consequences'[2].'exon'", containsString("1/2"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000462694"))
                .body("[0].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000462694.1:n.21G>T"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("PTEN"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000171862"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("9588"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("non_coding_transcript_exon_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'my_variant_info'.'license'", nullValue())
                .body("[0].'my_variant_info'.'annotation'.'variant'", containsString("chr10:g.89624245G>T"))
                .body("[0].'my_variant_info'.'annotation'.'query'", containsString("chr10:g.89624245G>T"))
                .body("[0].'my_variant_info'.'annotation'.'hgvs'", containsString("10:g.89624245G>T"))
                .body("[0].'my_variant_info'.'annotation'.'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[0].'my_variant_info'.'annotation'.'vcf'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'vcf'.'position'", containsString("89624245"))
                .body("[0].'my_variant_info'.'annotation'.'vcf'.'ref'", containsString("G"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'alleles'.size()", equalTo(1))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'alleles'[0]", nullValue())
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'chrom'", containsString("10"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'flags'.size()", equalTo(1))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'flags'[0]", nullValue())
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'hg19'.'start'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'hg19'.'end'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'ref'", containsString("G"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'rsid'", containsString("rs1554890335"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'vartype'", containsString("snv"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'chrom'", containsString("10"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'cosmicId'", containsString("COSM5298"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'hg19'.'start'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'hg19'.'end'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'mutFreq'", numberMatches(0.19))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'mutNt'", containsString("G>T"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'ref'", containsString("G"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'tumorSite'", containsString("large_intestine"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'alleleId'", numberMatches(486042.0))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'chrom'", containsString("10"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'cytogenic'", containsString("10q23.31"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'gene'.'symbol'", containsString("PTEN"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg19'.'start'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg19'.'end'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg38'.'start'", containsString("87864488"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg38'.'end'", containsString("87864488"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'coding'.size()", equalTo(3))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'coding'", hasItems("NM_000314.8:c.19G>T", "NM_001304717.5:c.538G>T", "NM_001304718.2:c.-687G>T"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'genomic'.size()", equalTo(6))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'genomic'", hasItems("LRG_1087:g.3950C>A", "LRG_311:g.6050G>T", "NC_000010.10:g.89624245G>T", "NC_000010.11:g.87864488G>T", "NG_007466.2:g.6050G>T", "NG_033079.1:g.3950C>A"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'protein'.size()", equalTo(3))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'protein'", hasItems("", "NP_000305.3:p.Glu7Ter", "NP_001291646.4:p.Glu180Ter"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'.size()", equalTo(2))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'accession'", containsString("RCV000585486"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'origin'", containsString("germline"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000314.8(PTEN):c.19G>T (p.Glu7Ter)"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'accession'", containsString("RCV003451323"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'origin'", containsString("unknown"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000314.8(PTEN):c.19G>T (p.Glu7Ter)"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'variantId'", numberMatches(493079.0))
                .body("[1].'variant'", containsString("10.0:g.89624242_89624243del"))
                .body("[1].'originalVariantQuery'", containsString("10.0,89624242,89624243,AA,-"))
                .body("[1].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_95() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": 25398284, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398283, " + 
                    " \"end\": 25398284, " + 
                    " \"referenceAllele\": \"AC\", " + 
                    " \"variantAllele\": \"AA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("12:g.25398284C>A"))
                .body("[0].'originalVariantQuery'", containsString("12,25398284,25398284,C,A"))
                .body("[0].'hgvsg'", containsString("12:g.25398284C>A"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("12"))
                .body("[0].'start'", numberMatches(2.5398284E7))
                .body("[0].'end'", numberMatches(2.5398284E7))
                .body("[0].'allele_string'", containsString("C/A"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("missense_variant"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("2/6"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000256078"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000256078.4:p.Gly12Val"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("gGt/gTt"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000256078"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("KRAS"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000133703"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("G/V"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("6407"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.972))
                .body("[0].'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_033360.2"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[1].'exon'", containsString("2/5"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000311936"))
                .body("[0].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000308495.3:p.Gly12Val"))
                .body("[0].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[1].'codons'", containsString("gGt/gTt"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000308495"))
                .body("[0].'transcript_consequences'[1].'protein_start'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[1].'protein_end'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("KRAS"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000133703"))
                .body("[0].'transcript_consequences'[1].'amino_acids'", containsString("G/V"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("6407"))
                .body("[0].'transcript_consequences'[1].'polyphen_score'", numberMatches(0.96))
                .body("[0].'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[1].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_004985.3"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[2].'exon'", containsString("2/3"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000556131"))
                .body("[0].'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000451856.1:p.Gly12Val"))
                .body("[0].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[2].'codons'", containsString("gGt/gTt"))
                .body("[0].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000451856"))
                .body("[0].'transcript_consequences'[2].'protein_start'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[2].'protein_end'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("KRAS"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000133703"))
                .body("[0].'transcript_consequences'[2].'amino_acids'", containsString("G/V"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("6407"))
                .body("[0].'transcript_consequences'[2].'polyphen_score'", numberMatches(0.884))
                .body("[0].'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[0].'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'nucleotide_context'.'license'", nullValue())
                .body("[0].'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("[0].'nucleotide_context'.'annotation'.'seq'", containsString("ACC"))
                .body("[0].'nucleotide_context'.'annotation'.'hgvs'", containsString("12:g.25398284C>A"))
                .body("[0].'annotation_summary'.'variant'", containsString("12:g.25398284C>A"))
                .body("[0].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("12"))
                .body("[0].'annotation_summary'.'genomicLocation'.'start'", numberMatches(2.5398284E7))
                .body("[0].'annotation_summary'.'genomicLocation'.'end'", numberMatches(2.5398284E7))
                .body("[0].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("A"))
                .body("[0].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[0].'annotation_summary'.'variantType'", containsString("SNP"))
                .body("[0].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[0].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.972))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.972))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000311936"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'refSeq'", containsString("NM_004985.3"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/5"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.96))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000556131"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("2/3"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.884))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_033360.2"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.972))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"))
                .body("[1].'variant'", containsString("12:g.25398284C>A"))
                .body("[1].'originalVariantQuery'", containsString("12,25398283,25398284,AC,AA"))
                .body("[1].'hgvsg'", containsString("12:g.25398284C>A"))
                .body("[1].'assembly_name'", containsString("GRCh37"))
                .body("[1].'seq_region_name'", containsString("12"))
                .body("[1].'start'", numberMatches(2.5398284E7))
                .body("[1].'end'", numberMatches(2.5398284E7))
                .body("[1].'allele_string'", containsString("C/A"))
                .body("[1].'strand'", numberMatches(1.0))
                .body("[1].'most_severe_consequence'", containsString("missense_variant"))
                .body("[1].'transcript_consequences'.size()", equalTo(4))
                .body("[1].'transcript_consequences'[0].'exon'", containsString("2/6"))
                .body("[1].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000256078"))
                .body("[1].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000256078.4:p.Gly12Val"))
                .body("[1].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[0].'codons'", containsString("gGt/gTt"))
                .body("[1].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000256078"))
                .body("[1].'transcript_consequences'[0].'protein_start'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[0].'protein_end'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[0].'gene_symbol'", containsString("KRAS"))
                .body("[1].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000133703"))
                .body("[1].'transcript_consequences'[0].'amino_acids'", containsString("G/V"))
                .body("[1].'transcript_consequences'[0].'hgnc_id'", containsString("6407"))
                .body("[1].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[1].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.972))
                .body("[1].'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[1].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[1].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("[1].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_033360.2"))
                .body("[1].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[1].'transcript_consequences'[1].'exon'", containsString("2/5"))
                .body("[1].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000311936"))
                .body("[1].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000308495.3:p.Gly12Val"))
                .body("[1].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[1].'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[1].'codons'", containsString("gGt/gTt"))
                .body("[1].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000308495"))
                .body("[1].'transcript_consequences'[1].'protein_start'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[1].'protein_end'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[1].'gene_symbol'", containsString("KRAS"))
                .body("[1].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000133703"))
                .body("[1].'transcript_consequences'[1].'amino_acids'", containsString("G/V"))
                .body("[1].'transcript_consequences'[1].'hgnc_id'", containsString("6407"))
                .body("[1].'transcript_consequences'[1].'polyphen_score'", numberMatches(0.96))
                .body("[1].'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[1].'transcript_consequences'[1].'sift_score'", numberMatches(0.0))
                .body("[1].'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("[1].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_004985.3"))
                .body("[1].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant"))
                .body("[1].'transcript_consequences'[2].'exon'", containsString("2/3"))
                .body("[1].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000556131"))
                .body("[1].'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000451856.1:p.Gly12Val"))
                .body("[1].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[1].'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[2].'codons'", containsString("gGt/gTt"))
                .body("[1].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000451856"))
                .body("[1].'transcript_consequences'[2].'protein_start'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[2].'protein_end'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[2].'gene_symbol'", containsString("KRAS"))
                .body("[1].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000133703"))
                .body("[1].'transcript_consequences'[2].'amino_acids'", containsString("G/V"))
                .body("[1].'transcript_consequences'[2].'hgnc_id'", containsString("6407"))
                .body("[1].'transcript_consequences'[2].'polyphen_score'", numberMatches(0.884))
                .body("[1].'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[1].'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("[1].'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("[1].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'successfully_annotated'", equalTo(true))
                .body("[1].'nucleotide_context'.'license'", nullValue())
                .body("[1].'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("[1].'nucleotide_context'.'annotation'.'seq'", containsString("ACC"))
                .body("[1].'nucleotide_context'.'annotation'.'hgvs'", containsString("12:g.25398284C>A"))
                .body("[1].'annotation_summary'.'variant'", containsString("12:g.25398284C>A"))
                .body("[1].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("12"))
                .body("[1].'annotation_summary'.'genomicLocation'.'start'", numberMatches(2.5398284E7))
                .body("[1].'annotation_summary'.'genomicLocation'.'end'", numberMatches(2.5398284E7))
                .body("[1].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[1].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("A"))
                .body("[1].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[1].'annotation_summary'.'variantType'", containsString("SNP"))
                .body("[1].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[1].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("2/6"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.972))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("2/6"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.972))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000311936"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'refSeq'", containsString("NM_004985.3"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/5"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.96))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000556131"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("2/3"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.884))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_033360.2"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("2/6"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.972))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_96() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/pfam/domain/g%5CK?EMextraParam123=_EM_43_XYZ_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_97() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/a%5C_ga757J00%60_U574UU4UnUUUBUUUCTUIIUGUFUTCAUKC%3FUAC")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_98() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/curious_cases/JVa%5C69bac1")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_99() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/nucleotide_context/OE_%5Eq%5C")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_100() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/canonical-transcript/hgnc/Os%5CXeY4TFkpeUIq?isoformOverrideSource=_EM_9073_XYZ_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_101() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/cancer_hotspots/transcript/_Ebk%5C752XXYZ_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_102() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/canonical-gene/hgnc/XZ%5C0j")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_103() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/MT:ga10%5C71G%3E?" + 
                    "isoformOverrideSource=fok&" + 
                    "token=%7B%0A++++++++++++++++++++%22_EM_25319_XYZ_%22%3A%22kazNE78D%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_25320_XYZ_%22%3A%22%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_25321UXYZ_%22%3A%22KiYO%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_25322_XYZ_%22%3A%221GIkA%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_25323_XYZ_%22%3A%22AOv1MlqVcm6%22%0A++++++++++++++++++++%7D&" + 
                    "fields=&fields=_EM_34363_XYZ_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_104() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/mutation_assessor/i%5CQ")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_105() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/canonical-gene/entrez/zD%5C0:tVm")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_106() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/pdb/header/1%5C87")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_107() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript/_EM_10526_XY%5C_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_108() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/genomic/f%5C?" + 
                    "isoformOverrideSource=SUd&" + 
                    "token=%3DUCnRi&" + 
                    "fields=muelelqmbb_ckpsfw&fields=WWYWUX1vqz4&fields=%3DUCnTk")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("text/html");
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_109() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/xrefs?" + 
                    "accession=&" + 
                    "EMextraParam123=_EM_3186_XYZ_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("{\"error\":\"ID 'ACCESSION' not found\"}"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_110() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/xrefs?" + 
                    "accession=_EM_63_XYZ_&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("{\"error\":\"ID '_EM_63_XYZ_' not found\"}"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_111() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/xrefs/_EM_80_XYZ_")
                .then()
                .statusCode(400)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("{\"error\":\"ID '_EM_80_XYZ_' not found\"}"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_112() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_1482_XYZ_")
                .contentType("application/json")
                .body(" [] ")
                .post(baseUrlOfSut + "/annotation/summary?" + 
                    "isoformOverrideSource=_EM_1480_XYZ_&" + 
                    "projection=EVOMASTER")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_113() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/summary/9:g.21994291C%3ET?" + 
                    "isoformOverrideSource=_EM_930_XYZ_&" + 
                    "projection=EVOMASTER&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_114() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"313U\", " + 
                    " \"{\\n                    \\\"RSsSY\\\":\\\"rE79lUafI52U\\\"\\n    !      \u001f        ,\\n                    \\\"eZlUaFS6EhNR\\\":\\\"qE79ltagI52g\\\"\\n                    ,\\n                    \\\"Y78bJSeZjqlLRH9\\\":\\\"Vs7rhjrLB\\\"\\n           \" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/dbsnp/?" + 
                    "token=%7B%0A++++++++++++++++++++%22RSr%22%3A%22rE79lUafI52U%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22eZkUaFS6EhN%22%3A%22qE79ltagI52%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22YU8bJSeZjWmLRH9%22%3A%22Vs7rhjrLA%22%0A++++++++++++++++++++%7D&" + 
                    "fields=jmcold&fields=US.nruGU6%40BQVBeh&fields=")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_115() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" {} ")
                .post(baseUrlOfSut + "/ptm/experimental")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_116() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ptm/experimental?ensemblTranscriptId=6R4%5BzhHWSDmYYsOJ")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_117() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"?J_cR\", " + 
                    " \"Tu9CS\", " + 
                    " \"ylMqCEUN9SDSZCy\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-transcript/hgnc?isoformOverrideSource=w0K3")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_118() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/pfam/domain/_EM_42_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("PFAM domain not found: _EM_42_XYZ_"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_119() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/cancer_hotspots/genomic/_EM_30_XYZ_?EMextraParam123=_EM_31_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_120() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/_EM_10_XYZ_?EMextraParam123=_EM_11_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_121() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=p.i&" + 
                    "limit=1017462204")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_122() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_2866_XYZ_")
                .get(baseUrlOfSut + "/ensembl/canonical-gene/entrez/238?EMextraParam123=_EM_2865_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_123() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_3169_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"604\", " + 
                    " \"_EM_3453_XYZ_\", " + 
                    " \"_EM_3449_XYZ_\", " + 
                    " \"_EM_3450_XYZ_\", " + 
                    " \"_EM_3167_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-gene/entrez")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_124() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/10:oaW5X4003%60%3F574UU4hnsGUACAUCRUUUUKCUUT;ACUCCCA")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_125() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/UU:ca7266a13ij5U%3C6U3UUUUUACAUMLCGIUKC%3FVTCUCUCCUAU")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_126() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/1U:pY771400U_75g7805ifUUUAKITDUUGI=ICHCUCACUD%3FBADU")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_127() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/1U:rY771400W_75g7824ifUUTAJITDUUGI=UCHCUCACUD%3FBADU")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_128() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/1U:mY771400W_74g7824ifUUTAJITDUWGI=VCHCUUAEUD%3FBABU")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_129() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/1U:nY771400W_74g7824ifUUTAJITDUWGI=VCHCUUAEUD%3FBABU")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_130() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/xrefs?accession=ENSG0000%2F198695")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("{\"error\":\"page not found. Please check your uri and refer to our documentation http://grch37.rest.ensembl.org/\"}"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_131() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/genomic/7,140453136,1504531U6,A,U?" + 
                    "isoformOverrideSource=fmo&" + 
                    "token=fon&" + 
                    "fields=annotation_summary&fields=nucleotide_contex")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_132() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/13:ga32893385_3289338:delTATTC?EMextraParam123=42")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_133() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/-3:ga32893378C%3ET")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_134() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_5_XYZ_")
                .get(baseUrlOfSut + "/ensembl/canonical-gene/hgnc/_EM_3_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_135() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_5390_XYZ_\", " + 
                    " \"_EM_5391_XYZ_\", " + 
                    " \"NeT\", " + 
                    " \"izLS{I\", " + 
                    " \"n\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-gene/hgnc?EMextraParam123=_EM_929_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_136() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/17:ga7574003_7574004insGAACATCTCGIAGCGCTCACGCCCAC?EMextraParam123=42")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_137() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_9_XYZ_")
                .get(baseUrlOfSut + "/ensembl/canonical-transcript/hgnc/_EM_6_XYZ_?EMextraParam123=_EM_8_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Ensembl transcript not found: _EM_6_XYZ_"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_138() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"iDFJR:\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/nucleotide_context")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_139() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/pdb/header/_EM_39_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("PDB web service error for id _EM_39_XYZ_: <!DOCTYPE HTML PUBLIC \"-//IETF//DTD HTML 2.0//EN\">\n<html><head>\n<title>404 Not Found</title>\n</head><body>\n<h1>Not Found</h1>\n<p>The requested URL was not found on this server.</p>\n<hr>\n<address>Apache/2.4.52 (Ubuntu) Server at files.rcsb.org Port 80</address>\n</body></html>\n"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_140() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/genomic/=?token=_EM_1477_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_141() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_58_XYZ_")
                .get(baseUrlOfSut + "/curious_cases/_EM_56_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_142() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/pdb/header/%3FU78")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("PDB header not found: ?U78"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_143() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_69_XYZ_")
                .get(baseUrlOfSut + "/my_variant_info/variant/_EM_67_XYZ_?EMextraParam123=_EM_68_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_144() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_71_XYZ_")
                .get(baseUrlOfSut + "/nucleotide_context/2YM")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_145() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/mutation_assessor/_EM_64_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .contentType("application/json")
                .body("'message'", containsString("Mutation Assessor annotation not found for variant: _EM_64_XYZ_"));
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_146() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"y:U\", " + 
                    " \"start\": 738008576, " + 
                    " \"end\": -1048576, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_147() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \",rqIxsW8G6My8a\", " + 
                    " \"start\": 55155679, " + 
                    " \"end\": 55152107, " + 
                    " \"referenceAllele\": \"y4O_ugLgl\", " + 
                    " \"variantAllele\": \"_EM_5114_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=_EM_5004_XYZ_&" + 
                    "fields=annotatiop_summary&fields=tvTWxlI&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_148() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_2625_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"_EM_7303_XYZ_\", " + 
                    " \"start\": 34, " + 
                    " \"end\": 0, " + 
                    " \"referenceAllele\": \"NA\", " + 
                    " \"variantAllele\": \"_EM_2617_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"8Mi3cSpE\", " + 
                    " \"start\": 965, " + 
                    " \"end\": 1286, " + 
                    " \"referenceAllele\": \"_t\", " + 
                    " \"variantAllele\": \"_tI\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_7304_XYZ_\", " + 
                    " \"start\": 0, " + 
                    " \"end\": 865, " + 
                    " \"referenceAllele\": \"_EM_2622_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_2623_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=_EM_2624_XYZ_")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_149() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1339_XYZ_\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 0, " + 
                    " \"referenceAllele\": \"_EM_1340_XYZ_\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_602_XYZ_\", " + 
                    " \"start\": 37880228, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"_EM_603_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_604_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879786, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_605_XYZ_\", " + 
                    " \"start\": 25398412, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1341_XYZ_\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 0, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"_EM_1342_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1343_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_150() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_602_XYZ_\", " + 
                    " \"start\": 37880228, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"_EM_604_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1690_XYZ_\", " + 
                    " \"start\": 0, " + 
                    " \"end\": -1038628526, " + 
                    " \"referenceAllele\": \"_EM_1691_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1692_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_605_XYZ_\", " + 
                    " \"start\": 25398412, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_2436_XYZ_\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"_EM_2437_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(404)
                .assertThat()
                .body(isEmptyOrNullString());
        
        expectationHandler.expect(ems)
            .that(sco, Arrays.asList(200).contains(res_0.extract().statusCode()));
    }


}
