import  org.junit.AfterClass;
import  org.junit.BeforeClass;
import  org.junit.Before;
import  org.junit.Test;
import static org.junit.Assert.*;
import  java.util.Map;
import  java.util.List;
import static org.evomaster.client.java.controller.api.EMTestUtils.*;
import  org.evomaster.client.java.controller.SutHandler;
import  io.restassured.RestAssured;
import static io.restassured.RestAssured.given;
import  io.restassured.response.ValidatableResponse;
import static org.hamcrest.Matchers.*;
import  io.restassured.config.JsonConfig;
import  io.restassured.path.json.config.JsonPathConfig;
import static io.restassured.config.RedirectConfig.redirectConfig;
import static org.evomaster.client.java.controller.contentMatchers.NumberMatcher.*;
import static org.evomaster.client.java.controller.contentMatchers.StringMatcher.*;
import static org.evomaster.client.java.controller.contentMatchers.SubStringMatcher.*;
import static org.evomaster.client.java.controller.expect.ExpectationHandler.expectationHandler;
import  org.evomaster.client.java.controller.expect.ExpectationHandler;
import  io.restassured.path.json.JsonPath;
import  java.util.Arrays;




/**
 * This file was automatically generated by EvoMaster on 2024-05-03T15:29:13.631+02:00[Europe/Paris]
 * <br>
 * The generated test suite contains 90 tests
 * <br>
 * Covered targets: 5634
 * <br>
 * Used time: 10h 3m 56s
 * <br>
 * Needed budget for current results: 91%
 * <br>
 * This file contains test cases that represent successful calls.
 */
public class EvoMaster_successes_Test {

    
    private static final SutHandler controller = new em.embedded.org.cbioportal.genome_nexus.EmbeddedEvoMasterController();
    private static String baseUrlOfSut;
    /** [ems] - expectations master switch - is the variable that activates/deactivates expectations individual test cases
    * by default, expectations are turned off. The variable needs to be set to [true] to enable expectations
    */
    private static boolean ems = false;
    /**
    * sco - supported code oracle - checking that the response status code is among those supported according to the schema
    */
    private static boolean sco = false;
    /**
    * rso - response structure oracle - checking that the response objects match the responses defined in the schema
    */
    private static boolean rso = false;
    
    
    @BeforeClass
    public static void initClass() {
        controller.setupForGeneratedTest();
        baseUrlOfSut = controller.startSut();
        controller.registerOrExecuteInitSqlCommandsIfNeeded();
        assertNotNull(baseUrlOfSut);
        RestAssured.enableLoggingOfRequestAndResponseIfValidationFails();
        RestAssured.useRelaxedHTTPSValidation();
        RestAssured.urlEncodingEnabled = false;
        RestAssured.config = RestAssured.config()
            .jsonConfig(JsonConfig.jsonConfig().numberReturnType(JsonPathConfig.NumberReturnType.DOUBLE))
            .redirect(redirectConfig().followRedirects(false));
    }
    
    
    @AfterClass
    public static void tearDown() {
        controller.stopSut();
    }
    
    
    @Before
    public void initTest() {
        controller.resetStateOfSUT();
    }
    
    
    
    
    @Test(timeout = 60000)
    public void test_0() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("*/*")
                .get(baseUrlOfSut + "/v2/api-docs")
                .then()
                .statusCode(200);
        
        expectationHandler.expect(ems)
            /*
             Note: No supported codes appear to be defined. https://swagger.io/docs/specification/describing-responses/.
             This is somewhat unexpected, so the code below is likely to lead to a failed expectation
            */
            .that(sco, Arrays.asList().contains(res_0.extract().statusCode()));
    }
    
    
    @Test(timeout = 60000)
    public void test_1() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation?" + 
                    "hugoGeneSymbol=BRCA2&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_2() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/cancer_hotspots/transcript/ENST00000288602?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_3() throws Exception {
        
        given().accept("*/*")
                .header("x-EMextraHeader123", "_EM_47_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript/_EM_45_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .body(isEmptyOrNullString());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_4() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_169_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/pfam/domain")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_5() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/ensembl/transcript?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_6() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_66_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_62_XYZ_&" + 
                    "EMextraParam123=_EM_65_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_7() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_211_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/signal/mutation")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_8() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_93_XYZ_")
                .get(baseUrlOfSut + "/version")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'genomeNexus'.'server'.'version'", containsString("1.0.2"))
                .body("'genomeNexus'.'server'.'static'", equalTo(true))
                .body("'genomeNexus'.'database'.'version'", containsString("3.6.2"))
                .body("'genomeNexus'.'database'.'static'", equalTo(true))
                .body("'vep'.'server'.'version'", containsString("NA"))
                .body("'vep'.'server'.'static'", equalTo(false))
                .body("'vep'.'cache'.'version'", containsString("NA"))
                .body("'vep'.'cache'.'static'", equalTo(false))
                .body("'vep'.'comment'", containsString("VEP annotations are currently externally sourced from ENSEMBL. Results are subject to change without notice."));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_9() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_410_XYZ_&" + 
                    "proteinId=_EM_411_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_10() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/ptm/experimental?" + 
                    "ensemblTranscriptId=ENST00000371953&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_11() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "proteinId=_EM_1533_XYZ_&" + 
                    "EMextraParam123=_EM_1535_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_12() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_2521_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_2518_XYZ_&" + 
                    "hugoSymbol=_EM_2519_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_13() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?keyword=")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_14() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_834_XYZ_")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "geneId=_EM_831_XYZ_&" + 
                    "proteinId=_EM_832_XYZ_&" + 
                    "hugoSymbol=kGaF_m1Zs2")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_15() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=&" + 
                    "limit=804")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_16() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/transcript?" + 
                    "proteinId=_EM_1494_XYZ_&" + 
                    "hugoSymbol=_EM_1495_XYZ_&" + 
                    "EMextraParam123=_EM_1496_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_17() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"transcriptIds\": [ " + 
                    " \"ENST00000420316\", " + 
                    " \"ENST00000646891\", " + 
                    " \"ENST00000371953\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ptm/experimental?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_18() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_160_XYZ_")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_148_XYZ_\", " + 
                    " \"_EM_149_XYZ_\", " + 
                    " \"_EM_150_XYZ_\", " + 
                    " \"_EM_151_XYZ_\", " + 
                    " \"_EM_152_XYZ_\" " + 
                    " ], " + 
                    " \"transcriptIds\": [ " + 
                    " \"_EM_157_XYZ_\", " + 
                    " \"_EM_158_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_19() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" { " + 
                    " \"proteinIds\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_20() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_4795_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_21() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" { " + 
                    " \"hugoSymbols\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_22() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" { " + 
                    " \"geneIds\": [ " + 
                    " \"_EM_2941_XYZ_\", " + 
                    " \"_EM_2942_XYZ_\", " + 
                    " \"_EM_2943_XYZ_\", " + 
                    " \"KVAq\", " + 
                    " \"_EM_2944_XYZ_\" " + 
                    " ], " + 
                    " \"hugoSymbols\": [ " + 
                    " \"QgTO27BDmWH40p\", " + 
                    " \"eTyZC5Cttkvbq\", " + 
                    " \"\" " + 
                    " ], " + 
                    " \"transcriptIds\": [] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_23() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" { " + 
                    " \"geneIds\": [], " + 
                    " \"hugoSymbols\": [ " + 
                    " \"_EM_456_XYZ_\", " + 
                    " \"_EM_457_XYZ_\", " + 
                    " \"_EM_458_XYZ_\", " + 
                    " \"_EM_459_XYZ_\", " + 
                    " \"_EM_460_XYZ_\" " + 
                    " ], " + 
                    " \"proteinIds\": [ " + 
                    " \"_EM_461_XYZ_\", " + 
                    " \"_EM_462_XYZ_\", " + 
                    " \"_EM_463_XYZ_\", " + 
                    " \"_EM_464_XYZ_\", " + 
                    " \"_EM_465_XYZ_\" " + 
                    " ] " + 
                    " } ")
                .post(baseUrlOfSut + "/ensembl/transcript")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_24() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_91_XYZ_")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=_EM_89_XYZ_&" + 
                    "EMextraParam123=_EM_90_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_25() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"ENST00000288602\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/transcript?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_26() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_1137_XYZ_")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=_EM_1135_XYZ_&" + 
                    "limit=419&" + 
                    "EMextraParam123=_EM_1136_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_27() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"277\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-gene/entrez?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_28() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_2314_XYZ_")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=c.&" + 
                    "limit=17302317&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'queryType'", containsString("HGVSG"))
                .body("[0].'results'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_29() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=g.&" + 
                    "limit=-194272024&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'queryType'", containsString("HGVSG"))
                .body("[0].'results'.size()", equalTo(1))
                .body("[0].'results'[0]", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_30() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/search?" + 
                    "keyword=g.&" + 
                    "limit=17297205&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'queryType'", containsString("HGVSG"))
                .body("[0].'results'.size()", equalTo(1))
                .body("[0].'results'[0]", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_31() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"transcriptId\": \"ENST00000397752\", " + 
                    " \"start\": 1009, " + 
                    " \"end\": 1009, " + 
                    " \"mutationType\": \"Splice_Region\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/proteinLocations?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'proteinLocation'.'transcriptId'", containsString("ENST00000397752"))
                .body("[0].'proteinLocation'.'start'", numberMatches(1009.0))
                .body("[0].'proteinLocation'.'end'", numberMatches(1009.0))
                .body("[0].'proteinLocation'.'mutationType'", containsString("Splice_Region"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_32() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_255_XYZ_\", " + 
                    " \"_EM_256_XYZ_\", " + 
                    " \"wXcAdH6Ecly0i26n\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-gene/hgnc?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_33() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_102_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_100_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/ensembl/canonical-transcript/hgnc?" + 
                    "isoformOverrideSource=uniprot&" + 
                    "EMextraParam123=_EM_101_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_34() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?" + 
                    "keyword=_EM_566_XYZ_&" + 
                    "EMextraParam123=_EM_567_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_35() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?" + 
                    "keyword=_EM_83_XYZ_&" + 
                    "limit=611")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_36() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/signal/search?" + 
                    "keyword=_EM_3389_XYZ_&" + 
                    "limit=-4193448")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_37() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_1650_XYZ_\", " + 
                    " \"_EM_1651_XYZ_\", " + 
                    " \"_EM_1652_XYZ_\", " + 
                    " \"\", " + 
                    " \"_EM_1653_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/pdb/header?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_38() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/ensembl/xrefs?accession=ENSG00000198685")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(7))
                .body("[0].'display_id'", containsString("OTTHUMG00000159688"))
                .body("[0].'primary_id'", containsString("OTTHUMG00000159688"))
                .body("[0].'version'", containsString("1"))
                .body("[0].'dbname'", containsString("OTTG"))
                .body("[0].'synonyms'.size()", equalTo(0))
                .body("[0].'info_text'", containsString(""))
                .body("[0].'info_types'", containsString("NONE"))
                .body("[0].'db_display_name'", containsString("Havana gene"))
                .body("[1].'display_id'", containsString("Hs.194283"))
                .body("[1].'primary_id'", containsString("Hs.194283"))
                .body("[1].'version'", containsString("0"))
                .body("[1].'description'", containsString("Chromosome 3 open reading frame 27"))
                .body("[1].'dbname'", containsString("UniGene"))
                .body("[1].'synonyms'.size()", equalTo(0))
                .body("[1].'info_text'", containsString(""))
                .body("[1].'info_types'", containsString("SEQUENCE_MATCH"))
                .body("[1].'db_display_name'", containsString("UniGene"))
                .body("[2].'display_id'", containsString("C3orf27"))
                .body("[2].'primary_id'", containsString("17099"))
                .body("[2].'version'", containsString("0"))
                .body("[2].'description'", containsString("chromosome 3 open reading frame 27"))
                .body("[2].'dbname'", containsString("HGNC"))
                .body("[2].'synonyms'.size()", equalTo(1))
                .body("[2].'synonyms'", hasItems("GR6"))
                .body("[2].'info_text'", containsString("Generated via ensembl_manual"))
                .body("[2].'info_types'", containsString("DIRECT"))
                .body("[2].'db_display_name'", containsString("HGNC Symbol"))
                ; // Skipping assertions on the remaining 4 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(4)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(5)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(6)).keySet().containsAll(Arrays.asList("db_display_name", "dbname", "description", "display_id", "primary_id", "version")));
    }
    
    
    @Test(timeout = 60000)
    public void test_39() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/10:g.89624230_89624231delAC?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_40() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/17:g.7574003_7574004insGAACATCTCGAAGCGCTCACGCCCAC?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_41() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/signal/mutation/hgvs/13:g.32893378C%3ET?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_42() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/cancer_hotspots/hgvs/undefined")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_43() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"zrj\", " + 
                    " \"zrj\", " + 
                    " \"_EM_31939_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/my_variant_info/variant")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_44() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_41_XYZ_")
                .get(baseUrlOfSut + "/cancer_hotspots/hgvs/_EM_39_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_45() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/pdb/header/1yy9?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'pdbId'", containsString("1yy9"))
                .body("'title'", containsString("structure of the extracellular domain of the epidermal growth factor receptor in complex with the fab fragment of cetuximab/erbitux/imc-c225"))
                .body("'compound'.'1'.'chain'.size()", equalTo(1))
                .body("'compound'.'1'.'chain'", hasItems("a"))
                .body("'compound'.'1'.'fragment'", containsString("unp residues 25-642"))
                .body("'compound'.'1'.'synonym'", containsString("receptor tyrosine-protein kinase erbb-1"))
                .body("'compound'.'1'.'engineered'", containsString("yes"))
                .body("'compound'.'1'.'molecule'", containsString("epidermal growth factor receptor"))
                .body("'compound'.'1'.'mol_id'", containsString("1"))
                .body("'compound'.'1'.'ec'", containsString("2.7.10.1"))
                .body("'compound'.'2'.'chain'.size()", equalTo(1))
                .body("'compound'.'2'.'chain'", hasItems("c"))
                .body("'compound'.'2'.'engineered'", containsString("yes"))
                .body("'compound'.'2'.'molecule'", containsString("cetuximab fab light chain"))
                .body("'compound'.'2'.'mol_id'", containsString("2"))
                .body("'compound'.'3'.'chain'.size()", equalTo(1))
                .body("'compound'.'3'.'chain'", hasItems("d"))
                .body("'compound'.'3'.'molecule'", containsString("cetuximab fab heavy chain"))
                .body("'compound'.'3'.'mol_id'", containsString("3"))
                .body("'source'.'1'.'gene'.size()", equalTo(2))
                .body("'source'.'1'.'gene'", hasItems("egfr", "erbb1"))
                .body("'source'.'1'.'organism_scientific'", containsString("homo sapiens"))
                .body("'source'.'1'.'organism_common'", containsString("human"))
                .body("'source'.'1'.'expression_system_vector_type'", containsString("baculovirus"))
                .body("'source'.'1'.'organism_taxid'", containsString("9606"))
                .body("'source'.'1'.'expression_system_common'", containsString("fall armyworm"))
                .body("'source'.'1'.'expression_system_atcc_number'", containsString("sf9"))
                .body("'source'.'1'.'expression_system_taxid'", containsString("7108"))
                .body("'source'.'1'.'expression_system'", containsString("spodoptera frugiperda"))
                .body("'source'.'1'.'mol_id'", containsString("1"))
                .body("'source'.'2'.'strain'", containsString(","))
                .body("'source'.'2'.'expression_system_plasmid'", containsString("pdhl2"))
                .body("'source'.'2'.'organism_common'", containsString("house mouse, human"))
                .body("'source'.'2'.'expression_system_vector_type'", containsString("plasmid"))
                .body("'source'.'2'.'expression_system_cell'", containsString("mouse myeloma cell line"))
                .body("'source'.'2'.'expression_system'", containsString("mus musculus"))
                .body("'source'.'2'.'organism_scientific'", containsString("mus musculus, homo sapiens"))
                .body("'source'.'2'.'organism_taxid'", containsString("10090,9606"))
                .body("'source'.'2'.'expression_system_common'", containsString("house mouse"))
                .body("'source'.'2'.'expression_system_strain'", containsString("sp2/0-ag15"))
                .body("'source'.'2'.'expression_system_atcc_number'", containsString("crl-1581"))
                .body("'source'.'2'.'expression_system_taxid'", containsString("10090"))
                .body("'source'.'2'.'mol_id'", containsString("2"))
                .body("'source'.'3'.'strain'", containsString(","))
                .body("'source'.'3'.'organism_scientific'", containsString("mus musculus, homo sapiens"))
                .body("'source'.'3'.'organism_common'", containsString("house mouse, human"))
                .body("'source'.'3'.'expression_system_vector_type'", containsString("plasmid"))
                .body("'source'.'3'.'organism_taxid'", containsString("10090,9606"))
                .body("'source'.'3'.'expression_system_common'", containsString("house mouse"))
                .body("'source'.'3'.'expression_system_strain'", containsString("sp2/0-ag15"))
                .body("'source'.'3'.'expression_system_atcc_number'", containsString("crl-1581"))
                .body("'source'.'3'.'expression_system_cell'", containsString("mouse myeloma cell line"))
                .body("'source'.'3'.'expression_system_taxid'", containsString("10090"))
                .body("'source'.'3'.'expression_system'", containsString("mus musculus"))
                .body("'source'.'3'.'mol_id'", containsString("3"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_46() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_194_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_188_XYZ_\", " + 
                    " \"_EM_189_XYZ_\", " + 
                    " \"_EM_190_XYZ_\", " + 
                    " \"_EM_191_XYZ_\", " + 
                    " \"_EM_192_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/mutation_assessor")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_47() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_135_XYZ_\", " + 
                    " \"_EM_136_XYZ_\", " + 
                    " \"_EM_137_XYZ_\", " + 
                    " \"CI\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/hgvs")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(4))
                .body("[0].'variant'", containsString("_EM_135_XYZ_"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'variant'", containsString("_EM_136_XYZ_"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'variant'", containsString("_EM_137_XYZ_"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("genomicLocation", "hotspots", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_48() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"1a37\", " + 
                    " \"1a4o\", " + 
                    " \"3njp\", " + 
                    " \"1yy9\", " + 
                    " \"INVALID\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/pdb/header?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(4))
                .body("[0].'pdbId'", containsString("1a37"))
                .body("[0].'title'", containsString("14-3-3 protein zeta bound to ps-raf259 peptide"))
                .body("[0].'compound'.'1'.'other_details'", containsString("complexed with phosphoserine-containing peptide derived from raf"))
                .body("[0].'compound'.'1'.'chain'.size()", equalTo(2))
                .body("[0].'compound'.'1'.'chain'", hasItems("a", "b"))
                .body("[0].'compound'.'1'.'engineered'", containsString("yes"))
                .body("[0].'compound'.'1'.'molecule'", containsString("14-3-3 protein zeta"))
                .body("[0].'compound'.'1'.'mol_id'", containsString("1"))
                .body("[0].'compound'.'2'.'chain'.size()", equalTo(2))
                .body("[0].'compound'.'2'.'chain'", hasItems("p", "q"))
                .body("[0].'compound'.'2'.'molecule'", containsString("ps-raf259 peptide lsqrqrst(sep)tpnvhm"))
                .body("[0].'compound'.'2'.'mol_id'", containsString("2"))
                .body("[0].'source'.'1'.'organism_scientific'", containsString("bos taurus"))
                .body("[0].'source'.'1'.'organism_common'", containsString("cattle"))
                .body("[0].'source'.'1'.'organism_taxid'", containsString("9913"))
                .body("[0].'source'.'1'.'expression_system_taxid'", containsString("562"))
                .body("[0].'source'.'1'.'expression_system'", containsString("escherichia coli"))
                .body("[0].'source'.'1'.'mol_id'", containsString("1"))
                .body("[1].'pdbId'", containsString("1a4o"))
                .body("[1].'title'", containsString("14-3-3 protein zeta isoform"))
                .body("[1].'compound'.'1'.'chain'.size()", equalTo(4))
                .body("[1].'compound'.'1'.'chain'", hasItems("a", "b", "c", "d"))
                .body("[1].'compound'.'1'.'molecule'", containsString("14-3-3 protein zeta"))
                .body("[1].'compound'.'1'.'mol_id'", containsString("1"))
                .body("[1].'source'.'1'.'organism_scientific'", containsString("bos taurus"))
                .body("[1].'source'.'1'.'organism_common'", containsString("cattle"))
                .body("[1].'source'.'1'.'organism_taxid'", containsString("9913"))
                .body("[1].'source'.'1'.'expression_system'", containsString("escherichia coli"))
                .body("[1].'source'.'1'.'mol_id'", containsString("1"))
                .body("[2].'pdbId'", containsString("3njp"))
                .body("[2].'title'", containsString("the extracellular and transmembrane domain interfaces in epidermal growth factor receptor signaling"))
                .body("[2].'compound'.'1'.'chain'.size()", equalTo(2))
                .body("[2].'compound'.'1'.'chain'", hasItems("a", "b"))
                .body("[2].'compound'.'1'.'synonym'", containsString("receptor tyrosine-protein kinase erbb-1, proto-oncogene c- erbb-1"))
                .body("[2].'compound'.'1'.'engineered'", containsString("yes"))
                .body("[2].'compound'.'1'.'molecule'", containsString("epidermal growth factor receptor"))
                .body("[2].'compound'.'1'.'mol_id'", containsString("1"))
                .body("[2].'compound'.'1'.'ec'", containsString("2.7.10.1"))
                .body("[2].'compound'.'2'.'chain'.size()", equalTo(2))
                .body("[2].'compound'.'2'.'chain'", hasItems("c", "d"))
                .body("[2].'compound'.'2'.'molecule'", containsString("epidermal growth factor"))
                .body("[2].'compound'.'2'.'mol_id'", containsString("2"))
                .body("[2].'source'.'1'.'expression_system_plasmid'", containsString("pcdna3.1/zeo(+)"))
                .body("[2].'source'.'1'.'gene'.size()", equalTo(2))
                .body("[2].'source'.'1'.'gene'", hasItems("egfr", "erbb1"))
                .body("[2].'source'.'1'.'organism_scientific'", containsString("homo sapiens"))
                .body("[2].'source'.'1'.'organism_common'", containsString("human"))
                .body("[2].'source'.'1'.'expression_system_vector_type'", containsString("plasmid"))
                .body("[2].'source'.'1'.'organism_taxid'", containsString("9606"))
                .body("[2].'source'.'1'.'expression_system_common'", containsString("chinese hamster"))
                .body("[2].'source'.'1'.'expression_system_strain'", containsString("lec8"))
                .body("[2].'source'.'1'.'expression_system_organ'", containsString("ovary"))
                .body("[2].'source'.'1'.'expression_system_taxid'", containsString("10029"))
                .body("[2].'source'.'1'.'expression_system'", containsString("cricetulus griseus"))
                .body("[2].'source'.'1'.'mol_id'", containsString("1"))
                .body("[2].'source'.'2'.'gene'.size()", equalTo(1))
                .body("[2].'source'.'2'.'gene'", hasItems("egf"))
                .body("[2].'source'.'2'.'organism_scientific'", containsString("homo sapiens"))
                .body("[2].'source'.'2'.'organism_common'", containsString("human"))
                .body("[2].'source'.'2'.'organism_taxid'", containsString("9606"))
                .body("[2].'source'.'2'.'expression_system_common'", containsString("chinese hamster"))
                .body("[2].'source'.'2'.'expression_system_taxid'", containsString("10029"))
                .body("[2].'source'.'2'.'expression_system'", containsString("cricetulus griseus"))
                .body("[2].'source'.'2'.'mol_id'", containsString("2"))
                ; // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_49() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/my_variant_info/variant/17:g.41276045_41276046del?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("chr17:g.41276045_41276046del"))
                .body("'query'", containsString("chr17:g.41276045_41276046del"))
                .body("'hgvs'", containsString("chr17:g.41276045_41276046del"))
                .body("'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("'vcf'.'alt'", containsString("A"))
                .body("'vcf'.'position'", containsString("41276044"))
                .body("'vcf'.'ref'", containsString("ACT"))
                .body("'gnomadExome'.'alleleCount'.'ac'", numberMatches(57.0))
                .body("'gnomadExome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_amr'", numberMatches(1.0))
                .body("'gnomadExome'.'alleleCount'.'ac_asj'", numberMatches(41.0))
                .body("'gnomadExome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_nfe'", numberMatches(11.0))
                .body("'gnomadExome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleCount'.'ac_sas'", numberMatches(4.0))
                .body("'gnomadExome'.'alleleNumber'.'an'", numberMatches(251050.0))
                .body("'gnomadExome'.'alleleNumber'.'an_afr'", numberMatches(16250.0))
                .body("'gnomadExome'.'alleleNumber'.'an_amr'", numberMatches(34592.0))
                .body("'gnomadExome'.'alleleNumber'.'an_asj'", numberMatches(10078.0))
                .body("'gnomadExome'.'alleleNumber'.'an_eas'", numberMatches(18394.0))
                .body("'gnomadExome'.'alleleNumber'.'an_fin'", numberMatches(21644.0))
                .body("'gnomadExome'.'alleleNumber'.'an_nfe'", numberMatches(113354.0))
                .body("'gnomadExome'.'alleleNumber'.'an_oth'", numberMatches(6130.0))
                .body("'gnomadExome'.'alleleNumber'.'an_sas'", numberMatches(30608.0))
                .body("'gnomadExome'.'alleleFrequency'.'af'", numberMatches(2.27046E-4))
                .body("'gnomadExome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_amr'", numberMatches(2.89084E-5))
                .body("'gnomadExome'.'alleleFrequency'.'af_asj'", numberMatches(0.00406827))
                .body("'gnomadExome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_nfe'", numberMatches(9.70411E-5))
                .body("'gnomadExome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("'gnomadExome'.'alleleFrequency'.'af_sas'", numberMatches(1.30685E-4))
                .body("'gnomadExome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("'gnomadExome'.'homozygotes'.'hom_sas'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac'", numberMatches(1.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_amr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_asj'", numberMatches(1.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_nfe'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleCount'.'ac_sas'", nullValue())
                .body("'gnomadGenome'.'alleleNumber'.'an'", numberMatches(31392.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_afr'", numberMatches(8712.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_amr'", numberMatches(848.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_asj'", numberMatches(290.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_eas'", numberMatches(1560.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_fin'", numberMatches(3468.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_nfe'", numberMatches(15426.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_oth'", numberMatches(1088.0))
                .body("'gnomadGenome'.'alleleNumber'.'an_sas'", nullValue())
                .body("'gnomadGenome'.'alleleFrequency'.'af'", numberMatches(3.18552E-5))
                .body("'gnomadGenome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_amr'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_asj'", numberMatches(0.00344828))
                .body("'gnomadGenome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_nfe'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("'gnomadGenome'.'alleleFrequency'.'af_sas'", nullValue())
                .body("'gnomadGenome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("'gnomadGenome'.'homozygotes'.'hom_sas'", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_50() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "_EM_113_XYZ_")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_107_XYZ_\", " + 
                    " \"COSM16\", " + 
                    " \"_EM_109_XYZ_\", " + 
                    " \"_EM_110_XYZ_\", " + 
                    " \"_EM_111_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/dbsnp/?" + 
                    "isoformOverrideSource=_EM_103_XYZ_&" + 
                    "token=JJK&" + 
                    "fields=_EM_104_XYZ_&fields=kWTNXUKCxrgDKs&fields=_EM_105_XYZ_&fields=_EM_106_XYZ_&" + 
                    "EMextraParam123=_EM_112_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(5))
                .body("[0].'variant'", containsString("_EM_107_XYZ_"))
                .body("[0].'originalVariantQuery'", containsString("_EM_107_XYZ_"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[1].'variant'", containsString("COSM16"))
                .body("[1].'originalVariantQuery'", containsString("COSM16"))
                .body("[1].'successfully_annotated'", equalTo(false))
                .body("[2].'variant'", containsString("_EM_109_XYZ_"))
                .body("[2].'originalVariantQuery'", containsString("_EM_109_XYZ_"))
                .body("[2].'successfully_annotated'", equalTo(false))
                ; // Skipping assertions on the remaining 2 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(4)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_51() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"n0hT87g\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation?" + 
                    "token=&" + 
                    "fields=n6hU%3A3g&fields=ptms&fields=O")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("n0hT87g"))
                .body("[0].'originalVariantQuery'", containsString("n0hT87g"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_52() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"cEcd6WUXXTBzXYgD\", " + 
                    " \"hVH\", " + 
                    " \"XRUN\", " + 
                    " \"Jx\", " + 
                    " \"NFxtI\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation?" + 
                    "isoformOverrideSource=&" + 
                    "token=%7B%0A++++++++++++++++++++%22TcUuKz4Wb%22%3A%22TocUvMz4Wb%2F%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22YVUVufUpNNMU%22%3A%22%7B%0A+++%1F++++++++++++++++%5C%22TcUvKz4Wb%2F%5C%22%3A%5C%22TpcUvMz4Wb%2F%5C%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%5C%22YVUVvfUpNNMUn%5C%22%3A%5C%22wrIUxWrirxvpNQ%5C%22%0A++++++++++++++++++++%2C%0A++%1F+++++++++++++++++%5C%22_EM_80583_XYZ_%5C%22%3A%5C%22euyp%5C%22%0A++++++++++%22%0A++++++++++++++++++++%2C%0A++++++++++++++++++++%22_EM_80583_XYZ_%22%3A%22euyp%22%0A++++++++++++++++++++%7D&" + 
                    "fields=clinvar")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(5))
                .body("[0].'variant'", containsString("cEcd6WUXXTBzXYgD"))
                .body("[0].'originalVariantQuery'", containsString("cEcd6WUXXTBzXYgD"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[0].'clinvar'.'annotation'", nullValue())
                .body("[1].'variant'", containsString("hVH"))
                .body("[1].'originalVariantQuery'", containsString("hVH"))
                .body("[1].'successfully_annotated'", equalTo(false))
                .body("[1].'clinvar'.'annotation'", nullValue())
                .body("[2].'variant'", containsString("XRUN"))
                .body("[2].'originalVariantQuery'", containsString("XRUN"))
                .body("[2].'successfully_annotated'", equalTo(false))
                .body("[2].'clinvar'.'annotation'", nullValue())
                ; // Skipping assertions on the remaining 2 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(4)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_53() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"osU3JZqbS4opc\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation?" + 
                    "token=dCFSUvU35zUO&" + 
                    "fields=signal&fields=V&fields=YjN")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("osU3JZqbS4opc"))
                .body("[0].'originalVariantQuery'", containsString("osU3JZqbS4opc"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_54() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"_EM_329_XYZ_\", " + 
                    " \"_EM_330_XYZ_\", " + 
                    " \"_EM_331_XYZ_\", " + 
                    " \"\", " + 
                    " \"_EM_332_XYZ_\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?isoformOverrideSource=XqQyM5pFtYP_x")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(5))
                .body("[0].'variant'", containsString("_EM_329_XYZ_"))
                .body("[0].'genomicLocation'.'chromosome'", nullValue())
                .body("[0].'genomicLocation'.'start'", nullValue())
                .body("[0].'genomicLocation'.'end'", nullValue())
                .body("[0].'genomicLocation'.'referenceAllele'", nullValue())
                .body("[0].'genomicLocation'.'variantAllele'", nullValue())
                .body("[0].'transcriptConsequences'.size()", equalTo(0))
                .body("[0].'transcriptConsequenceSummaries'.size()", equalTo(0))
                .body("[1].'variant'", containsString("_EM_330_XYZ_"))
                .body("[1].'genomicLocation'.'chromosome'", nullValue())
                .body("[1].'genomicLocation'.'start'", nullValue())
                .body("[1].'genomicLocation'.'end'", nullValue())
                .body("[1].'genomicLocation'.'referenceAllele'", nullValue())
                .body("[1].'genomicLocation'.'variantAllele'", nullValue())
                .body("[1].'transcriptConsequences'.size()", equalTo(0))
                .body("[1].'transcriptConsequenceSummaries'.size()", equalTo(0))
                .body("[2].'variant'", containsString("_EM_331_XYZ_"))
                .body("[2].'genomicLocation'.'chromosome'", nullValue())
                .body("[2].'genomicLocation'.'start'", nullValue())
                .body("[2].'genomicLocation'.'end'", nullValue())
                .body("[2].'genomicLocation'.'referenceAllele'", nullValue())
                .body("[2].'genomicLocation'.'variantAllele'", nullValue())
                .body("[2].'transcriptConsequences'.size()", equalTo(0))
                .body("[2].'transcriptConsequenceSummaries'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 2 elements. This limit of 3 elements can be increased in the configurations
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(2)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(3)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(4)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_55() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"9OBGei7wh2mU\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation?" + 
                    "isoformOverrideSource=&" + 
                    "token=_EM_64391_XYZ_&" + 
                    "fields=oncokb&fields=MCJa9YZUQ9H")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("9OBGei7wh2mU"))
                .body("[0].'originalVariantQuery'", containsString("9OBGei7wh2mU"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_56() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"12:g.43130875C\u003eG\", " + 
                    " \"12:g.43130996C\u003eG\", " + 
                    " \"INVALID\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(3))
                .body("[0].'variant'", containsString("12:g.43130875C>G"))
                .body("[0].'originalVariantQuery'", containsString("12:g.43130875C>G"))
                .body("[0].'hgvsg'", containsString("12:g.43130875C>G"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("12"))
                .body("[0].'start'", numberMatches(4.3130875E7))
                .body("[0].'end'", numberMatches(4.3130875E7))
                .body("[0].'allele_string'", containsString("C/G"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("intergenic_variant"))
                .body("[0].'intergenic_consequences'.size()", equalTo(1))
                .body("[0].'intergenic_consequences'[0].'impact'", containsString("MODIFIER"))
                .body("[0].'intergenic_consequences'[0].'variantAllele'", containsString("G"))
                .body("[0].'intergenic_consequences'[0].'consequenceTerms'.size()", equalTo(1))
                .body("[0].'intergenic_consequences'[0].'consequenceTerms'", hasItems("intergenic_variant"))
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[1].'variant'", containsString("12:g.43130996C>G"))
                .body("[1].'originalVariantQuery'", containsString("12:g.43130996C>G"))
                .body("[1].'hgvsg'", containsString("12:g.43130996C>G"))
                .body("[1].'assembly_name'", containsString("GRCh37"))
                .body("[1].'seq_region_name'", containsString("12"))
                .body("[1].'start'", numberMatches(4.3130996E7))
                .body("[1].'end'", numberMatches(4.3130996E7))
                .body("[1].'allele_string'", containsString("C/G"))
                .body("[1].'strand'", numberMatches(1.0))
                .body("[1].'most_severe_consequence'", containsString("intergenic_variant"))
                .body("[1].'intergenic_consequences'.size()", equalTo(1))
                .body("[1].'intergenic_consequences'[0].'impact'", containsString("MODIFIER"))
                .body("[1].'intergenic_consequences'[0].'variantAllele'", containsString("G"))
                .body("[1].'intergenic_consequences'[0].'consequenceTerms'.size()", equalTo(1))
                .body("[1].'intergenic_consequences'[0].'consequenceTerms'", hasItems("intergenic_variant"))
                .body("[1].'successfully_annotated'", equalTo(true))
                .body("[2].'variant'", containsString("INVALID"))
                .body("[2].'originalVariantQuery'", containsString("INVALID"))
                .body("[2].'successfully_annotated'", equalTo(false));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_57() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"7:g.140453136A\u003eT\", " + 
                    " \"17:g.41276045_41276046del\", " + 
                    " \"rs12190874\", " + 
                    " \"INVALID\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/my_variant_info/variant?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(3))
                .body("[0].'variant'", containsString("chr7:g.140453136A>T"))
                .body("[0].'query'", containsString("chr7:g.140453136A>T"))
                .body("[0].'hgvs'", containsString("7:g.140453136A>T"))
                .body("[0].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[0].'vcf'.'alt'", containsString("T"))
                .body("[0].'vcf'.'position'", containsString("140453136"))
                .body("[0].'vcf'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[0].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[0].'dbsnp'.'alleles'[0].'allele'", containsString("A"))
                .body("[0].'dbsnp'.'alleles'[1].'allele'", containsString("T"))
                .body("[0].'dbsnp'.'alt'", containsString("T"))
                .body("[0].'dbsnp'.'chrom'", containsString("7"))
                .body("[0].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[0].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[0].'dbsnp'.'flags'[0]", nullValue())
                .body("[0].'dbsnp'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'dbsnp'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'dbsnp'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'rsid'", containsString("rs113488022"))
                .body("[0].'dbsnp'.'vartype'", containsString("snv"))
                .body("[0].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[0].'cosmic'.'alt'", containsString("A"))
                .body("[0].'cosmic'.'chrom'", containsString("7"))
                .body("[0].'cosmic'.'cosmicId'", containsString("COSM476"))
                .body("[0].'cosmic'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'cosmic'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'cosmic'.'mutFreq'", numberMatches(2.83))
                .body("[0].'cosmic'.'mutNt'", containsString("T>A"))
                .body("[0].'cosmic'.'ref'", containsString("T"))
                .body("[0].'cosmic'.'tumorSite'", containsString("biliary_tract"))
                .body("[0].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[0].'clinVar'.'alleleId'", numberMatches(29000.0))
                .body("[0].'clinVar'.'alt'", containsString("T"))
                .body("[0].'clinVar'.'chrom'", containsString("7"))
                .body("[0].'clinVar'.'cytogenic'", containsString("7q34"))
                .body("[0].'clinVar'.'gene'.'symbol'", containsString("BRAF"))
                .body("[0].'clinVar'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'clinVar'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'clinVar'.'hg38'.'start'", containsString("140753336"))
                .body("[0].'clinVar'.'hg38'.'end'", containsString("140753336"))
                .body("[0].'clinVar'.'hgvs'.'coding'.size()", equalTo(14))
                .body("[0].'clinVar'.'hgvs'.'coding'", hasItems("LRG_299t1:c.1799T>A", "NM_001354609.2:c.1799T>A", "NM_001374244.1:c.1919T>A", "NM_001374258.1:c.1919T>A", "NM_001378467.1:c.1808T>A", "NM_001378468.1:c.1799T>A", "NM_001378469.1:c.1733T>A", "NM_001378470.1:c.1697T>A", "NM_001378471.1:c.1688T>A", "NM_001378472.1:c.1643T>A", "NM_001378473.1:c.1643T>A", "NM_001378474.1:c.1799T>A", "NM_001378475.1:c.1535T>A", "NM_004333.6:c.1799T>A"))
                .body("[0].'clinVar'.'hgvs'.'genomic'.size()", equalTo(5))
                .body("[0].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_299:g.176429T>A", "NC_000007.12:g.140099605A>T", "NC_000007.13:g.140453136A>T", "NC_000007.14:g.140753336A>T", "NG_007873.3:g.176429T>A"))
                .body("[0].'clinVar'.'hgvs'.'protein'.size()", equalTo(17))
                .body("[0].'clinVar'.'hgvs'.'protein'", hasItems("LRG_299p1:p.Val600Glu", "NP_001341538.1:p.Val600Glu", "NP_001361173.1:p.Val640Glu", "NP_001361187.1:p.Val640Glu", "NP_001365396.1:p.Val603Glu", "NP_001365397.1:p.Val600Glu", "NP_001365398.1:p.Val578Glu", "NP_001365399.1:p.Val566Glu", "NP_001365400.1:p.Val563Glu", "NP_001365401.1:p.Val548Glu", "NP_001365402.1:p.Val548Glu", "NP_001365403.1:p.Val600Glu", "NP_001365404.1:p.Val512Glu", "NP_004324.2:p.Val600Glu", "NP_004324.2:p.Val600Glu", "P15056:p.Val600Glu", "p.V600E"))
                .body("[0].'clinVar'.'rcv'.size()", equalTo(31))
                .body("[0].'clinVar'.'rcv'[0].'accession'", containsString("RCV000014992"))
                .body("[0].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_004333.6(BRAF):c.1799T>A (p.Val600Glu)"))
                .body("[0].'clinVar'.'rcv'[1].'accession'", containsString("RCV000014993"))
                .body("[0].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[1].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_004333.6(BRAF):c.1799T>A (p.Val600Glu)"))
                .body("[0].'clinVar'.'rcv'[2].'accession'", containsString("RCV000014994"))
                .body("[0].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[2].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_004333.6(BRAF):c.1799T>A (p.Val600Glu)"))
                // Skipping assertions on the remaining 28 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'clinVar'.'variantId'", numberMatches(13961.0))
                .body("[0].'mutdb'.'rsid'", containsString("rs113488022"))
                .body("[0].'mutdb'.'ref'", containsString("T"))
                .body("[0].'mutdb'.'alt'", containsString("A"))
                .body("[0].'mutdb'.'uniprotId'", containsString("VAR_018629"))
                .body("[0].'mutdb'.'mutpredScore'", numberMatches(0.705))
                .body("[0].'mutdb'.'cosmicId'", containsString("476, 1131"))
                .body("[0].'mutdb'.'chrom'", containsString("7"))
                .body("[0].'mutdb'.'hg19'.'start'", numberMatches(1.40453136E8))
                .body("[0].'mutdb'.'hg19'.'end'", numberMatches(1.40453136E8))
                .body("[0].'gnomadExome'.'alleleCount'.'ac'", numberMatches(1.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_amr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_asj'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_nfe'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleCount'.'ac_sas'", numberMatches(1.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an'", numberMatches(251260.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_afr'", numberMatches(16252.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_amr'", numberMatches(34528.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_asj'", numberMatches(10076.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_eas'", numberMatches(18392.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_fin'", numberMatches(21638.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_nfe'", numberMatches(113638.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_oth'", numberMatches(6124.0))
                .body("[0].'gnomadExome'.'alleleNumber'.'an_sas'", numberMatches(30612.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af'", numberMatches(3.97994E-6))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_amr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_asj'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_nfe'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'alleleFrequency'.'af_sas'", numberMatches(3.26669E-5))
                .body("[0].'gnomadExome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("[0].'gnomadExome'.'homozygotes'.'hom_sas'", numberMatches(0.0))
                .body("[1].'variant'", containsString("chr17:g.41276045_41276046del"))
                .body("[1].'query'", containsString("chr17:g.41276045_41276046del"))
                .body("[1].'hgvs'", containsString("17:g.41276045_41276046del"))
                .body("[1].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[1].'vcf'.'alt'", containsString("A"))
                .body("[1].'vcf'.'position'", containsString("41276044"))
                .body("[1].'vcf'.'ref'", containsString("ACT"))
                .body("[1].'gnomadExome'.'alleleCount'.'ac'", numberMatches(57.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_amr'", numberMatches(1.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_asj'", numberMatches(41.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_nfe'", numberMatches(11.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleCount'.'ac_sas'", numberMatches(4.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an'", numberMatches(251050.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_afr'", numberMatches(16250.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_amr'", numberMatches(34592.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_asj'", numberMatches(10078.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_eas'", numberMatches(18394.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_fin'", numberMatches(21644.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_nfe'", numberMatches(113354.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_oth'", numberMatches(6130.0))
                .body("[1].'gnomadExome'.'alleleNumber'.'an_sas'", numberMatches(30608.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af'", numberMatches(2.27046E-4))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_amr'", numberMatches(2.89084E-5))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_asj'", numberMatches(0.00406827))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_nfe'", numberMatches(9.70411E-5))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'alleleFrequency'.'af_sas'", numberMatches(1.30685E-4))
                .body("[1].'gnomadExome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("[1].'gnomadExome'.'homozygotes'.'hom_sas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac'", numberMatches(1.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_afr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_amr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_asj'", numberMatches(1.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_eas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_fin'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_nfe'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_oth'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleCount'.'ac_sas'", nullValue())
                .body("[1].'gnomadGenome'.'alleleNumber'.'an'", numberMatches(31392.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_afr'", numberMatches(8712.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_amr'", numberMatches(848.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_asj'", numberMatches(290.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_eas'", numberMatches(1560.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_fin'", numberMatches(3468.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_nfe'", numberMatches(15426.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_oth'", numberMatches(1088.0))
                .body("[1].'gnomadGenome'.'alleleNumber'.'an_sas'", nullValue())
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af'", numberMatches(3.18552E-5))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_afr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_amr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_asj'", numberMatches(0.00344828))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_eas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_fin'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_nfe'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_oth'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'alleleFrequency'.'af_sas'", nullValue())
                .body("[1].'gnomadGenome'.'homozygotes'.'hom'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_afr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_amr'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_asj'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_eas'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_fin'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_nfe'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_oth'", numberMatches(0.0))
                .body("[1].'gnomadGenome'.'homozygotes'.'hom_sas'", nullValue())
                .body("[2].'variant'", containsString("chr6:g.42454850G>A"))
                .body("[2].'query'", containsString("rs12190874"))
                .body("[2].'hgvs'", containsString("rs12190874"))
                .body("[2].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[2].'vcf'.'alt'", containsString("A"))
                .body("[2].'vcf'.'position'", containsString("42454850"))
                .body("[2].'vcf'.'ref'", containsString("G"))
                .body("[2].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[2].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[2].'dbsnp'.'alleles'[0].'allele'", containsString("G"))
                .body("[2].'dbsnp'.'alleles'[1].'allele'", containsString("A"))
                .body("[2].'dbsnp'.'alt'", containsString("A"))
                .body("[2].'dbsnp'.'chrom'", containsString("6"))
                .body("[2].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[2].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[2].'dbsnp'.'flags'[0]", nullValue())
                .body("[2].'dbsnp'.'hg19'.'start'", numberMatches(4.245485E7))
                .body("[2].'dbsnp'.'hg19'.'end'", numberMatches(4.245485E7))
                .body("[2].'dbsnp'.'ref'", containsString("G"))
                .body("[2].'dbsnp'.'rsid'", containsString("rs12190874"))
                .body("[2].'dbsnp'.'vartype'", containsString("snv"))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac'", numberMatches(3282.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_afr'", numberMatches(646.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_amr'", numberMatches(94.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_asj'", numberMatches(12.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_eas'", numberMatches(271.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_fin'", numberMatches(433.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_nfe'", numberMatches(1698.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_oth'", numberMatches(128.0))
                .body("[2].'gnomadGenome'.'alleleCount'.'ac_sas'", nullValue())
                .body("[2].'gnomadGenome'.'alleleNumber'.'an'", numberMatches(31340.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_afr'", numberMatches(8696.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_amr'", numberMatches(846.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_asj'", numberMatches(290.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_eas'", numberMatches(1556.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_fin'", numberMatches(3474.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_nfe'", numberMatches(15394.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_oth'", numberMatches(1084.0))
                .body("[2].'gnomadGenome'.'alleleNumber'.'an_sas'", nullValue())
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af'", numberMatches(0.104722))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_afr'", numberMatches(0.074287))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_amr'", numberMatches(0.111111))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_asj'", numberMatches(0.0413793))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_eas'", numberMatches(0.174165))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_fin'", numberMatches(0.12464))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_nfe'", numberMatches(0.110303))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_oth'", numberMatches(0.118081))
                .body("[2].'gnomadGenome'.'alleleFrequency'.'af_sas'", nullValue())
                .body("[2].'gnomadGenome'.'homozygotes'.'hom'", numberMatches(179.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_afr'", numberMatches(15.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_amr'", numberMatches(7.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_asj'", numberMatches(1.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_eas'", numberMatches(19.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_fin'", numberMatches(26.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_nfe'", numberMatches(103.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_oth'", numberMatches(8.0))
                .body("[2].'gnomadGenome'.'homozygotes'.'hom_sas'", nullValue());
        
    }
    
    
    @Test(timeout = 60000)
    public void test_58() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"chr17:g.7577566_7577567insA\", " + 
                    " \"chr17:g.7577567A\u003eC\", " + 
                    " \"chr17:g.7577567_7577567del\", " + 
                    " \"chr17:g.7577567_7577568insCACATGTAGTTGT\", " + 
                    " \"chr17:g.7577568C\u003eA\", " + 
                    " \"chr17:g.7577568C\u003eG\", " + 
                    " \"chr17:g.7577568C\u003eT\", " + 
                    " \"chr17:g.7577569A\u003eC\", " + 
                    " \"chr17:g.7577569A\u003eG\", " + 
                    " \"chr17:g.7577569_7577569del\", " + 
                    " \"chr17:g.7577569_7577572del\", " + 
                    " \"chr17:g.7577569_7577580del\", " + 
                    " \"chr17:g.7577570C\u003eA\", " + 
                    " \"chr17:g.7577570C\u003eG\", " + 
                    " \"chr17:g.7577570C\u003eT\", " + 
                    " \"chr17:g.7577571A\u003eT\", " + 
                    " \"chr17:g.7577571_7577572insTGTAGT\", " + 
                    " \"chr17:g.7577572_7577572del\", " + 
                    " \"chr17:g.7577572_7577574del\", " + 
                    " \"chr17:g.7577573G\u003eC\", " + 
                    " \"chr17:g.7577573G\u003eT\", " + 
                    " \"chr17:g.7577573_7577573del\", " + 
                    " \"chr17:g.7577574T\u003eC\", " + 
                    " \"chr17:g.7577574T\u003eG\", " + 
                    " \"chr17:g.7577575A\u003eC\", " + 
                    " \"chr17:g.7577575A\u003eG\", " + 
                    " \"chr17:g.7577575A\u003eT\", " + 
                    " \"chr17:g.7577576_7577578del\", " + 
                    " \"chr17:g.7577576_7577582del\", " + 
                    " \"chr17:g.7577577T\u003eC\", " + 
                    " \"chr17:g.7577578T\u003eC\", " + 
                    " \"chr17:g.7577579G\u003eC\", " + 
                    " \"chr17:g.7577579G\u003eT\", " + 
                    " \"chr17:g.7577579_7577581del\", " + 
                    " \"chr17:g.7577580T\u003eC\", " + 
                    " \"chr17:g.7577580T\u003eG\", " + 
                    " \"chr17:g.7577581A\u003eC\", " + 
                    " \"chr17:g.7577581A\u003eG\", " + 
                    " \"chr17:g.7577581A\u003eT\", " + 
                    " \"chr17:g.7577581_7577595del\", " + 
                    " \"chr17:g.7577584_7577585insCTAC\", " + 
                    " \"chr17:g.7577585_7577590del\", " + 
                    " \"chr17:g.7577586A\u003eC\", " + 
                    " \"chr17:g.7577586A\u003eT\", " + 
                    " \"chr17:g.7577586_7577587insTGGTGGTACAGTCAG\", " + 
                    " \"chr17:g.7577587T\u003eA\", " + 
                    " \"chr17:g.7577587_7577587del\", " + 
                    " \"chr17:g.7577588_7577595del\", " + 
                    " \"chr17:g.7577589G\u003eT\", " + 
                    " \"chr17:g.7577589_7577590insT\", " + 
                    " \"chr17:g.7577591_7577596del\", " + 
                    " \"chr17:g.7577592_7577602del\", " + 
                    " \"chr17:g.7577593_7577593del\", " + 
                    " \"chr17:g.7577593_7577594del\", " + 
                    " \"chr17:g.7577594A\u003eT\", " + 
                    " \"chr17:g.7577594_7577595del\", " + 
                    " \"chr17:g.7577595C\u003eT\", " + 
                    " \"chr17:g.7577596_7577596del\", " + 
                    " \"chr17:g.7577599C\u003eG\", " + 
                    " \"chr17:g.7577599_7577600insA\", " + 
                    " \"chr17:g.7577604C\u003eT\", " + 
                    " \"chr17:g.7577608C\u003eT\", " + 
                    " \"chr17:g.7577609C\u003eA\", " + 
                    " \"chr17:g.7577609C\u003eG\", " + 
                    " \"chr17:g.7577609C\u003eT\", " + 
                    " \"chr17:g.7577610T\u003eA\", " + 
                    " \"chr17:g.7577610T\u003eC\", " + 
                    " \"chr17:g.7577610T\u003eG\", " + 
                    " \"chr17:g.7578098_7578176del\", " + 
                    " \"chr17:g.7578175A\u003eG\", " + 
                    " \"chr17:g.7578175_7578233del\", " + 
                    " \"chr17:g.7578176C\u003eA\", " + 
                    " \"chr17:g.7578176C\u003eT\", " + 
                    " \"chr17:g.7578177C\u003eA\", " + 
                    " \"chr17:g.7578177C\u003eG\", " + 
                    " \"chr17:g.7578179C\u003eA\", " + 
                    " \"chr17:g.7578181G\u003eA\", " + 
                    " \"chr17:g.7578181_7578181del\", " + 
                    " \"chr17:g.7578184_7578184del\", " + 
                    " \"chr17:g.7578184_7578187del\", " + 
                    " \"chr17:g.7578184_7578208del\", " + 
                    " \"chr17:g.7578188C\u003eA\", " + 
                    " \"chr17:g.7578189A\u003eC\", " + 
                    " \"chr17:g.7578189A\u003eT\", " + 
                    " \"chr17:g.7578190T\u003eC\", " + 
                    " \"chr17:g.7578190T\u003eG\", " + 
                    " \"chr17:g.7578191A\u003eC\", " + 
                    " \"chr17:g.7578191A\u003eG\", " + 
                    " \"chr17:g.7578191A\u003eT\", " + 
                    " \"chr17:g.7578192_7578192del\", " + 
                    " \"chr17:g.7578192_7578193delinsA\", " + 
                    " \"chr17:g.7578192_7578193insGGCACCACCACACTATGTCGAAAAGTGT\", " + 
                    " \"chr17:g.7578193G\u003eC\", " + 
                    " \"chr17:g.7578194_7578195insCAC\", " + 
                    " \"chr17:g.7578195_7578197del\", " + 
                    " \"chr17:g.7578196A\u003eC\", " + 
                    " \"chr17:g.7578196_7578197insC\", " + 
                    " \"chr17:g.7578196_7578197insCC\", " + 
                    " \"chr17:g.7578196_7578197insCCACCACACTAT\", " + 
                    " \"chr17:g.7578197C\u003eA\", " + 
                    " \"chr17:g.7578197_7578198delinsTT\", " + 
                    " \"chr17:g.7578199A\u003eC\", " + 
                    " \"chr17:g.7578199_7578199del\", " + 
                    " \"chr17:g.7578200_7578222del\", " + 
                    " \"chr17:g.7578201_7578202del\", " + 
                    " \"chr17:g.7578202A\u003eC\", " + 
                    " \"chr17:g.7578202A\u003eT\", " + 
                    " \"chr17:g.7578203C\u003eA\", " + 
                    " \"chr17:g.7578203C\u003eG\", " + 
                    " \"chr17:g.7578203C\u003eT\", " + 
                    " \"chr17:g.7578204A\u003eC\", " + 
                    " \"chr17:g.7578204A\u003eT\", " + 
                    " \"chr17:g.7578205C\u003eA\", " + 
                    " \"chr17:g.7578205C\u003eG\", " + 
                    " \"chr17:g.7578205C\u003eT\", " + 
                    " \"chr17:g.7578205_7578206insTATGTCGAAAA\", " + 
                    " \"chr17:g.7578205_7578207delinsTTT\", " + 
                    " \"chr17:g.7578206T\u003eA\", " + 
                    " \"chr17:g.7578206T\u003eC\", " + 
                    " \"chr17:g.7578206T\u003eG\", " + 
                    " \"chr17:g.7578206_7578207del\", " + 
                    " \"chr17:g.7578207A\u003eC\", " + 
                    " \"chr17:g.7578207_7578215del\", " + 
                    " \"chr17:g.7578208T\u003eA\", " + 
                    " \"chr17:g.7578208T\u003eC\", " + 
                    " \"chr17:g.7578208T\u003eG\", " + 
                    " \"chr17:g.7578208_7578208del\", " + 
                    " \"chr17:g.7578208_7578217del\", " + 
                    " \"chr17:g.7578208_7578219del\", " + 
                    " \"chr17:g.7578209_7578209del\", " + 
                    " \"chr17:g.7578210_7578210del\", " + 
                    " \"chr17:g.7578211C\u003eA\", " + 
                    " \"chr17:g.7578211C\u003eT\", " + 
                    " \"chr17:g.7578211_7578212delinsA\", " + 
                    " \"chr17:g.7578212G\u003eA\", " + 
                    " \"chr17:g.7578212G\u003eC\", " + 
                    " \"chr17:g.7578212_7578213insA\", " + 
                    " \"chr17:g.7578213_7578213del\", " + 
                    " \"chr17:g.7578213_7578214del\", " + 
                    " \"chr17:g.7578213_7578231del\", " + 
                    " \"chr17:g.7578215_7578227del\", " + 
                    " \"chr17:g.7578217G\u003eA\", " + 
                    " \"chr17:g.7578219_7578220insT\", " + 
                    " \"chr17:g.7578220_7578221del\", " + 
                    " \"chr17:g.7578220_7578221insTTCTGTCATCCAAATACTCCACACG\", " + 
                    " \"chr17:g.7578222_7578223del\", " + 
                    " \"chr17:g.7578223C\u003eT\", " + 
                    " \"chr17:g.7578224T\u003eA\", " + 
                    " \"chr17:g.7578225G\u003eT\", " + 
                    " \"chr17:g.7578226T\u003eA\", " + 
                    " \"chr17:g.7578226_7578229del\", " + 
                    " \"chr17:g.7578227C\u003eT\", " + 
                    " \"chr17:g.7578229_7578237del\", " + 
                    " \"chr17:g.7578230_7578230del\", " + 
                    " \"chr17:g.7578231_7578232insAAAT\", " + 
                    " \"chr17:g.7578235T\u003eA\", " + 
                    " \"chr17:g.7578235T\u003eC\", " + 
                    " \"chr17:g.7578235T\u003eG\", " + 
                    " \"chr17:g.7578236A\u003eG\", " + 
                    " \"chr17:g.7578238_7578255del\", " + 
                    " \"chr17:g.7578239C\u003eA\", " + 
                    " \"chr17:g.7578239_7578240delinsAG\", " + 
                    " \"chr17:g.7578239_7578240insCA\", " + 
                    " \"chr17:g.7578242C\u003eA\", " + 
                    " \"chr17:g.7578244_7578261del\", " + 
                    " \"chr17:g.7578245G\u003eA\", " + 
                    " \"chr17:g.7578245_7578277del\", " + 
                    " \"chr17:g.7578247A\u003eT\", " + 
                    " \"chr17:g.7578247_7578247del\", " + 
                    " \"chr17:g.7578247_7578248del\", " + 
                    " \"chr17:g.7578248_7578249insAT\", " + 
                    " \"chr17:g.7578250_7578250del\", " + 
                    " \"chr17:g.7578252_7578299del\", " + 
                    " \"chr17:g.7578253C\u003eA\", " + 
                    " \"chr17:g.7578253C\u003eT\", " + 
                    " \"chr17:g.7578254C\u003eA\", " + 
                    " \"chr17:g.7578257C\u003eA\", " + 
                    " \"chr17:g.7578257_7578257del\", " + 
                    " \"chr17:g.7578257_7578258delinsAA\", " + 
                    " \"chr17:g.7578259A\u003eT\", " + 
                    " \"chr17:g.7578260C\u003eT\", " + 
                    " \"chr17:g.7578262C\u003eA\", " + 
                    " \"chr17:g.7578262C\u003eG\", " + 
                    " \"chr17:g.7578262C\u003eT\", " + 
                    " \"chr17:g.7578263G\u003eA\", " + 
                    " \"chr17:g.7578263_7578282del\", " + 
                    " \"chr17:g.7578264_7578265delinsTT\", " + 
                    " \"chr17:g.7578264_7578338delinsTT\", " + 
                    " \"chr17:g.7578265A\u003eC\", " + 
                    " \"chr17:g.7578265A\u003eG\", " + 
                    " \"chr17:g.7578265A\u003eT\", " + 
                    " \"chr17:g.7578265_7578266insT\", " + 
                    " \"chr17:g.7578266T\u003eA\", " + 
                    " \"chr17:g.7578266_7578266del\", " + 
                    " \"chr17:g.7578266_7578267insA\", " + 
                    " \"chr17:g.7578267_7578268insAG\", " + 
                    " \"chr17:g.7578268A\u003eC\", " + 
                    " \"chr17:g.7578268A\u003eG\", " + 
                    " \"chr17:g.7578268_7578280del\", " + 
                    " \"chr17:g.7578269G\u003eA\", " + 
                    " \"chr17:g.7578270A\u003eT\", " + 
                    " \"chr17:g.7578271T\u003eA\", " + 
                    " \"chr17:g.7578271T\u003eC\", " + 
                    " \"chr17:g.7578271T\u003eG\", " + 
                    " \"chr17:g.7578272G\u003eA\", " + 
                    " \"chr17:g.7578272G\u003eC\", " + 
                    " \"chr17:g.7578272G\u003eT\", " + 
                    " \"chr17:g.7578272_7578273delinsAA\", " + 
                    " \"chr17:g.7578274T\u003eA\", " + 
                    " \"chr17:g.7578275G\u003eA\", " + 
                    " \"chr17:g.7578275_7578277del\", " + 
                    " \"chr17:g.7578275_7578293del\", " + 
                    " \"chr17:g.7578277_7578281del\", " + 
                    " \"chr17:g.7578280G\u003eA\", " + 
                    " \"chr17:g.7578280_7578280del\", " + 
                    " \"chr17:g.7578281G\u003eA\", " + 
                    " \"chr17:g.7578281G\u003eT\", " + 
                    " \"chr17:g.7578282_7578306del\", " + 
                    " \"chr17:g.7578283_7578284insC\", " + 
                    " \"chr17:g.7578284_7578284del\", " + 
                    " \"chr17:g.7578284_7578288del\", " + 
                    " \"chr17:g.7578284_7578297del\", " + 
                    " \"chr17:g.7578284_7578310del\", " + 
                    " \"chr17:g.7578285_7578296del\", " + 
                    " \"chr17:g.7578286A\u003eT\", " + 
                    " \"chr17:g.7578286_7578286del\", " + 
                    " \"chr17:g.7578287_7578287del\", " + 
                    " \"chr17:g.7578289C\u003eT\", " + 
                    " \"chr17:g.7578289_7578290delinsTT\", " + 
                    " \"chr17:g.7578290C\u003eA\", " + 
                    " \"chr17:g.7578290C\u003eG\", " + 
                    " \"chr17:g.7578290C\u003eT\", " + 
                    " \"chr17:g.7578291T\u003eA\", " + 
                    " \"chr17:g.7578291T\u003eC\", " + 
                    " \"chr17:g.7578291T\u003eG\", " + 
                    " \"chr17:g.7578293_7578977del\", " + 
                    " \"chr17:g.7578313_7578425del\", " + 
                    " \"chr17:g.7578359_7578373del\", " + 
                    " \"chr17:g.7578359_7578375del\", " + 
                    " \"chr17:g.7578360_7578373del\", " + 
                    " \"chr17:g.7578362_7578385del\", " + 
                    " \"chr17:g.7578365_7578383del\", " + 
                    " \"chr17:g.7578370C\u003eA\", " + 
                    " \"chr17:g.7578370C\u003eG\", " + 
                    " \"chr17:g.7578370C\u003eT\", " + 
                    " \"chr17:g.7578371C\u003eG\", " + 
                    " \"chr17:g.7578371_7578382del\", " + 
                    " \"chr17:g.7578372_7578373delinsC\", " + 
                    " \"chr17:g.7578376_7578376del\", " + 
                    " \"chr17:g.7578376_7578379del\", " + 
                    " \"chr17:g.7578377_7578377del\", " + 
                    " \"chr17:g.7578377_7578378del\", " + 
                    " \"chr17:g.7578378A\u003eT\", " + 
                    " \"chr17:g.7578378_7578379insTC\", " + 
                    " \"chr17:g.7578378_7578395del\", " + 
                    " \"chr17:g.7578379_7578379del\", " + 
                    " \"chr17:g.7578382G\u003eC\", " + 
                    " \"chr17:g.7578382_7578382del\", " + 
                    " \"chr17:g.7578384_7578401del\", " + 
                    " \"chr17:g.7578388C\u003eG\", " + 
                    " \"chr17:g.7578388C\u003eT\", " + 
                    " \"chr17:g.7578389G\u003eA\", " + 
                    " \"chr17:g.7578389_7578402delinsT\", " + 
                    " \"chr17:g.7578390C\u003eG\", " + 
                    " \"chr17:g.7578392C\u003eA\", " + 
                    " \"chr17:g.7578392C\u003eT\", " + 
                    " \"chr17:g.7578393A\u003eC\", " + 
                    " \"chr17:g.7578393A\u003eT\", " + 
                    " \"chr17:g.7578393_7578405del\", " + 
                    " \"chr17:g.7578394T\u003eA\", " + 
                    " \"chr17:g.7578394T\u003eC\", " + 
                    " \"chr17:g.7578394T\u003eG\", " + 
                    " \"chr17:g.7578394_7578395del\", " + 
                    " \"chr17:g.7578394_7578396del\", " + 
                    " \"chr17:g.7578395G\u003eA\", " + 
                    " \"chr17:g.7578395G\u003eC\", " + 
                    " \"chr17:g.7578395G\u003eT\", " + 
                    " \"chr17:g.7578397T\u003eG\", " + 
                    " \"chr17:g.7578397_7578397del\", " + 
                    " \"chr17:g.7578397_7578398insG\", " + 
                    " \"chr17:g.7578398G\u003eC\", " + 
                    " \"chr17:g.7578398_7578398del\", " + 
                    " \"chr17:g.7578400G\u003eC\", " + 
                    " \"chr17:g.7578401_7578402delinsT\", " + 
                    " \"chr17:g.7578402G\u003eC\", " + 
                    " \"chr17:g.7578402G\u003eT\", " + 
                    " \"chr17:g.7578403C\u003eA\", " + 
                    " \"chr17:g.7578403C\u003eT\", " + 
                    " \"chr17:g.7578404A\u003eC\", " + 
                    " \"chr17:g.7578404A\u003eG\", " + 
                    " \"chr17:g.7578404A\u003eT\", " + 
                    " \"chr17:g.7578406C\u003eA\", " + 
                    " \"chr17:g.7578406C\u003eT\", " + 
                    " \"chr17:g.7578406_7578406del\", " + 
                    " \"chr17:g.7578407G\u003eA\", " + 
                    " \"chr17:g.7578407G\u003eC\", " + 
                    " \"chr17:g.7578407_7578412delinsCCC\", " + 
                    " \"chr17:g.7578410T\u003eA\", " + 
                    " \"chr17:g.7578410T\u003eC\", " + 
                    " \"chr17:g.7578411_7578411del\", " + 
                    " \"chr17:g.7578412A\u003eC\", " + 
                    " \"chr17:g.7578412A\u003eG\", " + 
                    " \"chr17:g.7578413C\u003eA\", " + 
                    " \"chr17:g.7578413C\u003eG\", " + 
                    " \"chr17:g.7578413C\u003eT\", " + 
                    " \"chr17:g.7578413_7578413del\", " + 
                    " \"chr17:g.7578414_7578414del\", " + 
                    " \"chr17:g.7578415A\u003eT\", " + 
                    " \"chr17:g.7578416C\u003eA\", " + 
                    " \"chr17:g.7578417_7578419delinsTG\", " + 
                    " \"chr17:g.7578418T\u003eC\", " + 
                    " \"chr17:g.7578419C\u003eA\", " + 
                    " \"chr17:g.7578419_7578419del\", " + 
                    " \"chr17:g.7578419_7578420insCGTCATGTG\", " + 
                    " \"chr17:g.7578421G\u003eA\", " + 
                    " \"chr17:g.7578422_7578423insCATGTGCTG\", " + 
                    " \"chr17:g.7578422_7578424del\", " + 
                    " \"chr17:g.7578423_7578424insA\", " + 
                    " \"chr17:g.7578423_7579121del\", " + 
                    " \"chr17:g.7578424_7578425del\", " + 
                    " \"chr17:g.7578424_7578427del\", " + 
                    " \"chr17:g.7578427T\u003eC\", " + 
                    " \"chr17:g.7578427T\u003eG\", " + 
                    " \"chr17:g.7578428_7578431delinsACTA\", " + 
                    " \"chr17:g.7578428_7578627del\", " + 
                    " \"chr17:g.7578429_7578430insTG\", " + 
                    " \"chr17:g.7578430_7578431del\", " + 
                    " \"chr17:g.7578431G\u003eA\", " + 
                    " \"chr17:g.7578433G\u003eC\", " + 
                    " \"chr17:g.7578433G\u003eT\", " + 
                    " \"chr17:g.7578435C\u003eA\", " + 
                    " \"chr17:g.7578437G\u003eA\", " + 
                    " \"chr17:g.7578439_7578439del\", " + 
                    " \"chr17:g.7578440T\u003eA\", " + 
                    " \"chr17:g.7578440T\u003eC\", " + 
                    " \"chr17:g.7578441G\u003eT\", " + 
                    " \"chr17:g.7578441_7578442delinsA\", " + 
                    " \"chr17:g.7578442T\u003eC\", " + 
                    " \"chr17:g.7578442_7578450delinsCCG\", " + 
                    " \"chr17:g.7578443A\u003eC\", " + 
                    " \"chr17:g.7578443A\u003eG\", " + 
                    " \"chr17:g.7578443A\u003eT\", " + 
                    " \"chr17:g.7578443_7578691del\", " + 
                    " \"chr17:g.7578445A\u003eC\", " + 
                    " \"chr17:g.7578445A\u003eT\", " + 
                    " \"chr17:g.7578445_7578449delinsCTGGG\", " + 
                    " \"chr17:g.7578446T\u003eA\", " + 
                    " \"chr17:g.7578447_7578447del\", " + 
                    " \"chr17:g.7578448G\u003eT\", " + 
                    " \"chr17:g.7578449C\u003eA\", " + 
                    " \"chr17:g.7578449C\u003eT\", " + 
                    " \"chr17:g.7578452_7578453insGGCGCG\", " + 
                    " \"chr17:g.7578453_7578455del\", " + 
                    " \"chr17:g.7578453_7578458del\", " + 
                    " \"chr17:g.7578454G\u003eA\", " + 
                    " \"chr17:g.7578454_7578455del\", " + 
                    " \"chr17:g.7578454_7578457del\", " + 
                    " \"chr17:g.7578454_7578464delinsACGC\", " + 
                    " \"chr17:g.7578454_7578467del\", " + 
                    " \"chr17:g.7578455C\u003eG\", " + 
                    " \"chr17:g.7578455_7578460del\", " + 
                    " \"chr17:g.7578457C\u003eA\", " + 
                    " \"chr17:g.7578457C\u003eG\", " + 
                    " \"chr17:g.7578457C\u003eT\", " + 
                    " \"chr17:g.7578457_7578458delinsAA\", " + 
                    " \"chr17:g.7578458G\u003eA\", " + 
                    " \"chr17:g.7578458G\u003eC\", " + 
                    " \"chr17:g.7578458_7578459insA\", " + 
                    " \"chr17:g.7578458_7578464del\", " + 
                    " \"chr17:g.7578460A\u003eC\", " + 
                    " \"chr17:g.7578460A\u003eG\", " + 
                    " \"chr17:g.7578461C\u003eA\", " + 
                    " \"chr17:g.7578461_7578473del\", " + 
                    " \"chr17:g.7578462_7578463insC\", " + 
                    " \"chr17:g.7578463C\u003eA\", " + 
                    " \"chr17:g.7578463C\u003eG\", " + 
                    " \"chr17:g.7578463C\u003eT\", " + 
                    " \"chr17:g.7578463_7578464insA\", " + 
                    " \"chr17:g.7578464G\u003eC\", " + 
                    " \"chr17:g.7578464_7578464del\", " + 
                    " \"chr17:g.7578464_7578476del\", " + 
                    " \"chr17:g.7578465_7578470del\", " + 
                    " \"chr17:g.7578466G\u003eA\", " + 
                    " \"chr17:g.7578466G\u003eT\", " + 
                    " \"chr17:g.7578467T\u003eG\", " + 
                    " \"chr17:g.7578467_7578468insG\", " + 
                    " \"chr17:g.7578467_7578468insGCCGGGC\", " + 
                    " \"chr17:g.7578467_7578498del\", " + 
                    " \"chr17:g.7578469C\u003eA\", " + 
                    " \"chr17:g.7578469C\u003eT\", " + 
                    " \"chr17:g.7578469_7578469del\", " + 
                    " \"chr17:g.7578469_7578470insCGGG\", " + 
                    " \"chr17:g.7578469_7578470insGGG\", " + 
                    " \"chr17:g.7578469_7578474del\", " + 
                    " \"chr17:g.7578470_7578471delinsA\", " + 
                    " \"chr17:g.7578470_7578471insG\", " + 
                    " \"chr17:g.7578470_7578473del\", " + 
                    " \"chr17:g.7578470_7578482del\", " + 
                    " \"chr17:g.7578470_7578502del\", " + 
                    " \"chr17:g.7578471_7578471del\", " + 
                    " \"chr17:g.7578471_7578478del\", " + 
                    " \"chr17:g.7578471_7578483del\", " + 
                    " \"chr17:g.7578472G\u003eA\", " + 
                    " \"chr17:g.7578472_7578491del\", " + 
                    " \"chr17:g.7578473_7578474insA\", " + 
                    " \"chr17:g.7578473_7578474insC\", " + 
                    " \"chr17:g.7578474_7578475insG\", " + 
                    " \"chr17:g.7578475G\u003eA\", " + 
                    " \"chr17:g.7578475G\u003eC\", " + 
                    " \"chr17:g.7578475_7578475del\", " + 
                    " \"chr17:g.7578475_7578476delinsTT\", " + 
                    " \"chr17:g.7578476G\u003eA\", " + 
                    " \"chr17:g.7578476G\u003eT\", " + 
                    " \"chr17:g.7578476_7578477insGGGT\", " + 
                    " \"chr17:g.7578478G\u003eC\", " + 
                    " \"chr17:g.7578478G\u003eT\", " + 
                    " \"chr17:g.7578478_7578479delinsAA\", " + 
                    " \"chr17:g.7578478_7578480del\", " + 
                    " \"chr17:g.7578479G\u003eA\", " + 
                    " \"chr17:g.7578479G\u003eC\", " + 
                    " \"chr17:g.7578479G\u003eT\", " + 
                    " \"chr17:g.7578480_7578488del\", " + 
                    " \"chr17:g.7578480_7578495del\", " + 
                    " \"chr17:g.7578481_7578481del\", " + 
                    " \"chr17:g.7578481_7578482insT\", " + 
                    " \"chr17:g.7578482_7578482del\", " + 
                    " \"chr17:g.7578484_7578485insA\", " + 
                    " \"chr17:g.7578484_7578502del\", " + 
                    " \"chr17:g.7578485_7578485del\", " + 
                    " \"chr17:g.7578486A\u003eT\", " + 
                    " \"chr17:g.7578488_7578488del\", " + 
                    " \"chr17:g.7578490A\u003eC\", " + 
                    " \"chr17:g.7578491_7578491del\", " + 
                    " \"chr17:g.7578492C\u003eG\", " + 
                    " \"chr17:g.7578492C\u003eT\", " + 
                    " \"chr17:g.7578493C\u003eT\", " + 
                    " \"chr17:g.7578493_7578500del\", " + 
                    " \"chr17:g.7578494A\u003eC\", " + 
                    " \"chr17:g.7578495_7578507del\", " + 
                    " \"chr17:g.7578496A\u003eC\", " + 
                    " \"chr17:g.7578496A\u003eG\", " + 
                    " \"chr17:g.7578496A\u003eT\", " + 
                    " \"chr17:g.7578499T\u003eA\", " + 
                    " \"chr17:g.7578499T\u003eC\", " + 
                    " \"chr17:g.7578499T\u003eG\", " + 
                    " \"chr17:g.7578499_7578499del\", " + 
                    " \"chr17:g.7578500G\u003eA\", " + 
                    " \"chr17:g.7578500_7578500del\", " + 
                    " \"chr17:g.7578500_7578524del\", " + 
                    " \"chr17:g.7578501_7578502insA\", " + 
                    " \"chr17:g.7578502A\u003eC\", " + 
                    " \"chr17:g.7578502A\u003eG\", " + 
                    " \"chr17:g.7578503C\u003eT\", " + 
                    " \"chr17:g.7578503_7578507del\", " + 
                    " \"chr17:g.7578503_7578518del\", " + 
                    " \"chr17:g.7578504_7578505insGGG\", " + 
                    " \"chr17:g.7578505G\u003eA\", " + 
                    " \"chr17:g.7578505_7578505del\", " + 
                    " \"chr17:g.7578505_7578510del\", " + 
                    " \"chr17:g.7578507G\u003eC\", " + 
                    " \"chr17:g.7578507G\u003eT\", " + 
                    " \"chr17:g.7578508C\u003eT\", " + 
                    " \"chr17:g.7578508_7578535delinsG\", " + 
                    " \"chr17:g.7578509A\u003eC\", " + 
                    " \"chr17:g.7578509A\u003eG\", " + 
                    " \"chr17:g.7578510_7578510del\", " + 
                    " \"chr17:g.7578511G\u003eA\", " + 
                    " \"chr17:g.7578513C\u003eG\", " + 
                    " \"chr17:g.7578513_7578549del\", " + 
                    " \"chr17:g.7578514T\u003eG\", " + 
                    " \"chr17:g.7578515T\u003eA\", " + 
                    " \"chr17:g.7578515_7578516insGGCCAGTTGGCAAAACATCTTGTTGAG\", " + 
                    " \"chr17:g.7578516_7578516del\", " + 
                    " \"chr17:g.7578517G\u003eA\", " + 
                    " \"chr17:g.7578518C\u003eG\", " + 
                    " \"chr17:g.7578520A\u003eT\", " + 
                    " \"chr17:g.7578520_7578525del\", " + 
                    " \"chr17:g.7578521_7578521del\", " + 
                    " \"chr17:g.7578521_7578531del\", " + 
                    " \"chr17:g.7578522_7578522del\", " + 
                    " \"chr17:g.7578523T\u003eG\", " + 
                    " \"chr17:g.7578523_7578548del\", " + 
                    " \"chr17:g.7578524G\u003eA\", " + 
                    " \"chr17:g.7578524G\u003eC\", " + 
                    " \"chr17:g.7578525G\u003eC\", " + 
                    " \"chr17:g.7578525G\u003eT\", " + 
                    " \"chr17:g.7578526C\u003eA\", " + 
                    " \"chr17:g.7578526C\u003eG\", " + 
                    " \"chr17:g.7578526C\u003eT\", " + 
                    " \"chr17:g.7578526_7578527insA\", " + 
                    " \"chr17:g.7578527A\u003eC\", " + 
                    " \"chr17:g.7578527A\u003eG\", " + 
                    " \"chr17:g.7578527A\u003eT\", " + 
                    " \"chr17:g.7578527_7578527del\", " + 
                    " \"chr17:g.7578527_7578528del\", " + 
                    " \"chr17:g.7578528A\u003eC\", " + 
                    " \"chr17:g.7578528A\u003eT\", " + 
                    " \"chr17:g.7578529A\u003eC\", " + 
                    " \"chr17:g.7578530A\u003eC\", " + 
                    " \"chr17:g.7578530A\u003eG\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/my_variant_info/variant?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(280))
                .body("[0].'variant'", containsString("chr17:g.7577567A>C"))
                .body("[0].'query'", containsString("chr17:g.7577567A>C"))
                .body("[0].'hgvs'", containsString("chr17:g.7577567A>C"))
                .body("[0].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[0].'vcf'.'alt'", containsString("C"))
                .body("[0].'vcf'.'position'", containsString("7577567"))
                .body("[0].'vcf'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[0].'dbsnp'.'alleles'.size()", equalTo(1))
                .body("[0].'dbsnp'.'alleles'[0]", nullValue())
                .body("[0].'dbsnp'.'alt'", containsString("C"))
                .body("[0].'dbsnp'.'chrom'", containsString("17"))
                .body("[0].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[0].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[0].'dbsnp'.'flags'[0]", nullValue())
                .body("[0].'dbsnp'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'dbsnp'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[0].'dbsnp'.'ref'", containsString("A"))
                .body("[0].'dbsnp'.'rsid'", containsString("rs193920789"))
                .body("[0].'dbsnp'.'vartype'", containsString("snv"))
                .body("[0].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[0].'cosmic'.'alt'", containsString("G"))
                .body("[0].'cosmic'.'chrom'", containsString("17"))
                .body("[0].'cosmic'.'cosmicId'", containsString("COSM44676"))
                .body("[0].'cosmic'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'cosmic'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[0].'cosmic'.'mutFreq'", numberMatches(0.03))
                .body("[0].'cosmic'.'mutNt'", containsString("T>G"))
                .body("[0].'cosmic'.'ref'", containsString("T"))
                .body("[0].'cosmic'.'tumorSite'", containsString("urinary_tract"))
                .body("[0].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[0].'clinVar'.'alleleId'", numberMatches(171615.0))
                .body("[0].'clinVar'.'alt'", containsString("C"))
                .body("[0].'clinVar'.'chrom'", containsString("17"))
                .body("[0].'clinVar'.'cytogenic'", containsString("17p13.1"))
                .body("[0].'clinVar'.'gene'.'symbol'", containsString("TP53"))
                .body("[0].'clinVar'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'clinVar'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[0].'clinVar'.'hg38'.'start'", containsString("7674249"))
                .body("[0].'clinVar'.'hg38'.'end'", containsString("7674249"))
                .body("[0].'clinVar'.'hgvs'.'coding'.size()", equalTo(17))
                .body("[0].'clinVar'.'hgvs'.'coding'", hasItems("LRG_321t1:c.714T>G", "LRG_321t5:c.318T>G", "NM_000546.6:c.714T>G", "NM_001126112.3:c.714T>G", "NM_001126113.3:c.714T>G", "NM_001126114.3:c.714T>G", "NM_001126115.2:c.318T>G", "NM_001126116.2:c.318T>G", "NM_001126117.2:c.318T>G", "NM_001126118.2:c.597T>G", "NM_001276695.3:c.597T>G", "NM_001276696.3:c.597T>G", "NM_001276697.3:c.237T>G", "NM_001276698.3:c.237T>G", "NM_001276699.3:c.237T>G", "NM_001276760.3:c.597T>G", "NM_001276761.3:c.597T>G"))
                .body("[0].'clinVar'.'hgvs'.'genomic'.size()", equalTo(4))
                .body("[0].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_321:g.18302T>G", "NC_000017.10:g.7577567A>C", "NC_000017.11:g.7674249A>C", "NG_017013.2:g.18302T>G"))
                .body("[0].'clinVar'.'hgvs'.'protein'.size()", equalTo(19))
                .body("[0].'clinVar'.'hgvs'.'protein'", hasItems("LRG_321p1:p.Cys238Trp", "LRG_321p5:p.Cys106Trp", "NP_000537.3:p.Cys238Trp", "NP_000537.3:p.Cys238Trp", "NP_001119584.1:p.Cys238Trp", "NP_001119585.1:p.Cys238Trp", "NP_001119586.1:p.Cys238Trp", "NP_001119587.1:p.Cys106Trp", "NP_001119587.1:p.Cys106Trp", "NP_001119588.1:p.Cys106Trp", "NP_001119589.1:p.Cys106Trp", "NP_001119590.1:p.Cys199Trp", "NP_001263624.1:p.Cys199Trp", "NP_001263625.1:p.Cys199Trp", "NP_001263626.1:p.Cys79Trp", "NP_001263627.1:p.Cys79Trp", "NP_001263628.1:p.Cys79Trp", "NP_001263689.1:p.Cys199Trp", "NP_001263690.1:p.Cys199Trp"))
                .body("[0].'clinVar'.'rcv'.size()", equalTo(3))
                .body("[0].'clinVar'.'rcv'[0].'accession'", containsString("RCV000149049"))
                .body("[0].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Uncertain significance"))
                .body("[0].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[0].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000546.6(TP53):c.714T>G (p.Cys238Trp)"))
                .body("[0].'clinVar'.'rcv'[1].'accession'", containsString("RCV000203823"))
                .body("[0].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'clinVar'.'rcv'[1].'origin'", containsString("germline"))
                .body("[0].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000546.6(TP53):c.714T>G (p.Cys238Trp)"))
                .body("[0].'clinVar'.'rcv'[2].'accession'", containsString("RCV002362784"))
                .body("[0].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[0].'clinVar'.'rcv'[2].'origin'", containsString("germline"))
                .body("[0].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_000546.6(TP53):c.714T>G (p.Cys238Trp)"))
                .body("[0].'clinVar'.'variantId'", numberMatches(161515.0))
                .body("[0].'mutdb'.'ref'", containsString("T"))
                .body("[0].'mutdb'.'alt'", containsString("G"))
                .body("[0].'mutdb'.'uniprotId'", containsString("VAR_045203"))
                .body("[0].'mutdb'.'mutpredScore'", numberMatches(0.963))
                .body("[0].'mutdb'.'cosmicId'", containsString("44676"))
                .body("[0].'mutdb'.'chrom'", containsString("17"))
                .body("[0].'mutdb'.'hg19'.'start'", numberMatches(7577567.0))
                .body("[0].'mutdb'.'hg19'.'end'", numberMatches(7577567.0))
                .body("[1].'variant'", containsString("chr17:g.7577568C>A"))
                .body("[1].'query'", containsString("chr17:g.7577568C>A"))
                .body("[1].'hgvs'", containsString("chr17:g.7577568C>A"))
                .body("[1].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[1].'vcf'.'alt'", containsString("A"))
                .body("[1].'vcf'.'position'", containsString("7577568"))
                .body("[1].'vcf'.'ref'", containsString("C"))
                .body("[1].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[1].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[1].'dbsnp'.'alleles'[0].'allele'", containsString("C"))
                .body("[1].'dbsnp'.'alleles'[1].'allele'", containsString("T"))
                .body("[1].'dbsnp'.'alt'", containsString("A"))
                .body("[1].'dbsnp'.'chrom'", containsString("17"))
                .body("[1].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[1].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[1].'dbsnp'.'flags'[0]", nullValue())
                .body("[1].'dbsnp'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'dbsnp'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[1].'dbsnp'.'ref'", containsString("C"))
                .body("[1].'dbsnp'.'rsid'", containsString("rs730882005"))
                .body("[1].'dbsnp'.'vartype'", containsString("snv"))
                .body("[1].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[1].'cosmic'.'alt'", containsString("T"))
                .body("[1].'cosmic'.'chrom'", containsString("17"))
                .body("[1].'cosmic'.'cosmicId'", containsString("COSM43778"))
                .body("[1].'cosmic'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'cosmic'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[1].'cosmic'.'mutFreq'", numberMatches(0.63))
                .body("[1].'cosmic'.'mutNt'", containsString("G>T"))
                .body("[1].'cosmic'.'ref'", containsString("G"))
                .body("[1].'cosmic'.'tumorSite'", containsString("meninges"))
                .body("[1].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[1].'clinVar'.'alleleId'", numberMatches(363453.0))
                .body("[1].'clinVar'.'alt'", containsString("A"))
                .body("[1].'clinVar'.'chrom'", containsString("17"))
                .body("[1].'clinVar'.'cytogenic'", containsString("17p13.1"))
                .body("[1].'clinVar'.'gene'.'symbol'", containsString("TP53"))
                .body("[1].'clinVar'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'clinVar'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[1].'clinVar'.'hg38'.'start'", containsString("7674250"))
                .body("[1].'clinVar'.'hg38'.'end'", containsString("7674250"))
                .body("[1].'clinVar'.'hgvs'.'coding'.size()", equalTo(16))
                .body("[1].'clinVar'.'hgvs'.'coding'", hasItems("LRG_321t1:c.713G>T", "NM_000546.6:c.713G>T", "NM_001126112.3:c.713G>T", "NM_001126113.3:c.713G>T", "NM_001126114.3:c.713G>T", "NM_001126115.2:c.317G>T", "NM_001126116.2:c.317G>T", "NM_001126117.2:c.317G>T", "NM_001126118.2:c.596G>T", "NM_001276695.3:c.596G>T", "NM_001276696.3:c.596G>T", "NM_001276697.3:c.236G>T", "NM_001276698.3:c.236G>T", "NM_001276699.3:c.236G>T", "NM_001276760.3:c.596G>T", "NM_001276761.3:c.596G>T"))
                .body("[1].'clinVar'.'hgvs'.'genomic'.size()", equalTo(4))
                .body("[1].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_321:g.18301G>T", "NC_000017.10:g.7577568C>A", "NC_000017.11:g.7674250C>A", "NG_017013.2:g.18301G>T"))
                .body("[1].'clinVar'.'hgvs'.'protein'.size()", equalTo(17))
                .body("[1].'clinVar'.'hgvs'.'protein'", hasItems("LRG_321p1:p.Cys238Phe", "NP_000537.3:p.Cys238Phe", "NP_000537.3:p.Cys238Phe", "NP_001119584.1:p.Cys238Phe", "NP_001119585.1:p.Cys238Phe", "NP_001119586.1:p.Cys238Phe", "NP_001119587.1:p.Cys106Phe", "NP_001119588.1:p.Cys106Phe", "NP_001119589.1:p.Cys106Phe", "NP_001119590.1:p.Cys199Phe", "NP_001263624.1:p.Cys199Phe", "NP_001263625.1:p.Cys199Phe", "NP_001263626.1:p.Cys79Phe", "NP_001263627.1:p.Cys79Phe", "NP_001263628.1:p.Cys79Phe", "NP_001263689.1:p.Cys199Phe", "NP_001263690.1:p.Cys199Phe"))
                .body("[1].'clinVar'.'rcv'.size()", equalTo(19))
                .body("[1].'clinVar'.'rcv'[0].'accession'", containsString("RCV000417498"))
                .body("[1].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[1].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[1].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000546.6(TP53):c.713G>T (p.Cys238Phe)"))
                .body("[1].'clinVar'.'rcv'[1].'accession'", containsString("RCV000418555"))
                .body("[1].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[1].'clinVar'.'rcv'[1].'origin'", containsString("somatic"))
                .body("[1].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000546.6(TP53):c.713G>T (p.Cys238Phe)"))
                .body("[1].'clinVar'.'rcv'[2].'accession'", containsString("RCV000422481"))
                .body("[1].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[1].'clinVar'.'rcv'[2].'origin'", containsString("somatic"))
                .body("[1].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_000546.6(TP53):c.713G>T (p.Cys238Phe)"))
                // Skipping assertions on the remaining 16 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'clinVar'.'variantId'", numberMatches(376574.0))
                .body("[1].'mutdb'.'ref'", containsString("G"))
                .body("[1].'mutdb'.'alt'", containsString("T"))
                .body("[1].'mutdb'.'uniprotId'", containsString("VAR_005966"))
                .body("[1].'mutdb'.'mutpredScore'", numberMatches(0.976))
                .body("[1].'mutdb'.'cosmicId'", containsString("43778, 99626, 99624, 99625"))
                .body("[1].'mutdb'.'chrom'", containsString("17"))
                .body("[1].'mutdb'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[1].'mutdb'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'variant'", containsString("chr17:g.7577568C>G"))
                .body("[2].'query'", containsString("chr17:g.7577568C>G"))
                .body("[2].'hgvs'", containsString("chr17:g.7577568C>G"))
                .body("[2].'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[2].'vcf'.'alt'", containsString("G"))
                .body("[2].'vcf'.'position'", containsString("7577568"))
                .body("[2].'vcf'.'ref'", containsString("C"))
                .body("[2].'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[2].'dbsnp'.'alleles'.size()", equalTo(2))
                .body("[2].'dbsnp'.'alleles'[0].'allele'", containsString("C"))
                .body("[2].'dbsnp'.'alleles'[1].'allele'", containsString("T"))
                .body("[2].'dbsnp'.'alt'", containsString("G"))
                .body("[2].'dbsnp'.'chrom'", containsString("17"))
                .body("[2].'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[2].'dbsnp'.'flags'.size()", equalTo(1))
                .body("[2].'dbsnp'.'flags'[0]", nullValue())
                .body("[2].'dbsnp'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'dbsnp'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'dbsnp'.'ref'", containsString("C"))
                .body("[2].'dbsnp'.'rsid'", containsString("rs730882005"))
                .body("[2].'dbsnp'.'vartype'", containsString("snv"))
                .body("[2].'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[2].'cosmic'.'alt'", containsString("C"))
                .body("[2].'cosmic'.'chrom'", containsString("17"))
                .body("[2].'cosmic'.'cosmicId'", containsString("COSM44653"))
                .body("[2].'cosmic'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'cosmic'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'cosmic'.'mutFreq'", numberMatches(0.04))
                .body("[2].'cosmic'.'mutNt'", containsString("G>C"))
                .body("[2].'cosmic'.'ref'", containsString("G"))
                .body("[2].'cosmic'.'tumorSite'", containsString("central_nervous_system"))
                .body("[2].'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[2].'clinVar'.'alleleId'", numberMatches(363454.0))
                .body("[2].'clinVar'.'alt'", containsString("G"))
                .body("[2].'clinVar'.'chrom'", containsString("17"))
                .body("[2].'clinVar'.'cytogenic'", containsString("17p13.1"))
                .body("[2].'clinVar'.'gene'.'symbol'", containsString("TP53"))
                .body("[2].'clinVar'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'clinVar'.'hg19'.'end'", numberMatches(7577568.0))
                .body("[2].'clinVar'.'hg38'.'start'", containsString("7674250"))
                .body("[2].'clinVar'.'hg38'.'end'", containsString("7674250"))
                .body("[2].'clinVar'.'hgvs'.'coding'.size()", equalTo(16))
                .body("[2].'clinVar'.'hgvs'.'coding'", hasItems("LRG_321t1:c.713G>C", "NM_000546.6:c.713G>C", "NM_001126112.3:c.713G>C", "NM_001126113.3:c.713G>C", "NM_001126114.3:c.713G>C", "NM_001126115.2:c.317G>C", "NM_001126116.2:c.317G>C", "NM_001126117.2:c.317G>C", "NM_001126118.2:c.596G>C", "NM_001276695.3:c.596G>C", "NM_001276696.3:c.596G>C", "NM_001276697.3:c.236G>C", "NM_001276698.3:c.236G>C", "NM_001276699.3:c.236G>C", "NM_001276760.3:c.596G>C", "NM_001276761.3:c.596G>C"))
                .body("[2].'clinVar'.'hgvs'.'genomic'.size()", equalTo(4))
                .body("[2].'clinVar'.'hgvs'.'genomic'", hasItems("LRG_321:g.18301G>C", "NC_000017.10:g.7577568C>G", "NC_000017.11:g.7674250C>G", "NG_017013.2:g.18301G>C"))
                .body("[2].'clinVar'.'hgvs'.'protein'.size()", equalTo(17))
                .body("[2].'clinVar'.'hgvs'.'protein'", hasItems("LRG_321p1:p.Cys238Ser", "NP_000537.3:p.Cys238Ser", "NP_000537.3:p.Cys238Ser", "NP_001119584.1:p.Cys238Ser", "NP_001119585.1:p.Cys238Ser", "NP_001119586.1:p.Cys238Ser", "NP_001119587.1:p.Cys106Ser", "NP_001119588.1:p.Cys106Ser", "NP_001119589.1:p.Cys106Ser", "NP_001119590.1:p.Cys199Ser", "NP_001263624.1:p.Cys199Ser", "NP_001263625.1:p.Cys199Ser", "NP_001263626.1:p.Cys79Ser", "NP_001263627.1:p.Cys79Ser", "NP_001263628.1:p.Cys79Ser", "NP_001263689.1:p.Cys199Ser", "NP_001263690.1:p.Cys199Ser"))
                .body("[2].'clinVar'.'rcv'.size()", equalTo(21))
                .body("[2].'clinVar'.'rcv'[0].'accession'", containsString("RCV000419485"))
                .body("[2].'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[2].'clinVar'.'rcv'[0].'origin'", containsString("somatic"))
                .body("[2].'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000546.6(TP53):c.713G>C (p.Cys238Ser)"))
                .body("[2].'clinVar'.'rcv'[1].'accession'", containsString("RCV000418923"))
                .body("[2].'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[2].'clinVar'.'rcv'[1].'origin'", containsString("somatic"))
                .body("[2].'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000546.6(TP53):c.713G>C (p.Cys238Ser)"))
                .body("[2].'clinVar'.'rcv'[2].'accession'", containsString("RCV000423226"))
                .body("[2].'clinVar'.'rcv'[2].'clinicalSignificance'", containsString("Likely pathogenic"))
                .body("[2].'clinVar'.'rcv'[2].'origin'", containsString("somatic"))
                .body("[2].'clinVar'.'rcv'[2].'preferredName'", containsString("NM_000546.6(TP53):c.713G>C (p.Cys238Ser)"))
                // Skipping assertions on the remaining 18 elements. This limit of 3 elements can be increased in the configurations
                .body("[2].'clinVar'.'variantId'", numberMatches(376575.0))
                .body("[2].'mutdb'.'ref'", containsString("G"))
                .body("[2].'mutdb'.'alt'", containsString("C"))
                .body("[2].'mutdb'.'uniprotId'", containsString("VAR_045202"))
                .body("[2].'mutdb'.'mutpredScore'", numberMatches(0.977))
                .body("[2].'mutdb'.'cosmicId'", containsString("44653"))
                .body("[2].'mutdb'.'chrom'", containsString("17"))
                .body("[2].'mutdb'.'hg19'.'start'", numberMatches(7577568.0))
                .body("[2].'mutdb'.'hg19'.'end'", numberMatches(7577568.0))
                ; // Skipping assertions on the remaining 277 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_59() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/nucleotide_context/7:g.140453136A%3ET?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'molecule'", containsString("dna"))
                .body("'seq'", containsString("CAC"))
                .body("'hgvs'", containsString("7:g.140453136A>T"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_60() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/dbsnp/rs3?" + 
                    "isoformOverrideSource=ensembl&" + 
                    "token=_EM_15828_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("rs3"))
                .body("'originalVariantQuery'", containsString("rs3"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("13"))
                .body("'start'", numberMatches(3.2446842E7))
                .body("'end'", numberMatches(3.2446842E7))
                .body("'allele_string'", containsString("C/T"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("intron_variant"))
                .body("'transcript_consequences'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000428783"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000428783.1:n.99+25766C>T"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("EEF1DP3"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000229715"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("30486"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(2))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("intron_variant", "non_coding_transcript_variant"))
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_61() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/dbsnp/rs116035550?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("rs116035550"))
                .body("'originalVariantQuery'", containsString("rs116035550"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("11"))
                .body("'start'", numberMatches(212464.0))
                .body("'end'", numberMatches(212464.0))
                .body("'allele_string'", containsString("G/A/C"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("missense_variant"))
                .body("'transcript_consequences'.size()", equalTo(46))
                .body("'transcript_consequences'[0].'exon'", containsString("6/10"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000325207"))
                .body("'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000325941.5:p.Glu340Lys"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000325207.5:c.1018G>A"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[0].'codons'", containsString("Gaa/Aaa"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000325941"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(340.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(340.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("RIC8A"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000177963"))
                .body("'transcript_consequences'[0].'amino_acids'", containsString("E/K"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("29550"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'polyphen_score'", numberMatches(0.932))
                .body("'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_021932.4"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[1].'exon'", containsString("6/10"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000325207"))
                .body("'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000325941.5:p.Glu340Gln"))
                .body("'transcript_consequences'[1].'hgvsc'", containsString("ENST00000325207.5:c.1018G>C"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("C"))
                .body("'transcript_consequences'[1].'codons'", containsString("Gaa/Caa"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000325941"))
                .body("'transcript_consequences'[1].'protein_start'", numberMatches(340.0))
                .body("'transcript_consequences'[1].'protein_end'", numberMatches(340.0))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("RIC8A"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000177963"))
                .body("'transcript_consequences'[1].'amino_acids'", containsString("E/Q"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("29550"))
                .body("'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("'transcript_consequences'[1].'polyphen_score'", numberMatches(0.969))
                .body("'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[1].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_021932.4"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000382743"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000372191"))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("SIRT3"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000142082"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("14931"))
                .body("'transcript_consequences'[2].'canonical'", containsString("1"))
                .body("'transcript_consequences'[2].'refseq_transcript_ids'.size()", equalTo(2))
                .body("'transcript_consequences'[2].'refseq_transcript_ids'", hasItems("NM_012239.5", "NM_001017524.2"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("downstream_gene_variant"))
                // Skipping assertions on the remaining 43 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_62() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/MT:g.10360G%3EA?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("MT:g.10360G>A"))
                .body("'originalVariantQuery'", containsString("MT:g.10360G>A"))
                .body("'hgvsg'", containsString("MT:g.10360G>A"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("MT"))
                .body("'start'", numberMatches(10360.0))
                .body("'end'", numberMatches(10360.0))
                .body("'allele_string'", containsString("G/A"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("missense_variant"))
                .body("'transcript_consequences'.size()", equalTo(26))
                .body("'transcript_consequences'[0].'exon'", containsString("1/1"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000361227"))
                .body("'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000355206.2:p.Ser101Asn"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000361227.2:c.302G>A"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[0].'codons'", containsString("aGt/aAt"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000355206"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(101.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(101.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("MT-ND3"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000198840"))
                .body("'transcript_consequences'[0].'amino_acids'", containsString("S/N"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("7458"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'polyphen_score'", numberMatches(0.569))
                .body("'transcript_consequences'[0].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[0].'sift_prediction'", containsString("deleterious_low_confidence"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000361335"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000354728"))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("MT-ND4L"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000212907"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("7460"))
                .body("'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("upstream_gene_variant"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000361381"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000354961"))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("MT-ND4"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000198886"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("7459"))
                .body("'transcript_consequences'[2].'canonical'", containsString("1"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("upstream_gene_variant"))
                // Skipping assertions on the remaining 23 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_63() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/dbsnp/rs3?" + 
                    "isoformOverrideSource=mskcc&" + 
                    "token=_EM_15703_XYZ_&" + 
                    "fields=PxXQYO&fields=xXsT&fields=_EM_15596_XYZ_&fields=_EM_15597_XYZ_&fields=_EM_15598_XYZ_")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("rs3"))
                .body("'originalVariantQuery'", containsString("rs3"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("13"))
                .body("'start'", numberMatches(3.2446842E7))
                .body("'end'", numberMatches(3.2446842E7))
                .body("'allele_string'", containsString("C/T"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("intron_variant"))
                .body("'transcript_consequences'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000428783"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000428783.1:n.99+25766C>T"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("EEF1DP3"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000229715"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("30486"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(2))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("intron_variant", "non_coding_transcript_variant"))
                .body("'successfully_annotated'", equalTo(true));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_64() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"\", " + 
                    " \"start\": 122, " + 
                    " \"end\": 320, " + 
                    " \"referenceAllele\": \"_\", " + 
                    " \"variantAllele\": \"_EM_21236_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"vC_DPO5Ew1_ieE\", " + 
                    " \"start\": 813, " + 
                    " \"end\": 334, " + 
                    " \"referenceAllele\": \"_EM_21237_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_21238_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_21239_XYZ_\", " + 
                    " \"start\": 996, " + 
                    " \"end\": 157, " + 
                    " \"referenceAllele\": \"_EM_21240_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_21241_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(3))
                .body("[0].'genomicLocation'.'chromosome'", containsString(""))
                .body("[0].'genomicLocation'.'start'", numberMatches(122.0))
                .body("[0].'genomicLocation'.'end'", numberMatches(320.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("_"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("_EM_21236_XYZ_"))
                .body("[0].'variant'", containsString(":g.122_123insEM_21236_XYZ_"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("vC_DPO5Ew1_ieE"))
                .body("[1].'genomicLocation'.'start'", numberMatches(813.0))
                .body("[1].'genomicLocation'.'end'", numberMatches(334.0))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("_EM_21237_XYZ_"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("_EM_21238_XYZ_"))
                .body("[1].'variant'", containsString("vC_DPO5Ew1_ieE:g.821_334delins8_XYZ_"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("_EM_21239_XYZ_"))
                .body("[2].'genomicLocation'.'start'", numberMatches(996.0))
                .body("[2].'genomicLocation'.'end'", numberMatches(157.0))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("_EM_21240_XYZ_"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("_EM_21241_XYZ_"))
                .body("[2].'variant'", containsString("_EM_21X9_XYZ_:g.1004_157delins1_XYZ_"))
                .body("[2].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_65() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"6\", " + 
                    " \"start\": 140453136, " + 
                    " \"end\": 140584272, " + 
                    " \"referenceAllele\": \"A\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=_EM_37152_XYZ_&" + 
                    "token=_EM_37153_XYZ_&" + 
                    "fields=annotation_summar&fields=nucleotide_context")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("6:g.140453136del"))
                .body("[0].'originalVariantQuery'", containsString("6,140453136,140584272,A,-"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_66() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 512, " + 
                    " \"end\": 0, " + 
                    " \"referenceAllele\": \"-w\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(512.0))
                .body("[0].'genomicLocation'.'end'", numberMatches(0.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("-w"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'variant'", containsString("7:g.513del"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_67() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"3\", " + 
                    " \"start\": 64, " + 
                    " \"end\": 8, " + 
                    " \"referenceAllele\": \"\", " + 
                    " \"variantAllele\": \"+\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"`EspF:Cbx\", " + 
                    " \"start\": 4194680, " + 
                    " \"end\": 67633508, " + 
                    " \"referenceAllele\": \"_MCc29NA\", " + 
                    " \"variantAllele\": \"_ECc29NA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'genomicLocation'.'chromosome'", containsString("3"))
                .body("[0].'genomicLocation'.'start'", numberMatches(64.0))
                .body("[0].'genomicLocation'.'end'", numberMatches(8.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString(""))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("+"))
                .body("[0].'variant'", containsString("3:g.64_65ins+"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("`EspF:Cbx"))
                .body("[1].'genomicLocation'.'start'", numberMatches(4194680.0))
                .body("[1].'genomicLocation'.'end'", numberMatches(6.7633508E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("_MCc29NA"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("_ECc29NA"))
                .body("[1].'variant'", containsString("`EspF:Cbx:g.4194681_67633508delinsECc29NA"))
                .body("[1].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_68() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1913_XYZ_\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 32, " + 
                    " \"referenceAllele\": \"NA\", " + 
                    " \"variantAllele\": \"j\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"s5L\", " + 
                    " \"start\": 39977372, " + 
                    " \"end\": 0, " + 
                    " \"referenceAllele\": \"_EM_1411_XYZ_\", " + 
                    " \"variantAllele\": \"NA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 4, " + 
                    " \"end\": 37880050, " + 
                    " \"referenceAllele\": \"_EM_1412_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1413_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1415_XYZ_\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"_EM_1416_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1914_XYZ_\", " + 
                    " \"start\": 89660172, " + 
                    " \"end\": -174548212, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(5))
                .body("[0].'genomicLocation'.'chromosome'", containsString("_EM_1913_XYZ_"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(32.0))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("NA"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("j"))
                .body("[0].'variant'", containsString("_EM_1913_XYZ_:g.140453193_140453194insj"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("s5L"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.9977372E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(0.0))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("_EM_1411_XYZ_"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("NA"))
                .body("[1].'variant'", containsString("s5L:g.39977372_0del"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(4.0))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.788005E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("_EM_1412_XYZ_"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("_EM_1413_XYZ_"))
                .body("[2].'variant'", containsString("17:g.11_37880050delins3_XYZ_"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 2 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_69() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/cancer_hotspots/genomic/19,10267171,10267171,T,C?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_70() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"7:g.140453136A\u003eT\", " + 
                    " \"12:g.25398285C\u003eA\", " + 
                    " \"17:g.41242962_41242963insGA\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/nucleotide_context?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'molecule'", containsString("dna"))
                .body("[0].'seq'", containsString("CAC"))
                .body("[0].'hgvs'", containsString("7:g.140453136A>T"))
                .body("[1].'molecule'", containsString("dna"))
                .body("[1].'seq'", containsString("CCA"))
                .body("[1].'hgvs'", containsString("12:g.25398285C>A"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_71() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"_EM_568_XYZ_\", " + 
                    " \"start\": 25398282, " + 
                    " \"end\": 25365516, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25529355, " + 
                    " \"end\": 25398284, " + 
                    " \"referenceAllele\": \"AC\", " + 
                    " \"variantAllele\": \"_EM_569_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("_EM_568_XYZ_:g.25398282C>A"))
                .body("[0].'originalVariantQuery'", containsString("_EM_568_XYZ_,25398282,25365516,C,A"))
                .body("[0].'successfully_annotated'", equalTo(false))
                .body("[1].'variant'", containsString("12:g.25529355_25398284delins_EM_569_XYZ_"))
                .body("[1].'originalVariantQuery'", containsString("12,25529355,25398284,AC,_EM_569_XYZ_"))
                .body("[1].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_72() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 55242467, " + 
                    " \"end\": 55242480, " + 
                    " \"referenceAllele\": \"GAATTAAGAGAAGCA\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_601_XYZ_\", " + 
                    " \"start\": -32, " + 
                    " \"end\": 55249003, " + 
                    " \"referenceAllele\": \"\", " + 
                    " \"variantAllele\": \"CAGCGTGGAV\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(5.5242467E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(5.524248E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("GAATTAAGAGAAGCA"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'variant'", containsString("7:g.55242467_55242480del"))
                .body("[0].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_73() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 55242338, " + 
                    " \"end\": 55242480, " + 
                    " \"referenceAllele\": \"GAATTAAGAGAAGCA\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_601_XYZ_\", " + 
                    " \"start\": 55249002, " + 
                    " \"end\": 55249003, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"CAGCGTGGA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(5.5242338E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(5.524248E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("GAATTAAGAGAAGCA"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'variant'", containsString("7:g.55242338_55242480del"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("_EM_601_XYZ_"))
                .body("[1].'genomicLocation'.'start'", numberMatches(5.5249002E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(5.5249003E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("-"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("CAGCGTGGA"))
                .body("[1].'variant'", containsString("_EM_601_XYZ_:g.55249002_55249003insCAGCGTGGA"))
                .body("[1].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_74() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 55242466, " + 
                    " \"end\": 55242480, " + 
                    " \"referenceAllele\": \"GAATTAAGAGAAGCA\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 55249002, " + 
                    " \"end\": 55249003, " + 
                    " \"referenceAllele\": \"-\", " + 
                    " \"variantAllele\": \"CAGCGTGGA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(5.5242466E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(5.524248E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("GAATTAAGAGAAGCA"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("-"))
                .body("[0].'variant'", containsString("7:g.55242466_55242480del"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[1].'genomicLocation'.'start'", numberMatches(5.5249002E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(5.5249003E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("-"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("CAGCGTGGA"))
                .body("[1].'variant'", containsString("7:g.55249002_55249003insCAGCGTGGA"))
                .body("[1].'hotspots'.size()", equalTo(0));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_75() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37880220, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879794, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("7:g.140453193T>C"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[1].'variant'", containsString("17:g.37880220T>C"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[2].'variant'", containsString("17:g.37879794G>C"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_76() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/summary/9:g.21994291C%3ET?" + 
                    "isoformOverrideSource=_EM_1199_XYZ_&" + 
                    "projection=CANONICAL&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("9:g.21994291C>T"))
                .body("'genomicLocation'.'chromosome'", containsString("9"))
                .body("'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("'genomicLocation'.'variantAllele'", containsString("T"))
                .body("'strandSign'", containsString("+"))
                .body("'variantType'", containsString("SNP"))
                .body("'assemblyName'", containsString("GRCh37"))
                .body("'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequences'.size()", equalTo(1))
                .body("'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequences'[0].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequences'[0].'variantClassification'", containsString("5'Flank"))
                .body("'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_77() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/summary/9:g.21994291C%3ET?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("9:g.21994291C>T"))
                .body("'genomicLocation'.'chromosome'", containsString("9"))
                .body("'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("'genomicLocation'.'variantAllele'", containsString("T"))
                .body("'strandSign'", containsString("+"))
                .body("'variantType'", containsString("SNP"))
                .body("'assemblyName'", containsString("GRCh37"))
                .body("'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequences'.size()", equalTo(23))
                .body("'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("'transcriptConsequences'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("'transcriptConsequences'[0].'aminoAcids'", containsString("A/T"))
                .body("'transcriptConsequences'[0].'aminoAcidRef'", containsString("A"))
                .body("'transcriptConsequences'[0].'aminoAcidAlt'", containsString("T"))
                .body("'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("'transcriptConsequences'[0].'hgvspShort'", containsString("p.A55T"))
                .body("'transcriptConsequences'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("'transcriptConsequences'[0].'refSeq'", containsString("NM_058195.3"))
                .body("'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'transcriptConsequences'[0].'exon'", containsString("1/3"))
                .body("'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.014))
                .body("'transcriptConsequences'[0].'polyphenPrediction'", containsString("benign"))
                .body("'transcriptConsequences'[0].'siftScore'", numberMatches(0.09))
                .body("'transcriptConsequences'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("'transcriptConsequences'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequences'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequences'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequences'[1].'variantClassification'", containsString("5'Flank"))
                .body("'transcriptConsequences'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("'transcriptConsequences'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequences'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequences'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("'transcriptConsequenceSummaries'.size()", equalTo(23))
                .body("'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("'transcriptConsequenceSummaries'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("A/T"))
                .body("'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("A"))
                .body("'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("T"))
                .body("'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.A55T"))
                .body("'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_058195.3"))
                .body("'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'transcriptConsequenceSummaries'[0].'exon'", containsString("1/3"))
                .body("'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.014))
                .body("'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("benign"))
                .body("'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.09))
                .body("'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("5'Flank"))
                .body("'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_78() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"9:g.21994291C\u003eT\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?" + 
                    "isoformOverrideSource=_EM_1209_XYZ_&" + 
                    "projection=CANONICAL&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("9:g.21994291C>T"))
                .body("[0].'genomicLocation'.'chromosome'", containsString("9"))
                .body("[0].'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'strandSign'", containsString("+"))
                .body("[0].'variantType'", containsString("SNP"))
                .body("[0].'assemblyName'", containsString("GRCh37"))
                .body("[0].'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'[0].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[0].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("transcriptConsequenceSummaries", "transcriptConsequenceSummary", "transcriptConsequences", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_79() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .get(baseUrlOfSut + "/annotation/genomic/7,140453136,140493136,A,TM?" + 
                    "isoformOverrideSource=hnd&" + 
                    "token=Qjg&" + 
                    "fields=annotation_summary")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("7:g.140453136delinsTM"))
                .body("'originalVariantQuery'", containsString("7,140453136,140493136,A,TM"))
                .body("'hgvsg'", containsString("7:g.140453136delinsTM"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("7"))
                .body("'start'", numberMatches(1.40453136E8))
                .body("'end'", numberMatches(1.40453136E8))
                .body("'allele_string'", containsString("A/TM"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("frameshift_variant"))
                .body("'transcript_consequences'.size()", equalTo(4))
                .body("'transcript_consequences'[0].'exon'", containsString("15/18"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000288602"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("TM"))
                .body("'transcript_consequences'[0].'codons'", containsString("gTg/gKAg"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000288602"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_004333.4"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("frameshift_variant"))
                .body("'transcript_consequences'[1].'exon'", containsString("2/6"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000479537"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("TM"))
                .body("'transcript_consequences'[1].'codons'", containsString("gTg/gKAg"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000418033"))
                .body("'transcript_consequences'[1].'protein_start'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'protein_end'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(2))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("frameshift_variant", "NMD_transcript_variant"))
                .body("'transcript_consequences'[2].'exon'", containsString("6/10"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000496384"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("TM"))
                .body("'transcript_consequences'[2].'codons'", containsString("gTg/gKAg"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000419060"))
                .body("'transcript_consequences'[2].'protein_start'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'protein_end'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("frameshift_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true))
                .body("'annotation_summary'.'variant'", containsString("7:g.140453136delinsTM"))
                .body("'annotation_summary'.'genomicLocation'.'chromosome'", containsString("7"))
                .body("'annotation_summary'.'genomicLocation'.'start'", numberMatches(1.40453136E8))
                .body("'annotation_summary'.'genomicLocation'.'end'", numberMatches(1.40453136E8))
                .body("'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("A"))
                .body("'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("TM"))
                .body("'annotation_summary'.'strandSign'", containsString("+"))
                .body("'annotation_summary'.'variantType'", containsString("INS"))
                .body("'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gTg/gKAg"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("frameshift_variant"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Frame_Shift_Ins"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gTg/gKAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("frameshift_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Frame_Shift_Ins"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000479537"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gTg/gKAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("frameshift_variant,NMD_transcript_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(28.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(28.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Frame_Shift_Ins"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/6"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000496384"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gTg/gKAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("frameshift_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(208.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(208.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Frame_Shift_Ins"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("6/10"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gTg/gKAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("frameshift_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Frame_Shift_Ins"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("15/18"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_80() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_522_XYZ_\", " + 
                    " \"start\": 39977372, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"_EM_523_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879794, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25380271, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("7:g.140453193T>C"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("_EM_522_XYZ_"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.9977372E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("_EM_523_XYZ_"))
                .body("[1].'variant'", containsString("_EM_522_XYZ_:g.39977372delins_EM_523_XYZ_"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[2].'variant'", containsString("17:g.37879794G>C"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_81() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " \"9:g.21994291C\u003eT\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?" + 
                    "isoformOverrideSource=genome_nexus&" + 
                    "projection=ALL&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("9:g.21994291C>T"))
                .body("[0].'genomicLocation'.'chromosome'", containsString("9"))
                .body("[0].'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'strandSign'", containsString("+"))
                .body("[0].'variantType'", containsString("SNP"))
                .body("[0].'assemblyName'", containsString("GRCh37"))
                .body("[0].'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'.size()", equalTo(23))
                .body("[0].'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("[0].'transcriptConsequences'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("[0].'transcriptConsequences'[0].'aminoAcids'", containsString("A/T"))
                .body("[0].'transcriptConsequences'[0].'aminoAcidRef'", containsString("A"))
                .body("[0].'transcriptConsequences'[0].'aminoAcidAlt'", containsString("T"))
                .body("[0].'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("[0].'transcriptConsequences'[0].'hgvspShort'", containsString("p.A55T"))
                .body("[0].'transcriptConsequences'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("[0].'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("[0].'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("[0].'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("[0].'transcriptConsequences'[0].'refSeq'", containsString("NM_058195.3"))
                .body("[0].'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'transcriptConsequences'[0].'exon'", containsString("1/3"))
                .body("[0].'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.014))
                .body("[0].'transcriptConsequences'[0].'polyphenPrediction'", containsString("benign"))
                .body("[0].'transcriptConsequences'[0].'siftScore'", numberMatches(0.09))
                .body("[0].'transcriptConsequences'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("[0].'transcriptConsequences'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[1].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequences'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("[0].'transcriptConsequences'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'transcriptConsequenceSummaries'.size()", equalTo(23))
                .body("[0].'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("[0].'transcriptConsequenceSummaries'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("A/T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.A55T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("[0].'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("[0].'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_058195.3"))
                .body("[0].'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'transcriptConsequenceSummaries'[0].'exon'", containsString("1/3"))
                .body("[0].'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.014))
                .body("[0].'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("benign"))
                .body("[0].'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.09))
                .body("[0].'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("[0].'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("[0].'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_82() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " \"9:g.21994291C\u003eT\" " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/summary?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("9:g.21994291C>T"))
                .body("[0].'genomicLocation'.'chromosome'", containsString("9"))
                .body("[0].'genomicLocation'.'start'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'end'", numberMatches(2.1994291E7))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'strandSign'", containsString("+"))
                .body("[0].'variantType'", containsString("SNP"))
                .body("[0].'assemblyName'", containsString("GRCh37"))
                .body("[0].'canonicalTranscriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'.size()", equalTo(23))
                .body("[0].'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("[0].'transcriptConsequences'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("[0].'transcriptConsequences'[0].'aminoAcids'", containsString("A/T"))
                .body("[0].'transcriptConsequences'[0].'aminoAcidRef'", containsString("A"))
                .body("[0].'transcriptConsequences'[0].'aminoAcidAlt'", containsString("T"))
                .body("[0].'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("[0].'transcriptConsequences'[0].'hgvspShort'", containsString("p.A55T"))
                .body("[0].'transcriptConsequences'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("[0].'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("[0].'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("[0].'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("[0].'transcriptConsequences'[0].'refSeq'", containsString("NM_058195.3"))
                .body("[0].'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'transcriptConsequences'[0].'exon'", containsString("1/3"))
                .body("[0].'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.014))
                .body("[0].'transcriptConsequences'[0].'polyphenPrediction'", containsString("benign"))
                .body("[0].'transcriptConsequences'[0].'siftScore'", numberMatches(0.09))
                .body("[0].'transcriptConsequences'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("[0].'transcriptConsequences'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequences'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[1].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequences'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("[0].'transcriptConsequences'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequences'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequences'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'transcriptConsequenceSummaries'.size()", equalTo(23))
                .body("[0].'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000361570"))
                .body("[0].'transcriptConsequenceSummaries'[0].'codonChange'", containsString("Gcg/Acg"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("A/T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("CDKN2A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.A55T"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Ala55Thr"))
                .body("[0].'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000361570.3:c.163G>A"))
                .body("[0].'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(55.0))
                .body("[0].'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(55.0))
                .body("[0].'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_058195.3"))
                .body("[0].'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'transcriptConsequenceSummaries'[0].'exon'", containsString("1/3"))
                .body("[0].'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.014))
                .body("[0].'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("benign"))
                .body("[0].'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.09))
                .body("[0].'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("tolerated_low_confidence"))
                .body("[0].'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("5'Flank"))
                .body("[0].'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000455933"))
                .body("[0].'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("5'Flank"))
                // Skipping assertions on the remaining 20 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000428597"))
                .body("[0].'transcriptConsequenceSummary'.'consequenceTerms'", containsString("upstream_gene_variant"))
                .body("[0].'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("CDKN2B-AS1"))
                .body("[0].'transcriptConsequenceSummary'.'variantClassification'", containsString("5'Flank"));
        
    }
    
    
    @Test(timeout = 60000)
    public void test_83() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"kTgS\", " + 
                    " \"start\": 40108444, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"_EM_1818_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_523_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879794, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1819_XYZ_\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": -511472627, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"_EM_1820_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25363887, " + 
                    " \"end\": 25380270, " + 
                    " \"referenceAllele\": \"_EM_1821_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1822_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692940, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic?EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("7:g.140453193T>C"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("kTgS"))
                .body("[1].'genomicLocation'.'start'", numberMatches(4.0108444E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("_EM_1818_XYZ_"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("_EM_523_XYZ_"))
                .body("[1].'variant'", containsString("kTgS:g.40108448_37880220delins523_XYZ_"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[2].'variant'", containsString("17:g.37879794G>C"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_84() throws Exception {
        
        given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453193, " + 
                    " \"end\": 140453193, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37880220, " + 
                    " \"end\": 37880220, " + 
                    " \"referenceAllele\": \"T\", " + 
                    " \"variantAllele\": \"_EM_1944_XYZ_\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"17\", " + 
                    " \"start\": 37879794, " + 
                    " \"end\": 37879794, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"C\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": 25398285, " + 
                    " \"referenceAllele\": \"CC\", " + 
                    " \"variantAllele\": \"GA\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1945_XYZ_\", " + 
                    " \"start\": 25380271, " + 
                    " \"end\": 25379759, " + 
                    " \"referenceAllele\": \"_EM_1946_XYZ_\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"_EM_1947_XYZ_\", " + 
                    " \"start\": 89692940, " + 
                    " \"end\": 89692972, " + 
                    " \"referenceAllele\": \"_EM_1948_XYZ_\", " + 
                    " \"variantAllele\": \"_EM_1949_XYZ_\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/cancer_hotspots/genomic")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(6))
                .body("[0].'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'genomicLocation'.'start'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'end'", numberMatches(1.40453193E8))
                .body("[0].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[0].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[0].'variant'", containsString("7:g.140453193T>C"))
                .body("[0].'hotspots'.size()", equalTo(0))
                .body("[1].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[1].'genomicLocation'.'start'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'end'", numberMatches(3.788022E7))
                .body("[1].'genomicLocation'.'referenceAllele'", containsString("T"))
                .body("[1].'genomicLocation'.'variantAllele'", containsString("_EM_1944_XYZ_"))
                .body("[1].'variant'", containsString("17:g.37880220delins_EM_1944_XYZ_"))
                .body("[1].'hotspots'.size()", equalTo(0))
                .body("[2].'genomicLocation'.'chromosome'", containsString("17"))
                .body("[2].'genomicLocation'.'start'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'end'", numberMatches(3.7879794E7))
                .body("[2].'genomicLocation'.'referenceAllele'", containsString("G"))
                .body("[2].'genomicLocation'.'variantAllele'", containsString("C"))
                .body("[2].'variant'", containsString("17:g.37879794G>C"))
                .body("[2].'hotspots'.size()", equalTo(0))
                ; // Skipping assertions on the remaining 3 elements. This limit of 3 elements can be increased in the configurations
        
    }
    
    
    @Test(timeout = 60000)
    public void test_85() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .get(baseUrlOfSut + "/annotation/genomic/7,140453136,140453136,A,T?" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("'variant'", containsString("7:g.140453136A>T"))
                .body("'originalVariantQuery'", containsString("7,140453136,140453136,A,T"))
                .body("'hgvsg'", containsString("7:g.140453136A>T"))
                .body("'assembly_name'", containsString("GRCh37"))
                .body("'seq_region_name'", containsString("7"))
                .body("'start'", numberMatches(1.40453136E8))
                .body("'end'", numberMatches(1.40453136E8))
                .body("'allele_string'", containsString("A/T"))
                .body("'strand'", numberMatches(1.0))
                .body("'most_severe_consequence'", containsString("missense_variant"))
                .body("'transcript_consequences'.size()", equalTo(4))
                .body("'transcript_consequences'[0].'exon'", containsString("15/18"))
                .body("'transcript_consequences'[0].'transcript_id'", containsString("ENST00000288602"))
                .body("'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000288602.6:p.Val600Glu"))
                .body("'transcript_consequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[0].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[0].'protein_id'", containsString("ENSP00000288602"))
                .body("'transcript_consequences'[0].'protein_start'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'protein_end'", numberMatches(600.0))
                .body("'transcript_consequences'[0].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[0].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[0].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[0].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("'transcript_consequences'[0].'polyphen_score'", numberMatches(0.963))
                .body("'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_004333.4"))
                .body("'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("'transcript_consequences'[1].'exon'", containsString("2/6"))
                .body("'transcript_consequences'[1].'transcript_id'", containsString("ENST00000479537"))
                .body("'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000418033.1:p.Val28Glu"))
                .body("'transcript_consequences'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[1].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[1].'protein_id'", containsString("ENSP00000418033"))
                .body("'transcript_consequences'[1].'protein_start'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'protein_end'", numberMatches(28.0))
                .body("'transcript_consequences'[1].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[1].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[1].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[1].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[1].'polyphen_score'", numberMatches(0.991))
                .body("'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("'transcript_consequences'[1].'sift_score'", numberMatches(0.01))
                .body("'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[1].'consequence_terms'.size()", equalTo(2))
                .body("'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant", "NMD_transcript_variant"))
                .body("'transcript_consequences'[2].'exon'", containsString("6/10"))
                .body("'transcript_consequences'[2].'transcript_id'", containsString("ENST00000496384"))
                .body("'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000419060.1:p.Val208Glu"))
                .body("'transcript_consequences'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("'transcript_consequences'[2].'codons'", containsString("gTg/gAg"))
                .body("'transcript_consequences'[2].'protein_id'", containsString("ENSP00000419060"))
                .body("'transcript_consequences'[2].'protein_start'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'protein_end'", numberMatches(208.0))
                .body("'transcript_consequences'[2].'gene_symbol'", containsString("BRAF"))
                .body("'transcript_consequences'[2].'gene_id'", containsString("ENSG00000157764"))
                .body("'transcript_consequences'[2].'amino_acids'", containsString("V/E"))
                .body("'transcript_consequences'[2].'hgnc_id'", containsString("1097"))
                .body("'transcript_consequences'[2].'polyphen_score'", numberMatches(0.817))
                .body("'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'successfully_annotated'", equalTo(true))
                .body("'nucleotide_context'.'license'", nullValue())
                .body("'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("'nucleotide_context'.'annotation'.'seq'", containsString("CAC"))
                .body("'nucleotide_context'.'annotation'.'hgvs'", containsString("7:g.140453136A>T"))
                .body("'annotation_summary'.'variant'", containsString("7:g.140453136A>T"))
                .body("'annotation_summary'.'genomicLocation'.'chromosome'", containsString("7"))
                .body("'annotation_summary'.'genomicLocation'.'start'", numberMatches(1.40453136E8))
                .body("'annotation_summary'.'genomicLocation'.'end'", numberMatches(1.40453136E8))
                .body("'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("A"))
                .body("'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("T"))
                .body("'annotation_summary'.'strandSign'", containsString("+"))
                .body("'annotation_summary'.'variantType'", containsString("SNP"))
                .body("'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.V600E"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.963))
                .body("'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.V600E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.963))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000479537"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant,NMD_transcript_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.V28E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Val28Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(28.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(28.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/6"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.991))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.01))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000496384"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.V208E"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Val208Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(208.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(208.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("6/10"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.817))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000288602"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gTg/gAg"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("V/E"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("V"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("E"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("BRAF"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.V600E"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Val600Glu"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(600.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_004333.4"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("15/18"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.963))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) res_0.extract().response().jsonPath().getJsonObject("")).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_86() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"4\", " + 
                    " \"start\": 55152095, " + 
                    " \"end\": 55152106, " + 
                    " \"referenceAllele\": \"ATCATGCATGATT\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=fmo&" + 
                    "token=fmo&" + 
                    "fields=annotation_summary&fields=nucleotide_context")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("4:g.55152096_55152106del"))
                .body("[0].'originalVariantQuery'", containsString("4,55152095,55152106,ATCATGCATGATT,A"))
                .body("[0].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_87() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"7\", " + 
                    " \"start\": 140453136, " + 
                    " \"end\": 140453136, " + 
                    " \"referenceAllele\": \"A\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "isoformOverrideSource=foo&" + 
                    "token=foo&" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(1))
                .body("[0].'variant'", containsString("7:g.140453136A>T"))
                .body("[0].'originalVariantQuery'", containsString("7,140453136,140453136,A,T"))
                .body("[0].'hgvsg'", containsString("7:g.140453136A>T"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("7"))
                .body("[0].'start'", numberMatches(1.40453136E8))
                .body("[0].'end'", numberMatches(1.40453136E8))
                .body("[0].'allele_string'", containsString("A/T"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("missense_variant"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("15/18"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000288602"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000288602.6:p.Val600Glu"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("gTg/gAg"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000288602"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(600.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(600.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("BRAF"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000157764"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("V/E"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("1097"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.963))
                .body("[0].'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_004333.4"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[1].'exon'", containsString("2/6"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000479537"))
                .body("[0].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000418033.1:p.Val28Glu"))
                .body("[0].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[1].'codons'", containsString("gTg/gAg"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000418033"))
                .body("[0].'transcript_consequences'[1].'protein_start'", numberMatches(28.0))
                .body("[0].'transcript_consequences'[1].'protein_end'", numberMatches(28.0))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("BRAF"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000157764"))
                .body("[0].'transcript_consequences'[1].'amino_acids'", containsString("V/E"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("1097"))
                .body("[0].'transcript_consequences'[1].'polyphen_score'", numberMatches(0.991))
                .body("[0].'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[1].'sift_score'", numberMatches(0.01))
                .body("[0].'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(2))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant", "NMD_transcript_variant"))
                .body("[0].'transcript_consequences'[2].'exon'", containsString("6/10"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000496384"))
                .body("[0].'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000419060.1:p.Val208Glu"))
                .body("[0].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[2].'codons'", containsString("gTg/gAg"))
                .body("[0].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000419060"))
                .body("[0].'transcript_consequences'[2].'protein_start'", numberMatches(208.0))
                .body("[0].'transcript_consequences'[2].'protein_end'", numberMatches(208.0))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("BRAF"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000157764"))
                .body("[0].'transcript_consequences'[2].'amino_acids'", containsString("V/E"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("1097"))
                .body("[0].'transcript_consequences'[2].'polyphen_score'", numberMatches(0.817))
                .body("[0].'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[0].'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'nucleotide_context'.'license'", nullValue())
                .body("[0].'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("[0].'nucleotide_context'.'annotation'.'seq'", containsString("CAC"))
                .body("[0].'nucleotide_context'.'annotation'.'hgvs'", containsString("7:g.140453136A>T"))
                .body("[0].'annotation_summary'.'variant'", containsString("7:g.140453136A>T"))
                .body("[0].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("7"))
                .body("[0].'annotation_summary'.'genomicLocation'.'start'", numberMatches(1.40453136E8))
                .body("[0].'annotation_summary'.'genomicLocation'.'end'", numberMatches(1.40453136E8))
                .body("[0].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("A"))
                .body("[0].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("T"))
                .body("[0].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[0].'annotation_summary'.'variantType'", containsString("SNP"))
                .body("[0].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[0].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.V600E"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_004333.4"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("15/18"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.963))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.V600E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Val600Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_004333.4"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("15/18"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.963))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000479537"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant,NMD_transcript_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.V28E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Val28Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000479537.1:c.83T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(28.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(28.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.991))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.01))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000496384"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.V208E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Val208Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000496384.2:c.623T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(208.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(208.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("6/10"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.817))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000288602"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gTg/gAg"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("V/E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("BRAF"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.V600E"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Val600Glu"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000288602.6:c.1799T>A"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(600.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_004333.4"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("15/18"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.963))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_88() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"10\", " + 
                    " \"start\": 89624245, " + 
                    " \"end\": 89624245, " + 
                    " \"referenceAllele\": \"G\", " + 
                    " \"variantAllele\": \"T\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"10.0\", " + 
                    " \"start\": 89624242, " + 
                    " \"end\": 89624243, " + 
                    " \"referenceAllele\": \"AA\", " + 
                    " \"variantAllele\": \"-\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "fields=my_variant_info&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("10:g.89624245G>T"))
                .body("[0].'originalVariantQuery'", containsString("10,89624245,89624245,G,T"))
                .body("[0].'hgvsg'", containsString("10:g.89624245G>T"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("10"))
                .body("[0].'start'", numberMatches(8.9624245E7))
                .body("[0].'end'", numberMatches(8.9624245E7))
                .body("[0].'allele_string'", containsString("G/T"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("stop_gained"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("1/9"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000371953"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000361021.3:p.Glu7Ter"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000371953.3:c.19G>T"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("Gag/Tag"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000361021"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(7.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(7.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("PTEN"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000171862"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("E/*"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("9588"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_000314.4"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("stop_gained"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000445946"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000392204"))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("KLLN"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000227268"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("37212"))
                .body("[0].'transcript_consequences'[1].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_001126049.1"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("upstream_gene_variant"))
                .body("[0].'transcript_consequences'[2].'exon'", containsString("1/2"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000462694"))
                .body("[0].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000462694.1:n.21G>T"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("T"))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("PTEN"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000171862"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("9588"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("non_coding_transcript_exon_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'my_variant_info'.'license'", nullValue())
                .body("[0].'my_variant_info'.'annotation'.'variant'", containsString("chr10:g.89624245G>T"))
                .body("[0].'my_variant_info'.'annotation'.'query'", containsString("chr10:g.89624245G>T"))
                .body("[0].'my_variant_info'.'annotation'.'hgvs'", containsString("10:g.89624245G>T"))
                .body("[0].'my_variant_info'.'annotation'.'snpeff'.'license'", containsString("http://bit.ly/2suyRKt"))
                .body("[0].'my_variant_info'.'annotation'.'vcf'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'vcf'.'position'", containsString("89624245"))
                .body("[0].'my_variant_info'.'annotation'.'vcf'.'ref'", containsString("G"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'license'", containsString("http://bit.ly/2AqoLOc"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'alleles'.size()", equalTo(1))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'alleles'[0]", nullValue())
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'chrom'", containsString("10"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'dbsnpBuild'", numberMatches(156.0))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'flags'.size()", equalTo(1))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'flags'[0]", nullValue())
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'hg19'.'start'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'hg19'.'end'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'ref'", containsString("G"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'rsid'", containsString("rs1554890335"))
                .body("[0].'my_variant_info'.'annotation'.'dbsnp'.'vartype'", containsString("snv"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'license'", containsString("http://bit.ly/2VMkY7R"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'chrom'", containsString("10"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'cosmicId'", containsString("COSM5298"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'hg19'.'start'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'hg19'.'end'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'mutFreq'", numberMatches(0.19))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'mutNt'", containsString("G>T"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'ref'", containsString("G"))
                .body("[0].'my_variant_info'.'annotation'.'cosmic'.'tumorSite'", containsString("large_intestine"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'license'", containsString("http://bit.ly/2SQdcI0"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'alleleId'", numberMatches(486042.0))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'alt'", containsString("T"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'chrom'", containsString("10"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'cytogenic'", containsString("10q23.31"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'gene'.'symbol'", containsString("PTEN"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg19'.'start'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg19'.'end'", numberMatches(8.9624245E7))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg38'.'start'", containsString("87864488"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hg38'.'end'", containsString("87864488"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'coding'.size()", equalTo(3))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'coding'", hasItems("NM_000314.8:c.19G>T", "NM_001304717.5:c.538G>T", "NM_001304718.2:c.-687G>T"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'genomic'.size()", equalTo(6))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'genomic'", hasItems("LRG_1087:g.3950C>A", "LRG_311:g.6050G>T", "NC_000010.10:g.89624245G>T", "NC_000010.11:g.87864488G>T", "NG_007466.2:g.6050G>T", "NG_033079.1:g.3950C>A"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'protein'.size()", equalTo(3))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'hgvs'.'protein'", hasItems("", "NP_000305.3:p.Glu7Ter", "NP_001291646.4:p.Glu180Ter"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'.size()", equalTo(2))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'accession'", containsString("RCV000585486"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'origin'", containsString("germline"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[0].'preferredName'", containsString("NM_000314.8(PTEN):c.19G>T (p.Glu7Ter)"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'accession'", containsString("RCV003451323"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'clinicalSignificance'", containsString("Pathogenic"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'origin'", containsString("unknown"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'rcv'[1].'preferredName'", containsString("NM_000314.8(PTEN):c.19G>T (p.Glu7Ter)"))
                .body("[0].'my_variant_info'.'annotation'.'clinVar'.'variantId'", numberMatches(493079.0))
                .body("[1].'variant'", containsString("10.0:g.89624242_89624243del"))
                .body("[1].'originalVariantQuery'", containsString("10.0,89624242,89624243,AA,-"))
                .body("[1].'successfully_annotated'", equalTo(false));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }
    
    
    @Test(timeout = 60000)
    public void test_89() throws Exception {
        ExpectationHandler expectationHandler = expectationHandler();
        
        ValidatableResponse res_0 = given().accept("application/json")
                .header("x-EMextraHeader123", "42")
                .contentType("application/json")
                .body(" [ " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398284, " + 
                    " \"end\": 25398284, " + 
                    " \"referenceAllele\": \"C\", " + 
                    " \"variantAllele\": \"A\" " + 
                    " }, " + 
                    " { " + 
                    " \"chromosome\": \"12\", " + 
                    " \"start\": 25398283, " + 
                    " \"end\": 25398284, " + 
                    " \"referenceAllele\": \"AC\", " + 
                    " \"variantAllele\": \"AA\" " + 
                    " } " + 
                    " ] ")
                .post(baseUrlOfSut + "/annotation/genomic?" + 
                    "fields=annotation_summary&fields=nucleotide_context&" + 
                    "EMextraParam123=42")
                .then()
                .statusCode(200)
                .assertThat()
                .contentType("application/json")
                .body("size()", equalTo(2))
                .body("[0].'variant'", containsString("12:g.25398284C>A"))
                .body("[0].'originalVariantQuery'", containsString("12,25398284,25398284,C,A"))
                .body("[0].'hgvsg'", containsString("12:g.25398284C>A"))
                .body("[0].'assembly_name'", containsString("GRCh37"))
                .body("[0].'seq_region_name'", containsString("12"))
                .body("[0].'start'", numberMatches(2.5398284E7))
                .body("[0].'end'", numberMatches(2.5398284E7))
                .body("[0].'allele_string'", containsString("C/A"))
                .body("[0].'strand'", numberMatches(1.0))
                .body("[0].'most_severe_consequence'", containsString("missense_variant"))
                .body("[0].'transcript_consequences'.size()", equalTo(4))
                .body("[0].'transcript_consequences'[0].'exon'", containsString("2/6"))
                .body("[0].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000256078"))
                .body("[0].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000256078.4:p.Gly12Val"))
                .body("[0].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[0].'codons'", containsString("gGt/gTt"))
                .body("[0].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000256078"))
                .body("[0].'transcript_consequences'[0].'protein_start'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[0].'protein_end'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[0].'gene_symbol'", containsString("KRAS"))
                .body("[0].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000133703"))
                .body("[0].'transcript_consequences'[0].'amino_acids'", containsString("G/V"))
                .body("[0].'transcript_consequences'[0].'hgnc_id'", containsString("6407"))
                .body("[0].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[0].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.972))
                .body("[0].'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_033360.2"))
                .body("[0].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[1].'exon'", containsString("2/5"))
                .body("[0].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000311936"))
                .body("[0].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000308495.3:p.Gly12Val"))
                .body("[0].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[0].'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[1].'codons'", containsString("gGt/gTt"))
                .body("[0].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000308495"))
                .body("[0].'transcript_consequences'[1].'protein_start'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[1].'protein_end'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[1].'gene_symbol'", containsString("KRAS"))
                .body("[0].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000133703"))
                .body("[0].'transcript_consequences'[1].'amino_acids'", containsString("G/V"))
                .body("[0].'transcript_consequences'[1].'hgnc_id'", containsString("6407"))
                .body("[0].'transcript_consequences'[1].'polyphen_score'", numberMatches(0.96))
                .body("[0].'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[0].'transcript_consequences'[1].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_004985.3"))
                .body("[0].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant"))
                .body("[0].'transcript_consequences'[2].'exon'", containsString("2/3"))
                .body("[0].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000556131"))
                .body("[0].'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000451856.1:p.Gly12Val"))
                .body("[0].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[0].'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("[0].'transcript_consequences'[2].'codons'", containsString("gGt/gTt"))
                .body("[0].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000451856"))
                .body("[0].'transcript_consequences'[2].'protein_start'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[2].'protein_end'", numberMatches(12.0))
                .body("[0].'transcript_consequences'[2].'gene_symbol'", containsString("KRAS"))
                .body("[0].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000133703"))
                .body("[0].'transcript_consequences'[2].'amino_acids'", containsString("G/V"))
                .body("[0].'transcript_consequences'[2].'hgnc_id'", containsString("6407"))
                .body("[0].'transcript_consequences'[2].'polyphen_score'", numberMatches(0.884))
                .body("[0].'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[0].'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("[0].'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("[0].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[0].'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'successfully_annotated'", equalTo(true))
                .body("[0].'nucleotide_context'.'license'", nullValue())
                .body("[0].'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("[0].'nucleotide_context'.'annotation'.'seq'", containsString("ACC"))
                .body("[0].'nucleotide_context'.'annotation'.'hgvs'", containsString("12:g.25398284C>A"))
                .body("[0].'annotation_summary'.'variant'", containsString("12:g.25398284C>A"))
                .body("[0].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("12"))
                .body("[0].'annotation_summary'.'genomicLocation'.'start'", numberMatches(2.5398284E7))
                .body("[0].'annotation_summary'.'genomicLocation'.'end'", numberMatches(2.5398284E7))
                .body("[0].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[0].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("A"))
                .body("[0].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[0].'annotation_summary'.'variantType'", containsString("SNP"))
                .body("[0].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[0].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.972))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.972))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000311936"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'refSeq'", containsString("NM_004985.3"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/5"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.96))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000556131"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("2/3"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.884))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000256078"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gGt/gTt"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("G/V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("G"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("KRAS"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.G12V"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Gly12Val"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(12.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_033360.2"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("2/6"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.972))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("[0].'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"))
                .body("[1].'variant'", containsString("12:g.25398284C>A"))
                .body("[1].'originalVariantQuery'", containsString("12,25398283,25398284,AC,AA"))
                .body("[1].'hgvsg'", containsString("12:g.25398284C>A"))
                .body("[1].'assembly_name'", containsString("GRCh37"))
                .body("[1].'seq_region_name'", containsString("12"))
                .body("[1].'start'", numberMatches(2.5398284E7))
                .body("[1].'end'", numberMatches(2.5398284E7))
                .body("[1].'allele_string'", containsString("C/A"))
                .body("[1].'strand'", numberMatches(1.0))
                .body("[1].'most_severe_consequence'", containsString("missense_variant"))
                .body("[1].'transcript_consequences'.size()", equalTo(4))
                .body("[1].'transcript_consequences'[0].'exon'", containsString("2/6"))
                .body("[1].'transcript_consequences'[0].'transcript_id'", containsString("ENST00000256078"))
                .body("[1].'transcript_consequences'[0].'hgvsp'", containsString("ENSP00000256078.4:p.Gly12Val"))
                .body("[1].'transcript_consequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'transcript_consequences'[0].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[0].'codons'", containsString("gGt/gTt"))
                .body("[1].'transcript_consequences'[0].'protein_id'", containsString("ENSP00000256078"))
                .body("[1].'transcript_consequences'[0].'protein_start'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[0].'protein_end'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[0].'gene_symbol'", containsString("KRAS"))
                .body("[1].'transcript_consequences'[0].'gene_id'", containsString("ENSG00000133703"))
                .body("[1].'transcript_consequences'[0].'amino_acids'", containsString("G/V"))
                .body("[1].'transcript_consequences'[0].'hgnc_id'", containsString("6407"))
                .body("[1].'transcript_consequences'[0].'canonical'", containsString("1"))
                .body("[1].'transcript_consequences'[0].'polyphen_score'", numberMatches(0.972))
                .body("[1].'transcript_consequences'[0].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[1].'transcript_consequences'[0].'sift_score'", numberMatches(0.0))
                .body("[1].'transcript_consequences'[0].'sift_prediction'", containsString("deleterious"))
                .body("[1].'transcript_consequences'[0].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[0].'refseq_transcript_ids'", hasItems("NM_033360.2"))
                .body("[1].'transcript_consequences'[0].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[0].'consequence_terms'", hasItems("missense_variant"))
                .body("[1].'transcript_consequences'[1].'exon'", containsString("2/5"))
                .body("[1].'transcript_consequences'[1].'transcript_id'", containsString("ENST00000311936"))
                .body("[1].'transcript_consequences'[1].'hgvsp'", containsString("ENSP00000308495.3:p.Gly12Val"))
                .body("[1].'transcript_consequences'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[1].'transcript_consequences'[1].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[1].'codons'", containsString("gGt/gTt"))
                .body("[1].'transcript_consequences'[1].'protein_id'", containsString("ENSP00000308495"))
                .body("[1].'transcript_consequences'[1].'protein_start'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[1].'protein_end'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[1].'gene_symbol'", containsString("KRAS"))
                .body("[1].'transcript_consequences'[1].'gene_id'", containsString("ENSG00000133703"))
                .body("[1].'transcript_consequences'[1].'amino_acids'", containsString("G/V"))
                .body("[1].'transcript_consequences'[1].'hgnc_id'", containsString("6407"))
                .body("[1].'transcript_consequences'[1].'polyphen_score'", numberMatches(0.96))
                .body("[1].'transcript_consequences'[1].'polyphen_prediction'", containsString("probably_damaging"))
                .body("[1].'transcript_consequences'[1].'sift_score'", numberMatches(0.0))
                .body("[1].'transcript_consequences'[1].'sift_prediction'", containsString("deleterious"))
                .body("[1].'transcript_consequences'[1].'refseq_transcript_ids'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[1].'refseq_transcript_ids'", hasItems("NM_004985.3"))
                .body("[1].'transcript_consequences'[1].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[1].'consequence_terms'", hasItems("missense_variant"))
                .body("[1].'transcript_consequences'[2].'exon'", containsString("2/3"))
                .body("[1].'transcript_consequences'[2].'transcript_id'", containsString("ENST00000556131"))
                .body("[1].'transcript_consequences'[2].'hgvsp'", containsString("ENSP00000451856.1:p.Gly12Val"))
                .body("[1].'transcript_consequences'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[1].'transcript_consequences'[2].'variant_allele'", containsString("A"))
                .body("[1].'transcript_consequences'[2].'codons'", containsString("gGt/gTt"))
                .body("[1].'transcript_consequences'[2].'protein_id'", containsString("ENSP00000451856"))
                .body("[1].'transcript_consequences'[2].'protein_start'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[2].'protein_end'", numberMatches(12.0))
                .body("[1].'transcript_consequences'[2].'gene_symbol'", containsString("KRAS"))
                .body("[1].'transcript_consequences'[2].'gene_id'", containsString("ENSG00000133703"))
                .body("[1].'transcript_consequences'[2].'amino_acids'", containsString("G/V"))
                .body("[1].'transcript_consequences'[2].'hgnc_id'", containsString("6407"))
                .body("[1].'transcript_consequences'[2].'polyphen_score'", numberMatches(0.884))
                .body("[1].'transcript_consequences'[2].'polyphen_prediction'", containsString("possibly_damaging"))
                .body("[1].'transcript_consequences'[2].'sift_score'", numberMatches(0.0))
                .body("[1].'transcript_consequences'[2].'sift_prediction'", containsString("deleterious"))
                .body("[1].'transcript_consequences'[2].'consequence_terms'.size()", equalTo(1))
                .body("[1].'transcript_consequences'[2].'consequence_terms'", hasItems("missense_variant"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'successfully_annotated'", equalTo(true))
                .body("[1].'nucleotide_context'.'license'", nullValue())
                .body("[1].'nucleotide_context'.'annotation'.'molecule'", containsString("dna"))
                .body("[1].'nucleotide_context'.'annotation'.'seq'", containsString("ACC"))
                .body("[1].'nucleotide_context'.'annotation'.'hgvs'", containsString("12:g.25398284C>A"))
                .body("[1].'annotation_summary'.'variant'", containsString("12:g.25398284C>A"))
                .body("[1].'annotation_summary'.'genomicLocation'.'chromosome'", containsString("12"))
                .body("[1].'annotation_summary'.'genomicLocation'.'start'", numberMatches(2.5398284E7))
                .body("[1].'annotation_summary'.'genomicLocation'.'end'", numberMatches(2.5398284E7))
                .body("[1].'annotation_summary'.'genomicLocation'.'referenceAllele'", containsString("C"))
                .body("[1].'annotation_summary'.'genomicLocation'.'variantAllele'", containsString("A"))
                .body("[1].'annotation_summary'.'strandSign'", containsString("+"))
                .body("[1].'annotation_summary'.'variantType'", containsString("SNP"))
                .body("[1].'annotation_summary'.'assemblyName'", containsString("GRCh37"))
                .body("[1].'annotation_summary'.'canonicalTranscriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequences'.size()", equalTo(1))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'exon'", containsString("2/6"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'polyphenScore'", numberMatches(0.972))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequences'[0].'siftPrediction'", containsString("deleterious"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'.size()", equalTo(4))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'transcriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'refSeq'", containsString("NM_033360.2"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'exon'", containsString("2/6"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenScore'", numberMatches(0.972))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[0].'siftPrediction'", containsString("deleterious"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'transcriptId'", containsString("ENST00000311936"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'hgvsc'", containsString("ENST00000311936.3:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'refSeq'", containsString("NM_004985.3"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'exon'", containsString("2/5"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenScore'", numberMatches(0.96))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[1].'siftPrediction'", containsString("deleterious"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'transcriptId'", containsString("ENST00000556131"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'hgvsc'", containsString("ENST00000556131.1:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'exon'", containsString("2/3"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenScore'", numberMatches(0.884))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'polyphenPrediction'", containsString("possibly_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummaries'[2].'siftPrediction'", containsString("deleterious"))
                // Skipping assertions on the remaining 1 elements. This limit of 3 elements can be increased in the configurations
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'transcriptId'", containsString("ENST00000256078"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'codonChange'", containsString("gGt/gTt"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcids'", containsString("G/V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidRef'", containsString("G"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'aminoAcidAlt'", containsString("V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'consequenceTerms'", containsString("missense_variant"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hugoGeneSymbol'", containsString("KRAS"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hgvspShort'", containsString("p.G12V"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsp'", containsString("p.Gly12Val"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'hgvsc'", containsString("ENST00000256078.4:c.35G>T"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'start'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'proteinPosition'.'end'", numberMatches(12.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'refSeq'", containsString("NM_033360.2"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'variantClassification'", containsString("Missense_Mutation"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'exon'", containsString("2/6"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenScore'", numberMatches(0.972))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'polyphenPrediction'", containsString("probably_damaging"))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'siftScore'", numberMatches(0.0))
                .body("[1].'annotation_summary'.'transcriptConsequenceSummary'.'siftPrediction'", containsString("deleterious"));
        
        expectationHandler.expect(ems)
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(0)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")))
            .that(rso, ((Map) ((List) res_0.extract().response().jsonPath().getJsonObject("")).get(1)).keySet().containsAll(Arrays.asList("id", "intergenic_consequences", "originalVariantQuery", "variant")));
    }


}
