


<!DOCTYPE html>
<html id="htmlId">
<head>
  <title>Coverage Report > NotationConverter</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">org.cbioportal.genome_nexus.component.annotation</a>
</div>

<h1>Coverage Summary for Class: NotationConverter (org.cbioportal.genome_nexus.component.annotation)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">NotationConverter</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    64,7%
  </span>
  <span class="absValue">
    (11/17)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    58,2%
  </span>
  <span class="absValue">
    (82/141)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package org.cbioportal.genome_nexus.component.annotation;
&nbsp;
&nbsp;import org.cbioportal.genome_nexus.model.GenomicLocation;
&nbsp;import org.cbioportal.genome_nexus.util.GenomicVariant;
&nbsp;import org.cbioportal.genome_nexus.util.GenomicVariantUtil;
&nbsp;
&nbsp;import org.jetbrains.annotations.NotNull;
&nbsp;import org.jetbrains.annotations.Nullable;
&nbsp;import org.springframework.stereotype.Component;
&nbsp;
&nbsp;import java.util.ArrayList;
&nbsp;import java.util.LinkedHashMap;
&nbsp;import java.util.List;
&nbsp;import java.util.Map;
&nbsp;
&nbsp;
&nbsp;@Component
<b class="fc">&nbsp;public class NotationConverter {</b>
&nbsp;    public static final String DEFAULT_DELIMITER = &quot;,&quot;;
&nbsp;
&nbsp;    public String hgvsNormalizer(String hgvs) {
<b class="fc">&nbsp;        return hgvs.replace(&quot;chr&quot;,&quot;&quot;).replace(&quot;23:g.&quot;,&quot;X:g.&quot;).replace(&quot;24:g.&quot;,&quot;Y:g.&quot;);</b>
&nbsp;    }
&nbsp;
&nbsp;    public String chromosomeNormalizer(String chromosome) {
<b class="fc">&nbsp;        return chromosome.trim().replace(&quot;chr&quot;,&quot;&quot;).replace(&quot;23&quot;,&quot;X&quot;).replace(&quot;24&quot;,&quot;Y&quot;);</b>
&nbsp;    }
&nbsp;
&nbsp;    public GenomicLocation hgvsgToGenomicLocation(String hgvsg) {
<b class="fc">&nbsp;        GenomicLocation gl = new GenomicLocation();</b>
<b class="fc">&nbsp;        GenomicVariant gv = GenomicVariantUtil.fromHgvs(hgvsg);</b>
<b class="fc">&nbsp;        gl.setChromosome(gv.getChromosome());</b>
<b class="fc">&nbsp;        gl.setStart(gv.getStart());</b>
<b class="fc">&nbsp;        gl.setEnd(gv.getEnd());</b>
<b class="fc">&nbsp;        gl.setReferenceAllele(gv.getRef());</b>
<b class="fc">&nbsp;        gl.setVariantAllele(gv.getAlt());</b>
<b class="fc">&nbsp;        gl.setOriginalInput(hgvsg);</b>
<b class="fc">&nbsp;        return gl;</b>
&nbsp;    }
&nbsp;
&nbsp;    public List&lt;GenomicLocation&gt; hgvsgToGenomicLocations(List&lt;String&gt; hgvsgs) {
<b class="nc">&nbsp;        List&lt;GenomicLocation&gt; genomicLocations = new ArrayList();</b>
<b class="nc">&nbsp;        for (String hgvsg : hgvsgs) {</b>
<b class="nc">&nbsp;            genomicLocations.add(hgvsgToGenomicLocation(hgvsg));</b>
<b class="nc">&nbsp;        }</b>
<b class="nc">&nbsp;        return genomicLocations;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public GenomicLocation parseGenomicLocation(String genomicLocation) {
<b class="fc">&nbsp;        return parseGenomicLocation(genomicLocation, DEFAULT_DELIMITER);</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public GenomicLocation parseGenomicLocation(String genomicLocation, String delimiter) {
<b class="fc">&nbsp;        if (genomicLocation == null) {</b>
<b class="nc">&nbsp;            return null;</b>
&nbsp;        }
<b class="fc">&nbsp;        String[] parts = genomicLocation.split(delimiter);</b>
<b class="fc">&nbsp;        GenomicLocation location = null;</b>
<b class="fc">&nbsp;        if (parts.length &gt;= 5) {</b>
&nbsp;            // trim all parts
<b class="fc">&nbsp;            for (int i = 0; i &lt; parts.length; i++) {</b>
<b class="fc">&nbsp;                parts[i] = parts[i].trim();</b>
&nbsp;            }
<b class="fc">&nbsp;            location = new GenomicLocation();</b>
<b class="fc">&nbsp;            location.setChromosome(chromosomeNormalizer(parts[0]));</b>
<b class="fc">&nbsp;            location.setStart(Integer.parseInt(parts[1]));</b>
<b class="fc">&nbsp;            location.setEnd(Integer.parseInt(parts[2]));</b>
<b class="fc">&nbsp;            location.setReferenceAllele(parts[3]);</b>
<b class="fc">&nbsp;            location.setVariantAllele(parts[4]);</b>
<b class="fc">&nbsp;            location.setOriginalInput(genomicLocation);</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return location;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String genomicToHgvs(String genomicLocation) {
<b class="fc">&nbsp;        return genomicToHgvs(parseGenomicLocation(genomicLocation));</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String genomicToEnsemblRestRegion(String genomicLocation) {
<b class="nc">&nbsp;        return genomicToEnsemblRestRegion(parseGenomicLocation(genomicLocation));</b>
&nbsp;    }
&nbsp;
&nbsp;    /*
&nbsp;     * Normalize genomic location:
&nbsp;     *
&nbsp;     * 1. Convert VCF style start,end,ref,alt to MAF by looking for common
&nbsp;     * prefix. (TODO: not sure if this is always a good idea)
&nbsp;     * 2. Normalize chromsome names.
&nbsp;     */
&nbsp;    public GenomicLocation normalizeGenomicLocation(GenomicLocation genomicLocation) {
<b class="fc">&nbsp;        GenomicLocation normalizedGenomicLocation = new GenomicLocation();</b>
&nbsp;        // if original input is set in the incoming genomic location object then use the same value
&nbsp;        // for the normalized genomic location object returned, otherwise set it to the
&nbsp;        // string representation of the incoming genomic location object
<b class="fc">&nbsp;        if (genomicLocation.getOriginalInput() != null &amp;&amp; !genomicLocation.getOriginalInput().isEmpty()) {</b>
<b class="fc">&nbsp;            normalizedGenomicLocation.setOriginalInput(genomicLocation.getOriginalInput());</b>
&nbsp;        } else {
<b class="nc">&nbsp;            normalizedGenomicLocation.setOriginalInput(genomicLocation.toString());</b>
&nbsp;        }
&nbsp;
&nbsp;        // normalize chromosome name
<b class="fc">&nbsp;        String chr = chromosomeNormalizer(genomicLocation.getChromosome().trim());</b>
<b class="fc">&nbsp;        normalizedGenomicLocation.setChromosome(chr);</b>
&nbsp;
&nbsp;        // convert vcf style start,end,ref,alt to MAF style
<b class="fc">&nbsp;        Integer start = genomicLocation.getStart();</b>
<b class="fc">&nbsp;        Integer end = genomicLocation.getEnd();</b>
<b class="fc">&nbsp;        String ref = genomicLocation.getReferenceAllele().trim();</b>
<b class="fc">&nbsp;        String var = genomicLocation.getVariantAllele().trim();</b>
&nbsp;
<b class="fc">&nbsp;        String prefix = &quot;&quot;;</b>
&nbsp;
<b class="fc">&nbsp;        if (!ref.equals(var)) {</b>
<b class="fc">&nbsp;            prefix = longestCommonPrefix(ref, var);</b>
&nbsp;        }
&nbsp;
&nbsp;        // Remove common prefix and adjust variant position accordingly
<b class="fc">&nbsp;        if (prefix.length() &gt; 0) {</b>
<b class="fc">&nbsp;            ref = ref.substring(prefix.length());</b>
<b class="fc">&nbsp;            var = var.substring(prefix.length());</b>
<b class="fc">&nbsp;            int nStart = start + prefix.length();</b>
<b class="fc">&nbsp;            if (ref.length() == 0) {</b>
<b class="nc">&nbsp;                nStart -= 1;</b>
&nbsp;            }
<b class="fc">&nbsp;            start = nStart;</b>
&nbsp;        }
<b class="fc">&nbsp;        normalizedGenomicLocation.setStart(start);</b>
<b class="fc">&nbsp;        normalizedGenomicLocation.setEnd(end);</b>
<b class="fc">&nbsp;        normalizedGenomicLocation.setReferenceAllele(ref);</b>
<b class="fc">&nbsp;        normalizedGenomicLocation.setVariantAllele(var);</b>
<b class="fc">&nbsp;        return normalizedGenomicLocation;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String genomicToHgvs(GenomicLocation genomicLocation) {
<b class="fc">&nbsp;        if (genomicLocation == null) {</b>
<b class="nc">&nbsp;            return null;</b>
&nbsp;        }
<b class="fc">&nbsp;        GenomicLocation normalizedGenomicLocation = normalizeGenomicLocation(genomicLocation);</b>
<b class="fc">&nbsp;        String chr = normalizedGenomicLocation.getChromosome();</b>
<b class="fc">&nbsp;        Integer start = normalizedGenomicLocation.getStart();</b>
<b class="fc">&nbsp;        Integer end = normalizedGenomicLocation.getEnd();</b>
<b class="fc">&nbsp;        String ref = normalizedGenomicLocation.getReferenceAllele().trim();</b>
<b class="fc">&nbsp;        String var = normalizedGenomicLocation.getVariantAllele().trim();</b>
&nbsp;        String hgvs;
<b class="fc">&nbsp;        if (start &lt; 1) {</b>
&nbsp;            // cannot convert invalid locations
&nbsp;            // Ensembl uses a one-based coordinate system
<b class="nc">&nbsp;            hgvs = null;</b>
<b class="fc">&nbsp;        } else if (ref.equals(&quot;-&quot;) || ref.length() == 0 || ref.equals(&quot;NA&quot;) || ref.contains(&quot;--&quot;)) {</b>
&nbsp;            /*
&nbsp;            Process Insertion
&nbsp;            Example insertion: 17 36002277 36002278 - A
&nbsp;            Example output: 17:g.36002277_36002278insA
&nbsp;            */
&nbsp;            try {
<b class="fc">&nbsp;                hgvs = chr + &quot;:g.&quot; + start + &quot;_&quot; + String.valueOf(start + 1) + &quot;ins&quot; + var;</b>
<b class="nc">&nbsp;            } catch (NumberFormatException e) {</b>
<b class="nc">&nbsp;                return null;</b>
<b class="fc">&nbsp;            }</b>
<b class="fc">&nbsp;        } else if (var.equals(&quot;-&quot;) || var.length() == 0 || var.equals(&quot;NA&quot;) || var.contains(&quot;--&quot;)) {</b>
<b class="fc">&nbsp;            if (ref.length() == 1) {</b>
&nbsp;                /*
&nbsp;                Process Deletion (single positon)
&nbsp;                Example deletion: 13 32914438 32914438 T -
&nbsp;                Example output:   13:g.32914438del
&nbsp;                */
<b class="nc">&nbsp;                hgvs = chr + &quot;:g.&quot; + start + &quot;del&quot;;</b>
&nbsp;            }
&nbsp;            else {
&nbsp;                /*
&nbsp;                Process Deletion (multiple postion)
&nbsp;                Example deletion: 1 206811015 206811016  AC -
&nbsp;                Example output:   1:g.206811015_206811016del
&nbsp;                */
<b class="fc">&nbsp;                hgvs = chr + &quot;:g.&quot; + start + &quot;_&quot; + end + &quot;del&quot;;</b>
&nbsp;            } 
<b class="fc">&nbsp;        } else if (ref.length() &gt; 1 &amp;&amp; var.length() &gt;= 1) {</b>
&nbsp;            /*
&nbsp;            Process ONP (multiple deletion insertion)
&nbsp;            Example INDEL   : 2 216809708 216809709 CA T
&nbsp;            Example output: 2:g.216809708_216809709delinsT
&nbsp;            */
<b class="fc">&nbsp;            hgvs = chr + &quot;:g.&quot; + start + &quot;_&quot; + end + &quot;delins&quot; + var;</b>
<b class="fc">&nbsp;        } else if (ref.length() == 1 &amp;&amp; var.length() &gt; 1) {</b>
&nbsp;            /*
&nbsp;            Process ONP (single deletion insertion)
&nbsp;            Example INDEL   : 17 7579363 7579363 A TTT
&nbsp;            Example output: 17:g.7579363delinsTTT
&nbsp;            */
<b class="nc">&nbsp;            hgvs = chr + &quot;:g.&quot; + start + &quot;delins&quot; + var;</b>
&nbsp;        } else {
&nbsp;            /*
&nbsp;            Process SNV
&nbsp;            Example SNP   : 2 216809708 216809708 C T
&nbsp;            Example output: 2:g.216809708C&gt;T
&nbsp;            */
<b class="fc">&nbsp;            hgvs = chr + &quot;:g.&quot; + start + ref + &quot;&gt;&quot; + var;</b>
&nbsp;        }
<b class="fc">&nbsp;        return hgvs;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String genomicToEnsemblRestRegion(GenomicLocation genomicLocation) {
<b class="nc">&nbsp;        if (genomicLocation == null) {</b>
<b class="nc">&nbsp;            return null;</b>
&nbsp;        }
<b class="nc">&nbsp;        GenomicLocation normalizedGenomicLocation = normalizeGenomicLocation(genomicLocation);</b>
<b class="nc">&nbsp;        String chr = normalizedGenomicLocation.getChromosome();</b>
<b class="nc">&nbsp;        Integer start = normalizedGenomicLocation.getStart();</b>
<b class="nc">&nbsp;        Integer end = normalizedGenomicLocation.getEnd();</b>
<b class="nc">&nbsp;        String ref = normalizedGenomicLocation.getReferenceAllele().trim();</b>
<b class="nc">&nbsp;        String var = normalizedGenomicLocation.getVariantAllele().trim();</b>
&nbsp;        String region;
&nbsp;        // cannot convert invalid locations
&nbsp;        // Ensembl uses a one-based coordinate system
<b class="nc">&nbsp;        if (start &lt; 1) {</b>
<b class="nc">&nbsp;            region = null;</b>
<b class="nc">&nbsp;        } else if (ref.equals(&quot;-&quot;) || ref.length() == 0 || ref.equals(&quot;NA&quot;) || ref.contains(&quot;--&quot;)) {</b>
&nbsp;            /*
&nbsp;            Process Insertion
&nbsp;            Example insertion: 17 36002277 36002278 - A
&nbsp;            Example output: 17:36002278-36002277:1/A
&nbsp;            */
&nbsp;            try {
<b class="nc">&nbsp;                region = chr + &quot;:&quot; + String.valueOf(start + 1) + &quot;-&quot; + start  + &quot;:1/&quot; + var;</b>
&nbsp;            }
<b class="nc">&nbsp;            catch (NumberFormatException e) {</b>
<b class="nc">&nbsp;                return null;</b>
<b class="nc">&nbsp;            }</b>
<b class="nc">&nbsp;        } else if (var.equals(&quot;-&quot;) || var.length() == 0) {</b>
&nbsp;            /*
&nbsp;            Process Deletion
&nbsp;            Example deletion: 1 206811015 206811016  AC -
&nbsp;            Example output:   1:206811015-206811016:1/-
&nbsp;            */
<b class="nc">&nbsp;            region = chr + &quot;:&quot; + start + &quot;-&quot; + end + &quot;:1/-&quot;;</b>
<b class="nc">&nbsp;        } else if (ref.length() &gt; 1 || var.length() &gt; 1) {</b>
&nbsp;            /*
&nbsp;            Process ONP
&nbsp;            Example SNP   : 2 216809708 216809709 CA T
&nbsp;            Example output: 2:216809708-216809709:1/T
&nbsp;            */
<b class="nc">&nbsp;            region = chr + &quot;:&quot; + start + &quot;-&quot; + end + &quot;:1/&quot; + var;</b>
&nbsp;        } else {
&nbsp;            /*
&nbsp;            Process SNV
&nbsp;            Example SNP   : 2 216809708 216809708 C T
&nbsp;            Example output: 2:216809708-216809708:1/T
&nbsp;            */
<b class="nc">&nbsp;            region = chr + &quot;:&quot; + start + &quot;-&quot; + start + &quot;:1/&quot; + var;</b>
&nbsp;        }
<b class="nc">&nbsp;        return region;</b>
&nbsp;    }
&nbsp;
&nbsp;    @NotNull
&nbsp;    public Map&lt;String, GenomicLocation&gt; genomicToHgvsMap(List&lt;GenomicLocation&gt; genomicLocations) {
<b class="nc">&nbsp;        Map&lt;String, GenomicLocation&gt; variantToGenomicLocation = new LinkedHashMap&lt;&gt;();</b>
&nbsp;        // convert genomic location to hgvs notation (there is always 1-1 mapping)
<b class="nc">&nbsp;        for (GenomicLocation location : genomicLocations) {</b>
<b class="nc">&nbsp;            String hgvs = genomicToHgvs(location);</b>
<b class="nc">&nbsp;            if (hgvs != null) {</b>
<b class="nc">&nbsp;                variantToGenomicLocation.put(hgvs, location);</b>
&nbsp;            }
<b class="nc">&nbsp;        }</b>
<b class="nc">&nbsp;        return variantToGenomicLocation;</b>
&nbsp;    }
&nbsp;
&nbsp;    public List&lt;String&gt; genomicToHgvs(List&lt;GenomicLocation&gt; genomicLocations) {
<b class="fc">&nbsp;        List&lt;String&gt; hgvsList = new ArrayList&lt;&gt;();</b>
<b class="fc">&nbsp;        for (GenomicLocation location : genomicLocations) {</b>
<b class="fc">&nbsp;            String hgvs = genomicToHgvs(location);</b>
<b class="fc">&nbsp;            if (hgvs != null) {</b>
<b class="fc">&nbsp;                hgvsList.add(hgvs);</b>
&nbsp;            }
<b class="fc">&nbsp;        }</b>
<b class="fc">&nbsp;        return hgvsList;</b>
&nbsp;    }
&nbsp;
&nbsp;    @NotNull
&nbsp;    public Map&lt;String, GenomicLocation&gt; genomicToEnsemblRestRegionMap(List&lt;GenomicLocation&gt; genomicLocations) {
<b class="nc">&nbsp;        Map&lt;String, GenomicLocation&gt; variantToGenomicLocation = new LinkedHashMap&lt;&gt;();</b>
&nbsp;        // convert genomic location to ensembl rest region notation (there is always 1-1 mapping)
<b class="nc">&nbsp;        for (GenomicLocation location : genomicLocations) {</b>
<b class="nc">&nbsp;            String ensemblRestRegion = genomicToEnsemblRestRegion(location);</b>
<b class="nc">&nbsp;            if (ensemblRestRegion != null) {</b>
<b class="nc">&nbsp;                variantToGenomicLocation.put(ensemblRestRegion, location);</b>
&nbsp;            }
<b class="nc">&nbsp;        }</b>
<b class="nc">&nbsp;        return variantToGenomicLocation;</b>
&nbsp;    }
&nbsp;
&nbsp;    public List&lt;String&gt; genomicToEnsemblRestRegion(List&lt;GenomicLocation&gt; genomicLocations) {
<b class="nc">&nbsp;        List&lt;String&gt; ensemblRestRegionsList = new ArrayList&lt;&gt;();</b>
<b class="nc">&nbsp;        for (GenomicLocation location : genomicLocations) {</b>
<b class="nc">&nbsp;            String ensemblRestRegion = genomicToEnsemblRestRegion(location);</b>
<b class="nc">&nbsp;            if (ensemblRestRegion != null) {</b>
<b class="nc">&nbsp;                ensemblRestRegionsList.add(ensemblRestRegion);</b>
&nbsp;            }
<b class="nc">&nbsp;        }</b>
<b class="nc">&nbsp;        return ensemblRestRegionsList;</b>
&nbsp;    }
&nbsp;
&nbsp;    // TODO factor out to a utility class as a static method if needed
&nbsp;    @NotNull
&nbsp;    public String longestCommonPrefix(String str1, String str2) {
<b class="fc">&nbsp;        if (str1 == null || str2 == null) {</b>
<b class="nc">&nbsp;            return &quot;&quot;;</b>
&nbsp;        }
<b class="fc">&nbsp;        for (int prefixLen = 0; prefixLen &lt; str1.length(); prefixLen++) {</b>
<b class="fc">&nbsp;            char c = str1.charAt(prefixLen);</b>
<b class="fc">&nbsp;            if (prefixLen &gt;= str2.length() || str2.charAt(prefixLen) != c) {</b>
&nbsp;                // mismatch found
<b class="fc">&nbsp;                return str2.substring(0, prefixLen);</b>
&nbsp;            }
&nbsp;        }
<b class="nc">&nbsp;        return str1;</b>
&nbsp;    }
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2024-05-01 05:19</div>
</div>
</body>
</html>
