


<!DOCTYPE html>
<html id="htmlId">
<head>
  <title>Coverage Report > GenomicLocationResolver</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">org.cbioportal.genome_nexus.component.annotation</a>
</div>

<h1>Coverage Summary for Class: GenomicLocationResolver (org.cbioportal.genome_nexus.component.annotation)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">GenomicLocationResolver</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (8/8)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    95,2%
  </span>
  <span class="absValue">
    (40/42)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package org.cbioportal.genome_nexus.component.annotation;
&nbsp;
&nbsp;import org.cbioportal.genome_nexus.model.GenomicLocation;
&nbsp;import org.cbioportal.genome_nexus.model.VariantAnnotation;
&nbsp;import org.jetbrains.annotations.NotNull;
&nbsp;import org.jetbrains.annotations.Nullable;
&nbsp;import org.springframework.stereotype.Component;
&nbsp;
&nbsp;@Component
<b class="fc">&nbsp;public class GenomicLocationResolver</b>
&nbsp;{
&nbsp;    @NotNull
&nbsp;    public GenomicLocation resolve(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        GenomicLocation genomicLocation = new GenomicLocation();</b>
&nbsp;
<b class="fc">&nbsp;        genomicLocation.setChromosome(this.resolveChromosome(variantAnnotation));</b>
<b class="fc">&nbsp;        genomicLocation.setStart(this.resolveStart(variantAnnotation));</b>
<b class="fc">&nbsp;        genomicLocation.setEnd(this.resolveEnd(variantAnnotation));</b>
<b class="fc">&nbsp;        genomicLocation.setReferenceAllele(this.resolveReferenceAllele(variantAnnotation));</b>
<b class="fc">&nbsp;        genomicLocation.setVariantAllele(this.resolveVariantAllele(variantAnnotation));</b>
&nbsp;
<b class="fc">&nbsp;        return genomicLocation;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    private String resolveChromosome(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        String chromosome = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (variantAnnotation != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getSeqRegionName() != null)</b>
&nbsp;        {
<b class="fc">&nbsp;            chromosome = variantAnnotation.getSeqRegionName();</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return chromosome;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public Integer resolveStart(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        Integer start = variantAnnotation.getStart();</b>
<b class="fc">&nbsp;        Integer end = variantAnnotation.getEnd();</b>
&nbsp;
<b class="fc">&nbsp;        if (start == null) {</b>
<b class="nc">&nbsp;            start = end;</b>
&nbsp;        }
&nbsp;
&nbsp;        // in case end &lt; start, use end instead of start
&nbsp;        // (this indicates that start &amp; end points are not properly reported though)
<b class="fc">&nbsp;        if (end != null) {</b>
<b class="fc">&nbsp;            start = Math.min(start, end);</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return start;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public Integer resolveEnd(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        Integer start = variantAnnotation.getStart();</b>
<b class="fc">&nbsp;        Integer end = variantAnnotation.getEnd();</b>
&nbsp;
<b class="fc">&nbsp;        if (end == null) {</b>
<b class="nc">&nbsp;            end = start;</b>
&nbsp;        }
&nbsp;
&nbsp;        // in case end &lt; start, use start instead of end
&nbsp;        // (this indicates that start &amp; end points are not properly reported though)
<b class="fc">&nbsp;        if (start != null) {</b>
<b class="fc">&nbsp;            end = Math.max(start, end);</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return end;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String resolveReferenceAllele(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        String referenceAllele = null;</b>
<b class="fc">&nbsp;        String[] alleles = extractAlleles(variantAnnotation);</b>
&nbsp;
<b class="fc">&nbsp;        if (alleles != null &amp;&amp; alleles.length == 2)</b>
&nbsp;        {
<b class="fc">&nbsp;            referenceAllele = alleles[0];</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return referenceAllele;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String resolveVariantAllele(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        String variantAllele = null;</b>
<b class="fc">&nbsp;        String[] alleles = extractAlleles(variantAnnotation);</b>
&nbsp;
<b class="fc">&nbsp;        if (alleles != null &amp;&amp; alleles.length == 2)</b>
&nbsp;        {
<b class="fc">&nbsp;            variantAllele = alleles[1];</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return variantAllele;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    private String[] extractAlleles(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        String[] alleles = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (variantAnnotation != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getAlleleString() != null)</b>
&nbsp;        {
<b class="fc">&nbsp;            alleles = variantAnnotation.getAlleleString().split(&quot;/&quot;, -1);</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return alleles;</b>
&nbsp;    }
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2024-05-01 05:19</div>
</div>
</body>
</html>
