


<!DOCTYPE html>
<html id="htmlId">
<head>
  <title>Coverage Report > ProteinChangeResolver</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">org.cbioportal.genome_nexus.component.annotation</a>
</div>

<h1>Coverage Summary for Class: ProteinChangeResolver (org.cbioportal.genome_nexus.component.annotation)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">ProteinChangeResolver</td>
<td class="coverageStat">
  <span class="percent">
    100%
  </span>
  <span class="absValue">
    (1/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    85,7%
  </span>
  <span class="absValue">
    (12/14)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    73,7%
  </span>
  <span class="absValue">
    (101/137)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package org.cbioportal.genome_nexus.component.annotation;
&nbsp;
&nbsp;import org.cbioportal.genome_nexus.model.TranscriptConsequence;
&nbsp;import org.cbioportal.genome_nexus.model.VariantAnnotation;
&nbsp;import org.jetbrains.annotations.Nullable;
&nbsp;import org.springframework.beans.factory.annotation.Autowired;
&nbsp;import org.springframework.stereotype.Component;
&nbsp;
&nbsp;import java.util.Arrays;
&nbsp;import java.util.HashSet;
&nbsp;import java.util.List;
&nbsp;import java.util.Set;
&nbsp;import java.util.regex.Matcher;
&nbsp;import java.util.regex.Pattern;
&nbsp;import java.util.stream.Collectors;
&nbsp;
&nbsp;@Component
&nbsp;public class ProteinChangeResolver
&nbsp;{
<b class="fc">&nbsp;    public static final String AA3TO1[][] = {</b>
&nbsp;        {&quot;Ala&quot;, &quot;A&quot;}, {&quot;Arg&quot;, &quot;R&quot;}, {&quot;Asn&quot;, &quot;N&quot;}, {&quot;Asp&quot;, &quot;D&quot;}, {&quot;Asx&quot;, &quot;B&quot;}, {&quot;Cys&quot;, &quot;C&quot;},
&nbsp;        {&quot;Glu&quot;, &quot;E&quot;}, {&quot;Gln&quot;, &quot;Q&quot;}, {&quot;Glx&quot;, &quot;Z&quot;}, {&quot;Gly&quot;, &quot;G&quot;}, {&quot;His&quot;, &quot;H&quot;}, {&quot;Ile&quot;, &quot;I&quot;},
&nbsp;        {&quot;Leu&quot;, &quot;L&quot;}, {&quot;Lys&quot;, &quot;K&quot;}, {&quot;Met&quot;, &quot;M&quot;}, {&quot;Phe&quot;, &quot;F&quot;}, {&quot;Pro&quot;, &quot;P&quot;}, {&quot;Ser&quot;, &quot;S&quot;},
&nbsp;        {&quot;Thr&quot;, &quot;T&quot;}, {&quot;Trp&quot;, &quot;W&quot;}, {&quot;Tyr&quot;, &quot;Y&quot;}, {&quot;Val&quot;, &quot;V&quot;}, {&quot;Xxx&quot;, &quot;X&quot;}, {&quot;Ter&quot;, &quot;*&quot;}
&nbsp;    };
&nbsp;
<b class="fc">&nbsp;    public static final Set&lt;String&gt; SPLICE_SITE_VARIANTS = new HashSet&lt;&gt;(</b>
<b class="fc">&nbsp;        Arrays.asList(&quot;splice_acceptor_variant&quot;, &quot;splice_donor_variant&quot;, &quot;splice_region_variant&quot;)</b>
&nbsp;    );
&nbsp;
&nbsp;    private final CanonicalTranscriptResolver canonicalTranscriptResolver;
&nbsp;    private final VariantClassificationResolver variantClassificationResolver;
&nbsp;    private final Pattern cDnaExtractor;
&nbsp;
&nbsp;    @Autowired
&nbsp;    public ProteinChangeResolver(CanonicalTranscriptResolver canonicalTranscriptResolver,
&nbsp;                                 VariantClassificationResolver variantClassificationResolver)
<b class="fc">&nbsp;    {</b>
<b class="fc">&nbsp;        this.canonicalTranscriptResolver = canonicalTranscriptResolver;</b>
<b class="fc">&nbsp;        this.variantClassificationResolver = variantClassificationResolver;</b>
<b class="fc">&nbsp;        this.cDnaExtractor = Pattern.compile(&quot;.*[cn].-?\\*?(\\d+).*&quot;);</b>
&nbsp;    }
&nbsp;
&nbsp;    /**
&nbsp;     * Resolves to a single hgvsp short for the given variant annotation.
&nbsp;     *
&nbsp;     * @param variantAnnotation
&nbsp;     * @return
&nbsp;     */
&nbsp;    @Nullable
&nbsp;    public String resolveHgvspShort(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="nc">&nbsp;        TranscriptConsequence canonicalTranscript =</b>
<b class="nc">&nbsp;            this.canonicalTranscriptResolver.resolve(variantAnnotation);</b>
&nbsp;
&nbsp;        // resolve for the canonical transcript
<b class="nc">&nbsp;        return this.resolveHgvspShort(variantAnnotation, canonicalTranscript);</b>
&nbsp;    }
&nbsp;
&nbsp;    /**
&nbsp;     * Resolves to a single hgvsp short for the given transcript consequence and variant annotation.
&nbsp;     *
&nbsp;     * @param transcriptConsequence
&nbsp;     * @return
&nbsp;     */
&nbsp;    @Nullable
&nbsp;    public String resolveHgvspShort(VariantAnnotation variantAnnotation,
&nbsp;                                    TranscriptConsequence transcriptConsequence)
&nbsp;    {
<b class="fc">&nbsp;        String hgvspShort = null;</b>
&nbsp;
<b class="fc">&nbsp;        String hgvsp = this.resolveHgvsp(transcriptConsequence);</b>
<b class="fc">&nbsp;        String hgvsc = this.resolveHgvsc(transcriptConsequence);</b>
&nbsp;
<b class="fc">&nbsp;        if (hgvsp != null) {</b>
<b class="fc">&nbsp;            hgvspShort = this.resolveHgvspShortFromHgvsp(hgvsp);</b>
&nbsp;        }
<b class="fc">&nbsp;        else if (hgvsc != null) {</b>
<b class="fc">&nbsp;            hgvspShort = this.resolveHgvspShortFromHgvsc(hgvsc, variantAnnotation, transcriptConsequence);</b>
&nbsp;        }
&nbsp;
&nbsp;        // if hgvspShort is still unresolved then
&nbsp;        // try to salvage using protein_start, amino_acids, and consequence_terms
<b class="fc">&nbsp;        if (hgvspShort == null &amp;&amp;</b>
&nbsp;            transcriptConsequence != null &amp;&amp;
<b class="fc">&nbsp;            transcriptConsequence.getAminoAcids() != null)</b>
&nbsp;        {
<b class="nc">&nbsp;            hgvspShort = this.resolveHgvspShortFromAAs(transcriptConsequence);</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvspShort;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    private String resolveHgvspShortFromHgvsp(String hgvsp)
&nbsp;    {
<b class="fc">&nbsp;        if (hgvsp == null) {</b>
<b class="nc">&nbsp;            return null;</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        String hgvspShort = hgvsp;</b>
&nbsp;
<b class="fc">&nbsp;        for (int i = 0; i &lt; 24; i++) {</b>
<b class="fc">&nbsp;            if (hgvsp.contains(AA3TO1[i][0])) {</b>
<b class="fc">&nbsp;                hgvspShort = hgvspShort.replaceAll(AA3TO1[i][0], AA3TO1[i][1]);</b>
&nbsp;            }
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvspShort;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    private String resolveHgvspShortFromHgvsc(String hgvsc,
&nbsp;                                              VariantAnnotation variantAnnotation,
&nbsp;                                              TranscriptConsequence transcriptConsequence)
&nbsp;    {
<b class="fc">&nbsp;        String hgvspShort = null;</b>
&nbsp;
<b class="fc">&nbsp;        Integer cPos = 0;</b>
<b class="fc">&nbsp;        Integer pPos = 0;</b>
&nbsp;
<b class="fc">&nbsp;        Matcher m = this.cDnaExtractor.matcher(hgvsc);</b>
&nbsp;
<b class="fc">&nbsp;        if (m.matches())</b>
&nbsp;        {
&nbsp;            // &quot;c.*&quot; represents for UTR
<b class="fc">&nbsp;            if (hgvsc.contains(&quot;c.*&quot;)) {</b>
<b class="fc">&nbsp;                return null;</b>
&nbsp;            }
&nbsp;
<b class="fc">&nbsp;            cPos = Integer.parseInt(m.group(1));</b>
<b class="fc">&nbsp;            cPos = cPos &lt; 1 ? 1 : cPos;</b>
<b class="fc">&nbsp;            pPos = (cPos + 2) / 3;</b>
&nbsp;
<b class="fc">&nbsp;            if (SPLICE_SITE_VARIANTS.contains(transcriptConsequence.getConsequenceTerms().get(0)))</b>
&nbsp;            {
<b class="nc">&nbsp;                hgvspShort = &quot;p.X&quot; + String.valueOf(pPos) + &quot;_splice&quot;;</b>
&nbsp;            }
<b class="fc">&nbsp;            else if (transcriptConsequence.getAminoAcids() == null)</b>
&nbsp;            {
<b class="fc">&nbsp;                String variantClassification = this.variantClassificationResolver.resolve(</b>
&nbsp;                    variantAnnotation, transcriptConsequence);
&nbsp;
<b class="fc">&nbsp;                hgvspShort = &quot;p.*&quot; + String.valueOf(pPos);</b>
&nbsp;
<b class="fc">&nbsp;                if (variantClassification != null &amp;&amp; variantClassification.toLowerCase().startsWith(&quot;frame_shift&quot;)) {</b>
<b class="nc">&nbsp;                    hgvspShort += &quot;fs*&quot;;</b>
&nbsp;                }
&nbsp;                else {
<b class="fc">&nbsp;                    hgvspShort += &quot;*&quot;;</b>
&nbsp;                }
&nbsp;            }
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvspShort;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    private String resolveHgvspShortFromAAs(TranscriptConsequence transcriptConsequence)
&nbsp;    {
<b class="nc">&nbsp;        String hgvspShort = null;</b>
&nbsp;
&nbsp;        try {
<b class="nc">&nbsp;            String[] aaParts = transcriptConsequence.getAminoAcids().split(&quot;/&quot;);</b>
&nbsp;
&nbsp;            // TODO: always using get(0) here, maybe better to get most
&nbsp;            // impactful consequence and use that?
<b class="nc">&nbsp;            if (transcriptConsequence.getConsequenceTerms() != null &amp;&amp;</b>
<b class="nc">&nbsp;                transcriptConsequence.getConsequenceTerms().get(0).toLowerCase().contains(&quot;inframe_insertion&quot;))</b>
&nbsp;            {
&nbsp;                // to prevent IndexOutOfBoundsException&#39;s we check for &#39;dup&#39; in the HGVSc field (ex: ENST00000357654.3:c.5266dupC)
&nbsp;                // since the &#39;amino_acids&#39; may not always be provided in such a way where the reference allele is available
&nbsp;                // ex: with the reference aa &#39;N/KN&#39; and without the reference aa &#39;-/K&#39;.
&nbsp;                // the second format will throw an IndexOutOfBoundsException when trying to access substring index 1
<b class="nc">&nbsp;                if (transcriptConsequence.getHgvsc() != null &amp;&amp;</b>
<b class="nc">&nbsp;                    transcriptConsequence.getHgvsc().contains(&quot;dup&quot;))</b>
&nbsp;                {
<b class="nc">&nbsp;                    if (aaParts[1].length() == 1) {</b>
<b class="nc">&nbsp;                        hgvspShort = &quot;p.&quot; + aaParts[1].substring(0,1) +</b>
<b class="nc">&nbsp;                            String.valueOf(transcriptConsequence.getProteinStart() - 1) +</b>
&nbsp;                            &quot;dup&quot;;
&nbsp;                    } else {
&nbsp;                        // e.g for KIT p.Q575_G592dup
&nbsp;                        // 4:g.55593656_55593657insCAACTTCCTTATGATCACAAATGGGAGTTTCCCAGAAACAGGCTGAGTTTTGGT
<b class="nc">&nbsp;                        hgvspShort = &quot;p.&quot; + aaParts[1].substring(0,1) +</b>
<b class="nc">&nbsp;                            String.valueOf(transcriptConsequence.getProteinStart() + 1) +</b>
&nbsp;                            &quot;_&quot; +
<b class="nc">&nbsp;                            aaParts[1].substring(aaParts[1].length() - 1) +</b>
&nbsp;                            // protein end for an instertion will be protein start + 1, so use length of amino acid insted
<b class="nc">&nbsp;                            String.valueOf(transcriptConsequence.getProteinStart() + aaParts[1].length()) +</b>
&nbsp;                            &quot;dup&quot;;
&nbsp;
&nbsp;                    }
&nbsp;                }
&nbsp;                else
&nbsp;                {
<b class="nc">&nbsp;                    hgvspShort = &quot;p.X&quot; +</b>
<b class="nc">&nbsp;                        transcriptConsequence.getProteinStart() + &quot;_X&quot; +</b>
<b class="nc">&nbsp;                        transcriptConsequence.getProteinEnd() + &quot;ins&quot; +</b>
&nbsp;                        aaParts[1];
&nbsp;                }
&nbsp;            }
<b class="nc">&nbsp;            else if (transcriptConsequence.getConsequenceTerms() != null &amp;&amp;</b>
<b class="nc">&nbsp;                transcriptConsequence.getConsequenceTerms().get(0).toLowerCase().contains(&quot;inframe_deletion&quot;))</b>
&nbsp;            {
<b class="nc">&nbsp;                hgvspShort = &quot;p.&quot; + aaParts[0] + &quot;del&quot;;</b>
&nbsp;            }
&nbsp;            else
&nbsp;            {
<b class="nc">&nbsp;                hgvspShort = &quot;p.&quot; + aaParts[0] + transcriptConsequence.getProteinStart();</b>
&nbsp;
<b class="nc">&nbsp;                if (transcriptConsequence.getConsequenceTerms() != null &amp;&amp;</b>
<b class="nc">&nbsp;                    transcriptConsequence.getConsequenceTerms().get(0).toLowerCase().contains(&quot;frameshift_variant&quot;))</b>
&nbsp;                {
<b class="nc">&nbsp;                    hgvspShort += &quot;fs&quot;;</b>
&nbsp;                }
&nbsp;                else
&nbsp;                {
<b class="nc">&nbsp;                    hgvspShort += aaParts[1];</b>
&nbsp;                }
&nbsp;            }
<b class="nc">&nbsp;        } catch (Exception e) {</b>
&nbsp;            // LOG.debug(&quot;Failed to salvage HGVSp_Short from protein start, amino acids, and consequence terms&quot;);
<b class="nc">&nbsp;        }</b>
&nbsp;
<b class="nc">&nbsp;        return hgvspShort;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String resolveHgvsc(TranscriptConsequence transcriptConsequence)
&nbsp;    {
<b class="fc">&nbsp;        String hgvsc = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (transcriptConsequence != null &amp;&amp;</b>
<b class="fc">&nbsp;            transcriptConsequence.getHgvsc() != null)</b>
&nbsp;        {
<b class="fc">&nbsp;            hgvsc = transcriptConsequence.getHgvsc();</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvsc;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public String resolveHgvsp(TranscriptConsequence transcriptConsequence)
&nbsp;    {
<b class="fc">&nbsp;        String hgvsp = null;</b>
&nbsp;
<b class="fc">&nbsp;        String variantClassification = this.variantClassificationResolver.resolve(null, transcriptConsequence);</b>
&nbsp;
&nbsp;        // only use hgvsp if the most severe impact is not a splice variant
<b class="fc">&nbsp;        if (transcriptConsequence != null &amp;&amp;</b>
<b class="fc">&nbsp;            transcriptConsequence.getHgvsp() != null &amp;&amp;</b>
<b class="fc">&nbsp;            !(variantClassification != null &amp;&amp; variantClassification.toLowerCase().contains(&quot;splice&quot;))</b>
&nbsp;            )
&nbsp;        {
<b class="fc">&nbsp;            hgvsp = this.normalizeHgvsp(transcriptConsequence.getHgvsp());</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvsp;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public List&lt;String&gt; resolveAllHgvsp(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        List&lt;String&gt; hgvsp = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (variantAnnotation != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences() != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences().size() &gt; 0)</b>
&nbsp;        {
<b class="fc">&nbsp;            Set&lt;String&gt; hgvspSet = new HashSet&lt;&gt;();</b>
<b class="fc">&nbsp;            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {</b>
<b class="fc">&nbsp;                if (transcriptConsequence.getHgvsp() != null) {</b>
<b class="fc">&nbsp;                    hgvspSet.add(transcriptConsequence.getHgvsp());</b>
&nbsp;                }
<b class="fc">&nbsp;            }</b>
<b class="fc">&nbsp;            hgvsp = hgvspSet.stream().collect(Collectors.toList());</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvsp;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public List&lt;String&gt; resolveAllHgvsc(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        List&lt;String&gt; hgvsc = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (variantAnnotation != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences() != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences().size() &gt; 0)</b>
&nbsp;        {
<b class="fc">&nbsp;            Set&lt;String&gt; hgvscSet = new HashSet&lt;&gt;();</b>
<b class="fc">&nbsp;            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {</b>
<b class="fc">&nbsp;                if (transcriptConsequence.getHgvsc() != null) {</b>
<b class="fc">&nbsp;                    hgvscSet.add(transcriptConsequence.getHgvsc());</b>
&nbsp;                }
<b class="fc">&nbsp;            }</b>
<b class="fc">&nbsp;            hgvsc = hgvscSet.stream().collect(Collectors.toList());</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvsc;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public List&lt;String&gt; resolveAllHgvspShort(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        List&lt;String&gt; hgvspShort = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (variantAnnotation != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences() != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences().size() &gt; 0)</b>
&nbsp;        {
<b class="fc">&nbsp;            Set&lt;String&gt; hgvspShortSet = new HashSet&lt;&gt;();</b>
<b class="fc">&nbsp;            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {</b>
<b class="fc">&nbsp;                String hgvspShortResult = this.resolveHgvspShort(variantAnnotation, transcriptConsequence);</b>
<b class="fc">&nbsp;                if (hgvspShortResult != null) {</b>
<b class="fc">&nbsp;                    hgvspShortSet.add(hgvspShortResult);</b>
&nbsp;                }
<b class="fc">&nbsp;            }</b>
<b class="fc">&nbsp;            hgvspShort = hgvspShortSet.stream().collect(Collectors.toList());</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return hgvspShort;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    public List&lt;String&gt; resolveAllCdna(VariantAnnotation variantAnnotation)
&nbsp;    {
<b class="fc">&nbsp;        List&lt;String&gt; cdna = null;</b>
&nbsp;
<b class="fc">&nbsp;        if (variantAnnotation != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences() != null &amp;&amp;</b>
<b class="fc">&nbsp;            variantAnnotation.getTranscriptConsequences().size() &gt; 0)</b>
&nbsp;        {
<b class="fc">&nbsp;            Set&lt;String&gt; cdnaSet = new HashSet&lt;&gt;();</b>
<b class="fc">&nbsp;            for (TranscriptConsequence transcriptConsequence : variantAnnotation.getTranscriptConsequences()) {</b>
<b class="fc">&nbsp;                if (transcriptConsequence.getHgvsc() != null &amp;&amp; transcriptConsequence.getHgvsc().contains(&quot;:c.&quot;)) {</b>
<b class="fc">&nbsp;                    cdnaSet.add(transcriptConsequence.getHgvsc().split(&quot;:&quot;)[1]);</b>
&nbsp;                }
<b class="fc">&nbsp;            }</b>
<b class="fc">&nbsp;            cdna = cdnaSet.stream().collect(Collectors.toList());</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        return cdna;</b>
&nbsp;    }
&nbsp;
&nbsp;    @Nullable
&nbsp;    private String normalizeHgvsp(String hgvsp)
&nbsp;    {
<b class="fc">&nbsp;        if (hgvsp == null) {</b>
<b class="nc">&nbsp;            return null;</b>
&nbsp;        }
&nbsp;
<b class="fc">&nbsp;        int index = hgvsp.indexOf(&quot;:&quot;);</b>
&nbsp;
<b class="fc">&nbsp;        if (hgvsp.contains(&quot;(p.%3D)&quot;)) {</b>
<b class="nc">&nbsp;            return &quot;p.=&quot;;</b>
&nbsp;        }
&nbsp;        else {
<b class="fc">&nbsp;            return hgvsp.substring(index+1);</b>
&nbsp;        }
&nbsp;    }
&nbsp;
&nbsp;    
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2024-05-01 05:19</div>
</div>
</body>
</html>
