


<!DOCTYPE html>
<html id="htmlId">
<head>
  <title>Coverage Report > MutationAssessorMixin</title>
  <style type="text/css">
    @import "../../css/coverage.css";
    @import "../../css/idea.min.css";
  </style>
  <script type="text/javascript" src="../../js/highlight.min.js"></script>
  <script type="text/javascript" src="../../js/highlightjs-line-numbers.min.js"></script>
</head>

<body>
<div class="content">
<div class="breadCrumbs">
Current scope:     <a href="../../index.html">all classes</a>
    <span class="separator">|</span>
    <a href="../index.html">org.cbioportal.genome_nexus.web.mixin</a>
</div>

<h1>Coverage Summary for Class: MutationAssessorMixin (org.cbioportal.genome_nexus.web.mixin)</h1>

<table class="coverageStats">
<tr>
  <th class="name">Class</th>
<th class="coverageStat 
">
  Class, %
</th>
<th class="coverageStat 
">
  Method, %
</th>
<th class="coverageStat 
">
  Line, %
</th>
</tr>
<tr>
  <td class="name">MutationAssessorMixin</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/1)
  </span>
</td>
<td class="coverageStat">
  <span class="percent">
    0%
  </span>
  <span class="absValue">
    (0/1)
  </span>
</td>
</tr>

</table>

<br/>
<br/>


<pre>
<code class="sourceCode" id="sourceCode">&nbsp;package org.cbioportal.genome_nexus.web.mixin;
&nbsp;
&nbsp;import com.fasterxml.jackson.annotation.JsonInclude;
&nbsp;
&nbsp;import io.swagger.annotations.ApiModelProperty;
&nbsp;
&nbsp;@JsonInclude(JsonInclude.Include.NON_NULL)
<b class="nc">&nbsp;public class MutationAssessorMixin</b>
&nbsp;{
&nbsp;    @ApiModelProperty(value = &quot;User-input variants&quot;, required = true)
&nbsp;    private String input;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Hugo gene symbol&quot;, required = false)
&nbsp;    private String hugoSymbol;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Reference genome variant&quot;, required = false)
&nbsp;    private String referenceGenomeVariant;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Reference genome variant type&quot;, required = false)
&nbsp;    private String referenceGenomeVariantType;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Functional impact&quot;, required = false)
&nbsp;    private String functionalImpact;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Functional impact score&quot;, required = false)
&nbsp;    private double functionalImpactScore;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Link to multiple sequence alignment&quot;, required = false)
&nbsp;    private String msaLink;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Link to 3d structure browser&quot;, required = false)
&nbsp;    private String pdbLink;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Variant conservation score&quot;, required = false)
&nbsp;    private double variantConservationScore;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Variant specificity score&quot;, required = false)
&nbsp;    private double variantSpecificityScore;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Mapping issue info&quot;, required = false)
&nbsp;    private String mappingIssue;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Amino acid substitution&quot;, required = false)
&nbsp;    private String variant;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Uniprot protein accession ID&quot;, required = false)
&nbsp;    private String uniprotId;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Refseq protein ID&quot;, required = false)
&nbsp;    private String refseqId;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Portion of gaps in variant position in multiple sequence alignment&quot;, required = false)
&nbsp;    private double msaGaps;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Number of diverse sequences in multiple sequence alignment (identical or highly similar sequences filtered out)&quot;, required = false)
&nbsp;    private int msaHeight;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Codon start position&quot;, required = false)
&nbsp;    private String codonStartPosition;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Variant position in Uniprot protein, can be different from the one in Refseq&quot;, required = false)
&nbsp;    private int uniprotPosition;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Reference residue in Uniprot protein, can be different from the one in Refseq&quot;, required = false)
&nbsp;    private String uniprotResidue;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Variant position in Refseq protein, can be different from the one in Uniprot&quot;, required = false)
&nbsp;    private int refseqPosition;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Reference residue in Refseq protein, can be different from the one in Uniprot&quot;, required = false)
&nbsp;    private String refseqResidue;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Number of mutations in COSMIC for this protein&quot;, required = false)
&nbsp;    private int cosmicCount;
&nbsp;
&nbsp;    @ApiModelProperty(value = &quot;Number of SNPs in dbSNP for this protein&quot;, required = false)
&nbsp;    private int snpCount;
&nbsp;}
</code>
</pre>
</div>

<script type="text/javascript">
(function() {
    var msie = false, msie9 = false;
    /*@cc_on
      msie = true;
      @if (@_jscript_version >= 9)
        msie9 = true;
      @end
    @*/

    if (!msie || msie && msie9) {
      hljs.highlightAll()
      hljs.initLineNumbersOnLoad();
    }
})();
</script>

<div class="footer">
    
    <div style="float:right;">generated on 2024-05-01 05:19</div>
</div>
</body>
</html>
